<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="en-GB" xml:lang="en-GB"><head>

<meta charset="utf-8">
<meta name="generator" content="quarto-1.5.56">

<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes">

<meta name="author" content="Tormey Reimer">

<title>Asparagopsis parameteristation</title>
<style>
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
div.columns{display: flex; gap: min(4vw, 1.5em);}
div.column{flex: auto; overflow-x: auto;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
ul.task-list li input[type="checkbox"] {
  width: 0.8em;
  margin: 0 0.8em 0.2em -1em; /* quarto-specific, see https://github.com/quarto-dev/quarto-cli/issues/4556 */ 
  vertical-align: middle;
}
/* CSS for syntax highlighting */
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { display: inline-block; text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
  }
pre.numberSource { margin-left: 3em;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
/* CSS for citations */
div.csl-bib-body { }
div.csl-entry {
  clear: both;
  margin-bottom: 0em;
}
.hanging-indent div.csl-entry {
  margin-left:2em;
  text-indent:-2em;
}
div.csl-left-margin {
  min-width:2em;
  float:left;
}
div.csl-right-inline {
  margin-left:2em;
  padding-left:1em;
}
div.csl-indent {
  margin-left: 2em;
}</style>


<script src="asparagopsis-parameterisation_files/libs/clipboard/clipboard.min.js"></script>
<script src="asparagopsis-parameterisation_files/libs/quarto-html/quarto.js"></script>
<script src="asparagopsis-parameterisation_files/libs/quarto-html/popper.min.js"></script>
<script src="asparagopsis-parameterisation_files/libs/quarto-html/tippy.umd.min.js"></script>
<script src="asparagopsis-parameterisation_files/libs/quarto-html/anchor.min.js"></script>
<link href="asparagopsis-parameterisation_files/libs/quarto-html/tippy.css" rel="stylesheet">
<link href="asparagopsis-parameterisation_files/libs/quarto-html/quarto-syntax-highlighting.css" rel="stylesheet" id="quarto-text-highlighting-styles">
<script src="asparagopsis-parameterisation_files/libs/bootstrap/bootstrap.min.js"></script>
<link href="asparagopsis-parameterisation_files/libs/bootstrap/bootstrap-icons.css" rel="stylesheet">
<link href="asparagopsis-parameterisation_files/libs/bootstrap/bootstrap.min.css" rel="stylesheet" id="quarto-bootstrap" data-mode="light">

  <script src="https://cdnjs.cloudflare.com/polyfill/v3/polyfill.min.js?features=es6"></script>
  <script src="https://cdn.jsdelivr.net/npm/mathjax@3/es5/tex-chtml-full.js" type="text/javascript"></script>

<script type="text/javascript">
const typesetMath = (el) => {
  if (window.MathJax) {
    // MathJax Typeset
    window.MathJax.typeset([el]);
  } else if (window.katex) {
    // KaTeX Render
    var mathElements = el.getElementsByClassName("math");
    var macros = [];
    for (var i = 0; i < mathElements.length; i++) {
      var texText = mathElements[i].firstChild;
      if (mathElements[i].tagName == "SPAN") {
        window.katex.render(texText.data, mathElements[i], {
          displayMode: mathElements[i].classList.contains('display'),
          throwOnError: false,
          macros: macros,
          fleqn: false
        });
      }
    }
  }
}
window.Quarto = {
  typesetMath
};
</script>

<link rel="stylesheet" href="../resources/extra.css">
</head>

<body>

<div id="quarto-content" class="page-columns page-rows-contents page-layout-article toc-left">
<div id="quarto-sidebar-toc-left" class="sidebar toc-left">
  <nav id="TOC" role="doc-toc" class="toc-active" data-toc-expanded="99">
    <h2 id="toc-title">Table of contents</h2>
   
  <ul>
  <li><a href="#introduction" id="toc-introduction" class="nav-link active" data-scroll-target="#introduction">Introduction</a>
  <ul class="collapse">
  <li><a href="#phosphorus-limitation" id="toc-phosphorus-limitation" class="nav-link" data-scroll-target="#phosphorus-limitation">Phosphorus limitation</a></li>
  <li><a href="#still-to-investigate" id="toc-still-to-investigate" class="nav-link" data-scroll-target="#still-to-investigate">Still to investigate</a></li>
  </ul></li>
  <li><a href="#tetrasporophytes" id="toc-tetrasporophytes" class="nav-link" data-scroll-target="#tetrasporophytes">Tetrasporophytes</a>
  <ul class="collapse">
  <li><a href="#sec-nitrogen-uptake-tetrasporophyte" id="toc-sec-nitrogen-uptake-tetrasporophyte" class="nav-link" data-scroll-target="#sec-nitrogen-uptake-tetrasporophyte">Nitrogen uptake</a>
  <ul class="collapse">
  <li><a href="#sec-nitrogen-uptake-tetrasporophyte-linear" id="toc-sec-nitrogen-uptake-tetrasporophyte-linear" class="nav-link" data-scroll-target="#sec-nitrogen-uptake-tetrasporophyte-linear">Linear uptake</a></li>
  <li><a href="#sec-nitrogen-uptake-tetrasporophyte-MM" id="toc-sec-nitrogen-uptake-tetrasporophyte-MM" class="nav-link" data-scroll-target="#sec-nitrogen-uptake-tetrasporophyte-MM">Michaelis–Menten uptake</a></li>
  </ul></li>
  <li><a href="#sec-internal-nitrogen-tetrasporophyte" id="toc-sec-internal-nitrogen-tetrasporophyte" class="nav-link" data-scroll-target="#sec-internal-nitrogen-tetrasporophyte">Internal nutrient state</a>
  <ul class="collapse">
  <li><a href="#sec-nitrogen-concentration-tetrasporophyte" id="toc-sec-nitrogen-concentration-tetrasporophyte" class="nav-link" data-scroll-target="#sec-nitrogen-concentration-tetrasporophyte">Nitrogen concentration</a></li>
  </ul></li>
  <li><a href="#sec-salinity-tetrasporophyte" id="toc-sec-salinity-tetrasporophyte" class="nav-link" data-scroll-target="#sec-salinity-tetrasporophyte">Salinity</a></li>
  <li><a href="#biomass-factors" id="toc-biomass-factors" class="nav-link" data-scroll-target="#biomass-factors">Biomass factors</a>
  <ul class="collapse">
  <li><a href="#sec-dry-weight-tetrasporophyte" id="toc-sec-dry-weight-tetrasporophyte" class="nav-link" data-scroll-target="#sec-dry-weight-tetrasporophyte">Dry weight to fresh weight</a></li>
  <li><a href="#height" id="toc-height" class="nav-link" data-scroll-target="#height">Height</a></li>
  <li><a href="#growth-rate-stocking-density" id="toc-growth-rate-stocking-density" class="nav-link" data-scroll-target="#growth-rate-stocking-density">Growth rate, stocking density</a></li>
  </ul></li>
  <li><a href="#sec-temperature-tetrasporophyte" id="toc-sec-temperature-tetrasporophyte" class="nav-link" data-scroll-target="#sec-temperature-tetrasporophyte">Temperature response</a></li>
  <li><a href="#sec-light-tetrasporophyte" id="toc-sec-light-tetrasporophyte" class="nav-link" data-scroll-target="#sec-light-tetrasporophyte">Light response</a></li>
  <li><a href="#fragmentation-with-exposure" id="toc-fragmentation-with-exposure" class="nav-link" data-scroll-target="#fragmentation-with-exposure">Fragmentation with exposure</a></li>
  </ul></li>
  <li><a href="#finishing-up" id="toc-finishing-up" class="nav-link" data-scroll-target="#finishing-up">Finishing up</a>
  <ul class="collapse">
  <li><a href="#sec-final-params-tetrasporophyte" id="toc-sec-final-params-tetrasporophyte" class="nav-link" data-scroll-target="#sec-final-params-tetrasporophyte">All parameters and sources</a></li>
  <li><a href="#sec-save-tetrasporophyte" id="toc-sec-save-tetrasporophyte" class="nav-link" data-scroll-target="#sec-save-tetrasporophyte">Saving data</a></li>
  </ul></li>
  <li><a href="#gametophytes" id="toc-gametophytes" class="nav-link" data-scroll-target="#gametophytes">Gametophytes</a>
  <ul class="collapse">
  <li><a href="#sec-nitrogen-uptake-gametophyte" id="toc-sec-nitrogen-uptake-gametophyte" class="nav-link" data-scroll-target="#sec-nitrogen-uptake-gametophyte">Nitrogen uptake</a></li>
  <li><a href="#sec-internal-nitrogen-gametophyte" id="toc-sec-internal-nitrogen-gametophyte" class="nav-link" data-scroll-target="#sec-internal-nitrogen-gametophyte">Internal nutrient state</a></li>
  <li><a href="#sec-salinity-gametophyte" id="toc-sec-salinity-gametophyte" class="nav-link" data-scroll-target="#sec-salinity-gametophyte">Salinity</a></li>
  <li><a href="#biomass-factors-1" id="toc-biomass-factors-1" class="nav-link" data-scroll-target="#biomass-factors-1">Biomass factors</a>
  <ul class="collapse">
  <li><a href="#sec-dry-weight-gametophyte" id="toc-sec-dry-weight-gametophyte" class="nav-link" data-scroll-target="#sec-dry-weight-gametophyte">Dry weight to fresh weight</a></li>
  <li><a href="#sec-height-gametophyte" id="toc-sec-height-gametophyte" class="nav-link" data-scroll-target="#sec-height-gametophyte">Height</a></li>
  <li><a href="#sec-growth-rate-gametophyte" id="toc-sec-growth-rate-gametophyte" class="nav-link" data-scroll-target="#sec-growth-rate-gametophyte">Growth rate, stocking density</a></li>
  </ul></li>
  <li><a href="#sec-temperature-gametophyte" id="toc-sec-temperature-gametophyte" class="nav-link" data-scroll-target="#sec-temperature-gametophyte">Temperature response</a></li>
  <li><a href="#light-response" id="toc-light-response" class="nav-link" data-scroll-target="#light-response">Light response</a></li>
  <li><a href="#fragmentation-with-exposure-1" id="toc-fragmentation-with-exposure-1" class="nav-link" data-scroll-target="#fragmentation-with-exposure-1">Fragmentation with exposure</a>
  <ul class="collapse">
  <li><a href="#velocity" id="toc-velocity" class="nav-link" data-scroll-target="#velocity">Velocity</a></li>
  <li><a href="#turbulence" id="toc-turbulence" class="nav-link" data-scroll-target="#turbulence">Turbulence</a></li>
  </ul></li>
  </ul></li>
  <li><a href="#finishing-up-1" id="toc-finishing-up-1" class="nav-link" data-scroll-target="#finishing-up-1">Finishing up</a>
  <ul class="collapse">
  <li><a href="#sec-final-params-gametophyte" id="toc-sec-final-params-gametophyte" class="nav-link" data-scroll-target="#sec-final-params-gametophyte">All parameters and sources</a></li>
  <li><a href="#sec-save-gametophyte" id="toc-sec-save-gametophyte" class="nav-link" data-scroll-target="#sec-save-gametophyte">Saving data</a></li>
  </ul></li>
  <li><a href="#potential-future-extensions" id="toc-potential-future-extensions" class="nav-link" data-scroll-target="#potential-future-extensions">Potential future extensions</a></li>
  <li><a href="#code" id="toc-code" class="nav-link" data-scroll-target="#code">Code</a></li>
  <li><a href="#references" id="toc-references" class="nav-link" data-scroll-target="#references">References</a></li>
  </ul>
</nav>
</div>
<div id="quarto-margin-sidebar" class="sidebar margin-sidebar zindex-bottom">
</div>
<main class="content page-columns page-full" id="quarto-document-content">

<header id="title-block-header" class="quarto-title-block default">
<div class="quarto-title">
<h1 class="title">Asparagopsis parameteristation</h1>
</div>



<div class="quarto-title-meta">

    <div>
    <div class="quarto-title-meta-heading">Author</div>
    <div class="quarto-title-meta-contents">
             <p>Tormey Reimer </p>
          </div>
  </div>
    
  
    <div>
    <div class="quarto-title-meta-heading">Modified</div>
    <div class="quarto-title-meta-contents">
      <p class="date-modified">27 March 2025</p>
    </div>
  </div>
    
  </div>
  


</header>


<div class="cell">
<details class="code-fold">
<summary>All the code in this markdown is set to ‘fold’ by default. Click each folded code chunk to see its inner workings.</summary>
<div class="sourceCode cell-code" id="cb1"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb1-1"><a href="#cb1-1" aria-hidden="true" tabindex="-1"></a><span class="co"># You did it!</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<section id="introduction" class="level1">
<h1>Introduction</h1>
<div class="cell">
<details class="code-fold">
<summary>Some global defaults I like to use</summary>
<div class="sourceCode cell-code" id="cb2"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb2-1"><a href="#cb2-1" aria-hidden="true" tabindex="-1"></a>base_path <span class="ot">&lt;-</span> <span class="fu">file.path</span>(<span class="st">"modelling-asparagopsis"</span>, <span class="st">"FRDC-seaweed"</span>)</span>
<span id="cb2-2"><a href="#cb2-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-3"><a href="#cb2-3" aria-hidden="true" tabindex="-1"></a>prettyplot <span class="ot">&lt;-</span> </span>
<span id="cb2-4"><a href="#cb2-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_classic</span>() <span class="sc">+</span></span>
<span id="cb2-5"><a href="#cb2-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="fu">element_text</span>(<span class="at">family =</span> <span class="st">"sans"</span>, <span class="at">size =</span> <span class="dv">12</span>, <span class="at">colour =</span> <span class="st">"black"</span>),</span>
<span id="cb2-6"><a href="#cb2-6" aria-hidden="true" tabindex="-1"></a>        <span class="at">axis.title.y =</span> <span class="fu">element_text</span>(<span class="at">vjust =</span> <span class="dv">2</span>),</span>
<span id="cb2-7"><a href="#cb2-7" aria-hidden="true" tabindex="-1"></a>        <span class="at">legend.position =</span> <span class="st">"none"</span>)</span>
<span id="cb2-8"><a href="#cb2-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-9"><a href="#cb2-9" aria-hidden="true" tabindex="-1"></a><span class="fu">remove_unit</span>(<span class="st">"gDW"</span>)</span>
<span id="cb2-10"><a href="#cb2-10" aria-hidden="true" tabindex="-1"></a><span class="fu">install_unit</span>(<span class="st">"gDW"</span>)</span>
<span id="cb2-11"><a href="#cb2-11" aria-hidden="true" tabindex="-1"></a><span class="fu">remove_unit</span>(<span class="st">"gFW"</span>)</span>
<span id="cb2-12"><a href="#cb2-12" aria-hidden="true" tabindex="-1"></a><span class="fu">install_unit</span>(<span class="st">"gFW"</span>)</span>
<span id="cb2-13"><a href="#cb2-13" aria-hidden="true" tabindex="-1"></a><span class="fu">remove_unit</span>(<span class="st">"photons"</span>)</span>
<span id="cb2-14"><a href="#cb2-14" aria-hidden="true" tabindex="-1"></a><span class="fu">install_unit</span>(<span class="st">"photons"</span>)</span>
<span id="cb2-15"><a href="#cb2-15" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-16"><a href="#cb2-16" aria-hidden="true" tabindex="-1"></a><span class="co"># Only works because I'm always working in nitrogen</span></span>
<span id="cb2-17"><a href="#cb2-17" aria-hidden="true" tabindex="-1"></a><span class="fu">remove_unit</span>(<span class="st">"mol"</span>)</span>
<span id="cb2-18"><a href="#cb2-18" aria-hidden="true" tabindex="-1"></a><span class="fu">install_unit</span>(<span class="st">"mol"</span>, <span class="st">"14.0067 g"</span>)</span>
<span id="cb2-19"><a href="#cb2-19" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-20"><a href="#cb2-20" aria-hidden="true" tabindex="-1"></a>S_uM <span class="ot">&lt;-</span> <span class="fu">set_units</span>(<span class="fu">seq</span>(<span class="dv">0</span>, <span class="dv">200</span>, <span class="fl">0.5</span>), <span class="st">"umol L-1"</span>)</span>
<span id="cb2-21"><a href="#cb2-21" aria-hidden="true" tabindex="-1"></a>S_mg <span class="ot">&lt;-</span> <span class="fu">set_units</span>(S_uM, <span class="st">"mg m-3"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<p>The <em>Asparagopsis</em> genus (and especially <em>A. armata</em>) is invasive in the northeastern Atlantic and the Mediterranean <span class="citation" data-cites="zanolla_concise_2022">(<a href="#ref-zanolla_concise_2022" role="doc-biblioref">Zanolla et al. 2022</a>)</span>. It has been studied more as an invasive genus than a native one. There are two species (or species complexes) within the genus <em>Asparagopsis</em>: <em>A. armata</em> is a temperate species while <em>A. taxiformis</em> is more commonly found in tropical areas (including northern Australia), and this might affect the results of the modelling - I may need to parameterise two different species, possibly distinct only in their temperature/light tolerance.</p>
<p>Phylogenetic analyses suggest that there is a clade within the species <em>A. armata</em> restricted only to Western Australia, Tasmania and New Zealand <span class="citation" data-cites="dijoux_more_2014">(<a href="#ref-dijoux_more_2014" role="doc-biblioref">Dijoux, Viard, and Payri 2014</a>)</span>. It has been proposed that the species <em>A. armata</em> is actually two cryptic species: <em>A. armata</em> 1 is native to Tasmania and is invasive in the northern hemisphere, while <em>A. armata</em> 2 is restricted to Western Australia, Tasmania and New Zealand <span class="citation" data-cites="dijoux_more_2014 zanolla_concise_2022">(<a href="#ref-dijoux_more_2014" role="doc-biblioref">Dijoux, Viard, and Payri 2014</a>; <a href="#ref-zanolla_concise_2022" role="doc-biblioref">Zanolla et al. 2022</a>)</span>. Therefore, while the <em>A. armata</em> studied as an invasive species (<em>A. armata</em> 1) is partially representative of the Australian species, the lack of sampling in Australia and New Zealand may be leaving knowledge gaps of the potentially more widespread <em>A. armata</em> 2. It’s not known how much will this affect environmental growth responses in a model, given that the genus is very plastic in its environmental tolerance <span class="citation" data-cites="zanolla_concise_2022">(<a href="#ref-zanolla_concise_2022" role="doc-biblioref">Zanolla et al. 2022</a>)</span>.</p>
<p>Both species follow a triphasic lifestyle, alternative between macroscopic diploid tetrasporophyte and haploid (male and female) gametophyte phases (<a href="#fig-morphology" class="quarto-xref">Figure&nbsp;1</a>). Gametophytes produce gametes for sexual reproduction while tetrasporophytes produce tetrasporangia for asexual reproduction, and both phases additionally reproduce asexually through fragmentation <span class="citation" data-cites="zanolla_concise_2022">(<a href="#ref-zanolla_concise_2022" role="doc-biblioref">Zanolla et al. 2022</a>)</span>. The phases are easy to distinguish morphologically - gametophytes look somewhat like asparagus shoots (giving the genus name) and grow as branching shoots up to ~30 cm long. Tetrasporophytes are much smaller with finer fronds, resembling pom-poms []. Tetrasporophytes are usually found year-round, while gametophytes are often only found in colder months <span class="citation" data-cites="zanolla_size_2018">(<a href="#ref-zanolla_size_2018" role="doc-biblioref">Zanolla, Altamirano, De La Rosa, et al. 2018</a>)</span>. The phases may differ in their responses to environmental temperature (particularly temperature and light) and knowing the phases is important because some studies do not specify which phase of <em>Asparagopsis</em> they have collected or are working with.</p>
<div id="fig-morphology" class="quarto-layout-panel">
<figure class="quarto-float quarto-float-fig figure">
<div aria-describedby="fig-morphology-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<div class="quarto-layout-row">
<div class="quarto-layout-cell" style="flex-basis: 50.0%;justify-content: center;">
<p><img src="images/10811_2021_2665_Fig1_HTML.jpg" class="img-fluid figure-img"></p>
</div>
<div class="quarto-layout-cell" style="flex-basis: 50.0%;justify-content: center;">
<p><img src="images/10811_2021_2665_Fig2_HTML.jpg" class="img-fluid figure-img"></p>
</div>
</div>
</div>
<figcaption class="quarto-float-caption-bottom quarto-float-caption quarto-float-fig" id="fig-morphology-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
Figure&nbsp;1: Left: Gametophytes of <em>Asparagopsis armata</em> (above) and <em>A. taxiformis</em> (below) at 5 m depth. Arrows point to harpoon-like branches in <em>A. armata</em>. Right: Tetrasporophyte of <em>Asparagopsis</em> at 3-m depth, presenting pompom structure and attached to other algae. Collected from Granada, Spain. Photo credits: J. De la Rosa, reproduced from <span class="citation" data-cites="zanolla_concise_2022">Zanolla et al. (<a href="#ref-zanolla_concise_2022" role="doc-biblioref">2022</a>)</span>.
</figcaption>
</figure>
</div>
<p>The different forms may also be very physiologically different in some respects - at the very least with regard to temperature and daylength. Catriona Hurd also thinks it’s likely they’ll be different in their uptake rates of the different forms of nitrogen.</p>
<section id="phosphorus-limitation" class="level2">
<h2 class="anchored" data-anchor-id="phosphorus-limitation">Phosphorus limitation</h2>
<p>It has been suggested that the model should include the possibility of phosphorus limitation, as well as nitrogen limitation (switching between the limiting nutrient based on availability/internal conditions). This was considered because the places where macroalgae is being grown for nitrogen bioremediation purposes (river outflows, next to fish farms) are likely to be very high in nitrogen but not necessarily phosphates.</p>
<p>However, it was decided not to include phosphorus limitation. While the mechanics of including it would be fairly simple, research into the required parameters (shape and rate of uptake, internal requirements) are almost non-existent for <em>Asparagopsis</em>. <span class="citation" data-cites="grisenthwaite_establishing_2023">Grisenthwaite (<a href="#ref-grisenthwaite_establishing_2023" role="doc-biblioref">2023</a>)</span> may possibly have some of this data for tetrasporophytes but I have yet to find anything similar for gametophytes. If the ‘standard’, not species-specific Atkinson ratio of N:P were assumed <span class="citation" data-cites="atkinson_cnp_1983">(<a href="#ref-atkinson_cnp_1983" role="doc-biblioref">Atkinson and Smith 1983</a>)</span>, then the algae would require 1/30<sup>th</sup> as much phosphorus as nitrogen to maintain the modelled growth. Given that macroalgae require so much more nitrogen than phosphorus, it seems unlikely that they are ever going to be phosphorus limited in coastal or oceanic waters. Phosphates are also not often measured in marine waters, so obtaining ambient data would be extremely difficult.</p>
<p>Still, if the Atkinson ratio (30:1 N:P) were assumed and phosphate uptake kinetics were ignored, then the minimum phosphate concentrations required to maintain the modelled growth can be estimated and reported as an additional consideration.</p>
</section>
<section id="still-to-investigate" class="level2">
<h2 class="anchored" data-anchor-id="still-to-investigate">Still to investigate</h2>
<ul>
<li><span class="citation" data-cites="chualain_invasive_2004">Chualáin et al. (<a href="#ref-chualain_invasive_2004" role="doc-biblioref">2004</a>)</span></li>
<li><span class="citation" data-cites="cole_using_2014">Cole et al. (<a href="#ref-cole_using_2014" role="doc-biblioref">2014</a>)</span></li>
<li><span class="citation" data-cites="dishon_effect_2023">Dishon et al. (<a href="#ref-dishon_effect_2023" role="doc-biblioref">2023</a>)</span></li>
<li><span class="citation" data-cites="kraan_commercial_2005">Kraan and Barrington (<a href="#ref-kraan_commercial_2005" role="doc-biblioref">2005</a>)</span></li>
<li><span class="citation" data-cites="mata_integrated_2008">Mata (<a href="#ref-mata_integrated_2008" role="doc-biblioref">2008</a>)</span></li>
<li><span class="citation" data-cites="mata_within-species_2017">Mata et al. (<a href="#ref-mata_within-species_2017" role="doc-biblioref">2017</a>)</span></li>
<li><span class="citation" data-cites="mihaila_moderate_2024">Mihaila et al. (<a href="#ref-mihaila_moderate_2024" role="doc-biblioref">2024</a>)</span></li>
<li><span class="citation" data-cites="monro_evolutionary_2007">Monro (<a href="#ref-monro_evolutionary_2007" role="doc-biblioref">2007</a>)</span></li>
<li><span class="citation" data-cites="padilla-gamino_seasonal_2007">Padilla-Gamiño and Carpenter (<a href="#ref-padilla-gamino_seasonal_2007" role="doc-biblioref">2007</a>)</span></li>
<li><span class="citation" data-cites="schuenhoff_tetrasporophyte_2006">Schuenhoff, Mata, and Santos (<a href="#ref-schuenhoff_tetrasporophyte_2006" role="doc-biblioref">2006</a>)</span></li>
<li><span class="citation" data-cites="torres_nitrogen_2021">Torres et al. (<a href="#ref-torres_nitrogen_2021" role="doc-biblioref">2021</a>)</span></li>
<li><span class="citation" data-cites="torres_effects_2024">Torres et al. (<a href="#ref-torres_effects_2024" role="doc-biblioref">2024</a>)</span></li>
<li><span class="citation" data-cites="wiltshire_feasibility_2015">Wiltshire et al. (<a href="#ref-wiltshire_feasibility_2015" role="doc-biblioref">2015</a>)</span></li>
<li><span class="citation" data-cites="zanolla_assessing_2018">Zanolla, Altamirano, Carmona, et al. (<a href="#ref-zanolla_assessing_2018" role="doc-biblioref">2018</a>)</span></li>
<li><span class="citation" data-cites="zanolla_there_2019">Zanolla et al. (<a href="#ref-zanolla_there_2019" role="doc-biblioref">2019</a>)</span></li>
<li><span class="citation" data-cites="zanolla_concise_2022">Zanolla et al. (<a href="#ref-zanolla_concise_2022" role="doc-biblioref">2022</a>)</span></li>
</ul>
</section>
</section>
<section id="tetrasporophytes" class="level1">
<h1>Tetrasporophytes</h1>
<section id="sec-nitrogen-uptake-tetrasporophyte" class="level2">
<h2 class="anchored" data-anchor-id="sec-nitrogen-uptake-tetrasporophyte">Nitrogen uptake</h2>
<section id="sec-nitrogen-uptake-tetrasporophyte-linear" class="level3">
<h3 class="anchored" data-anchor-id="sec-nitrogen-uptake-tetrasporophyte-linear">Linear uptake</h3>
<p>Some very interesting things from reading <span class="citation" data-cites="torres_nitrogen_2021">Torres et al. (<a href="#ref-torres_nitrogen_2021" role="doc-biblioref">2021</a>)</span>:</p>
<ul>
<li>Asparagopsis prefers ammonium during the internally-controlled (non-surge) phase</li>
<li>Nitrate also taken up, but not preferred - even when nitrate was the only form available uptake was limited (limited use by seaweed)</li>
<li>Amino acids (a mix of several different ones, with alanine dominating) were also tested. Amino acids are released by decomposing fish food. I may need to include amino acids as “other N” in the model.</li>
</ul>
<p><span class="citation" data-cites="torres_nitrogen_2021">Torres et al. (<a href="#ref-torres_nitrogen_2021" role="doc-biblioref">2021</a>)</span> tested the surge and internally-controlled nitrogen uptake rates of <em>A. armata</em> when given nitrogen in the form of ammonium, nitrate and a mix of amino acids (dominated by alanine). Nitrogen uptake (in all three forms) did not saturate at ~200 <span class="math inline">\(\mu\)</span>M concentration, so was was better described with a linear equation rather than with Michaelis-Menton uptake kinetics</p>
<div class="cell">
<details class="code-fold">
<summary>Import Torres et al. (2021) data</summary>
<div class="sourceCode cell-code" id="cb3"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb3-1"><a href="#cb3-1" aria-hidden="true" tabindex="-1"></a>torresNO3 <span class="ot">&lt;-</span> <span class="fu">read.csv</span>(<span class="fu">file.path</span>(<span class="st">"data_raw"</span>, <span class="st">"sources"</span>, <span class="st">"torres_2021"</span>, <span class="st">"tores-2021-Fig2-B.csv"</span>))</span>
<span id="cb3-2"><a href="#cb3-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-3"><a href="#cb3-3" aria-hidden="true" tabindex="-1"></a>C_ni_alone <span class="ot">&lt;-</span> <span class="fu">set_units</span>(<span class="fu">mean</span>(torresNO3<span class="sc">$</span>uptake[torresNO3<span class="sc">$</span>treatment <span class="sc">==</span> <span class="st">"alone"</span>]), <span class="st">"umol gDW-1 h-1"</span>)</span>
<span id="cb3-4"><a href="#cb3-4" aria-hidden="true" tabindex="-1"></a>C_ni_combo <span class="ot">&lt;-</span> <span class="fu">set_units</span>(<span class="fu">mean</span>(torresNO3<span class="sc">$</span>uptake[torresNO3<span class="sc">$</span>treatment <span class="sc">==</span> <span class="st">"combo"</span>]), <span class="st">"umol gDW-1 h-1"</span>)</span>
<span id="cb3-5"><a href="#cb3-5" aria-hidden="true" tabindex="-1"></a>N_15 <span class="ot">&lt;-</span> <span class="fu">set_units</span>(<span class="dv">15</span>, g<span class="sc">/</span>mol) <span class="co"># Torres 2021 used N-15 only</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<p><span class="citation" data-cites="torres_nitrogen_2021">Torres et al. (<a href="#ref-torres_nitrogen_2021" role="doc-biblioref">2021</a>)</span> did not calculate a linear uptake rate for nitrate when it was tested alone. They did not offer an explanation beyond the observation that since <em>A. armata</em> did not show an increase in uptake rate with concentration it likely has a limited capacity to use nitrogen. However this does not explain why they provided a slope and constant for linear uptake of nitrate in combination with other N forms, as that also did not show an increase in uptake rate with increasing concentration (although the relationship was not significant). I have decided to use a constant rate of nitrate uptake independent of concentration.</p>
<div class="cell">
<details class="code-fold">
<summary>Calculate uptake rates from data</summary>
<div class="sourceCode cell-code" id="cb4"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb4-1"><a href="#cb4-1" aria-hidden="true" tabindex="-1"></a><span class="co"># From Torres et. al. 2021, Table 3</span></span>
<span id="cb4-2"><a href="#cb4-2" aria-hidden="true" tabindex="-1"></a>M_am_alone <span class="ot">&lt;-</span> <span class="fu">set_units</span>(<span class="fl">0.12</span>, <span class="st">"L h-1 gDW-1"</span>) </span>
<span id="cb4-3"><a href="#cb4-3" aria-hidden="true" tabindex="-1"></a>C_am_alone <span class="ot">&lt;-</span> <span class="fu">set_units</span>(<span class="fl">10.45</span>, <span class="st">"umol h-1 gDW-1"</span>)</span>
<span id="cb4-4"><a href="#cb4-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-5"><a href="#cb4-5" aria-hidden="true" tabindex="-1"></a>M_am_combo <span class="ot">&lt;-</span> <span class="fu">set_units</span>(<span class="fl">0.14</span>, <span class="st">"L h-1 gDW-1"</span>)</span>
<span id="cb4-6"><a href="#cb4-6" aria-hidden="true" tabindex="-1"></a>C_am_combo <span class="ot">&lt;-</span> <span class="fu">set_units</span>(<span class="fl">8.79</span>, <span class="st">"umol h-1 gDW-1 "</span>)</span>
<span id="cb4-7"><a href="#cb4-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-8"><a href="#cb4-8" aria-hidden="true" tabindex="-1"></a>M_aa_alone <span class="ot">&lt;-</span> <span class="fu">set_units</span>(<span class="fl">0.007</span>, <span class="st">"L h-1 gDW-1"</span>)</span>
<span id="cb4-9"><a href="#cb4-9" aria-hidden="true" tabindex="-1"></a>C_aa_alone <span class="ot">&lt;-</span> <span class="fu">set_units</span>(<span class="fl">1.69</span>, <span class="st">"umol h-1 gDW-1 "</span>)</span>
<span id="cb4-10"><a href="#cb4-10" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-11"><a href="#cb4-11" aria-hidden="true" tabindex="-1"></a>M_aa_combo <span class="ot">&lt;-</span> <span class="fu">set_units</span>(<span class="fl">0.006</span>, <span class="st">"L h-1 gDW-1"</span>)</span>
<span id="cb4-12"><a href="#cb4-12" aria-hidden="true" tabindex="-1"></a>C_aa_combo <span class="ot">&lt;-</span> <span class="fu">set_units</span>(<span class="fl">0.88</span>, <span class="st">"umol h-1 gDW-1"</span>)</span>
<span id="cb4-13"><a href="#cb4-13" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-14"><a href="#cb4-14" aria-hidden="true" tabindex="-1"></a>df <span class="ot">&lt;-</span> <span class="fu">list</span>(</span>
<span id="cb4-15"><a href="#cb4-15" aria-hidden="true" tabindex="-1"></a>  <span class="fu">as.data.frame</span>(S_uM) <span class="sc">%&gt;%</span> </span>
<span id="cb4-16"><a href="#cb4-16" aria-hidden="true" tabindex="-1"></a>    <span class="fu">mutate</span>(<span class="at">alone =</span> <span class="fu">lin_uptake</span>(S_uM, M_am_alone, C_am_alone),</span>
<span id="cb4-17"><a href="#cb4-17" aria-hidden="true" tabindex="-1"></a>           <span class="at">combo =</span> <span class="fu">lin_uptake</span>(S_uM, M_am_combo, C_am_combo),</span>
<span id="cb4-18"><a href="#cb4-18" aria-hidden="true" tabindex="-1"></a>           <span class="at">form =</span> <span class="st">"ammonium"</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb4-19"><a href="#cb4-19" aria-hidden="true" tabindex="-1"></a>    <span class="fu">pivot_longer</span>(<span class="at">names_to =</span> <span class="st">"treatment"</span>, <span class="at">values_to =</span> <span class="st">"uptake_rate"</span>, <span class="at">cols =</span> <span class="fu">c</span>(alone, combo)),</span>
<span id="cb4-20"><a href="#cb4-20" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb4-21"><a href="#cb4-21" aria-hidden="true" tabindex="-1"></a>  <span class="fu">as.data.frame</span>(S_uM) <span class="sc">%&gt;%</span> </span>
<span id="cb4-22"><a href="#cb4-22" aria-hidden="true" tabindex="-1"></a>    <span class="fu">mutate</span>(<span class="at">alone =</span> C_ni_alone,</span>
<span id="cb4-23"><a href="#cb4-23" aria-hidden="true" tabindex="-1"></a>           <span class="at">combo =</span> C_ni_combo,</span>
<span id="cb4-24"><a href="#cb4-24" aria-hidden="true" tabindex="-1"></a>           <span class="at">form =</span> <span class="st">"nitrate"</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb4-25"><a href="#cb4-25" aria-hidden="true" tabindex="-1"></a>    <span class="fu">pivot_longer</span>(<span class="at">names_to =</span> <span class="st">"treatment"</span>, <span class="at">values_to =</span> <span class="st">"uptake_rate"</span>, <span class="at">cols =</span> <span class="fu">c</span>(alone, combo)),</span>
<span id="cb4-26"><a href="#cb4-26" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb4-27"><a href="#cb4-27" aria-hidden="true" tabindex="-1"></a>  <span class="fu">as.data.frame</span>(S_uM) <span class="sc">%&gt;%</span> </span>
<span id="cb4-28"><a href="#cb4-28" aria-hidden="true" tabindex="-1"></a>    <span class="fu">mutate</span>(<span class="at">alone =</span> <span class="fu">lin_uptake</span>(S_uM, M_aa_alone, C_aa_alone),</span>
<span id="cb4-29"><a href="#cb4-29" aria-hidden="true" tabindex="-1"></a>           <span class="at">combo =</span> <span class="fu">lin_uptake</span>(S_uM, M_aa_combo, C_aa_combo),</span>
<span id="cb4-30"><a href="#cb4-30" aria-hidden="true" tabindex="-1"></a>           <span class="at">form =</span> <span class="st">"amino acids"</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb4-31"><a href="#cb4-31" aria-hidden="true" tabindex="-1"></a>    <span class="fu">pivot_longer</span>(<span class="at">names_to =</span> <span class="st">"treatment"</span>, <span class="at">values_to =</span> <span class="st">"uptake_rate"</span>, <span class="at">cols =</span> <span class="fu">c</span>(alone, combo))</span>
<span id="cb4-32"><a href="#cb4-32" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb4-33"><a href="#cb4-33" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-34"><a href="#cb4-34" aria-hidden="true" tabindex="-1"></a>df <span class="ot">&lt;-</span> <span class="fu">bind_rows</span>(df) <span class="sc">%&gt;%</span> </span>
<span id="cb4-35"><a href="#cb4-35" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">form =</span> <span class="fu">as.factor</span>(form),</span>
<span id="cb4-36"><a href="#cb4-36" aria-hidden="true" tabindex="-1"></a>         <span class="at">treatment =</span> <span class="fu">as.factor</span>(treatment),</span>
<span id="cb4-37"><a href="#cb4-37" aria-hidden="true" tabindex="-1"></a>         <span class="at">S =</span> <span class="fu">set_units</span>(S_uM<span class="sc">*</span>N_15, <span class="st">"mg m-3"</span>),</span>
<span id="cb4-38"><a href="#cb4-38" aria-hidden="true" tabindex="-1"></a>         <span class="at">uptake_rate =</span> <span class="fu">set_units</span>(uptake_rate<span class="sc">*</span>N_15, <span class="st">"mg gDW-1 d-1"</span>))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<p>Ammonium is obviously the preferred form of nitrogen uptake in <em>A. armata</em>. Amino acids and nitrate are taken up at approximately the same rate.</p>
<div class="cell">
<details class="code-fold">
<summary>Plot N uptake based on Torres et al. (2021) data</summary>
<div class="sourceCode cell-code" id="cb5"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb5-1"><a href="#cb5-1" aria-hidden="true" tabindex="-1"></a>int1 <span class="ot">&lt;-</span> <span class="fu">set_units</span>(<span class="dv">20</span>, <span class="st">"umol L-1"</span>)</span>
<span id="cb5-2"><a href="#cb5-2" aria-hidden="true" tabindex="-1"></a>int2 <span class="ot">&lt;-</span> <span class="fu">set_units</span>(<span class="dv">50</span>, <span class="st">"umol L-1"</span>)</span>
<span id="cb5-3"><a href="#cb5-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-4"><a href="#cb5-4" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>(df, <span class="fu">aes</span>(<span class="at">x =</span> S, <span class="at">y =</span> uptake_rate, <span class="at">colour =</span> form)) <span class="sc">+</span></span>
<span id="cb5-5"><a href="#cb5-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_line</span>() <span class="sc">+</span></span>
<span id="cb5-6"><a href="#cb5-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">facet_wrap</span>(<span class="fu">vars</span>(df<span class="sc">$</span>treatment)) <span class="sc">+</span></span>
<span id="cb5-7"><a href="#cb5-7" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_vline</span>(<span class="fu">aes</span>(<span class="at">xintercept =</span> <span class="fu">set_units</span>(int1, <span class="st">"mg m-3"</span>)), <span class="at">linetype =</span> <span class="st">"dashed"</span>) <span class="sc">+</span></span>
<span id="cb5-8"><a href="#cb5-8" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_vline</span>(<span class="fu">aes</span>(<span class="at">xintercept =</span> <span class="fu">set_units</span>(int2, <span class="st">"mg m-3"</span>)), <span class="at">linetype =</span> <span class="st">"dashed"</span>) <span class="sc">+</span></span>
<span id="cb5-9"><a href="#cb5-9" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">y =</span> <span class="st">"Uptake rate"</span>, <span class="at">x =</span> <span class="st">"Substrate concentration"</span>) <span class="sc">+</span></span>
<span id="cb5-10"><a href="#cb5-10" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_classic</span>() <span class="sc">+</span></span>
<span id="cb5-11"><a href="#cb5-11" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">legend.position =</span> <span class="st">"none"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><img src="asparagopsis-parameterisation_files/figure-html/torres%202021%20plot-1.png" class="img-fluid figure-img" width="672"></p>
<figcaption>Linear uptake of ammonium (green), nitrate (blue) and amino acids (red) by <em>A. armata</em>. Dashed black lines show concentrations of 20 <span class="math inline">\(\mu\)</span>M and 50 <span class="math inline">\(\mu\)</span>M for reference.</figcaption>
</figure>
</div>
</div>
</div>
</section>
<section id="sec-nitrogen-uptake-tetrasporophyte-MM" class="level3">
<h3 class="anchored" data-anchor-id="sec-nitrogen-uptake-tetrasporophyte-MM">Michaelis–Menten uptake</h3>
<p><span class="citation" data-cites="schuenhoff_tetrasporophyte_2006">Schuenhoff, Mata, and Santos (<a href="#ref-schuenhoff_tetrasporophyte_2006" role="doc-biblioref">2006</a>)</span> fit a Michaelis–Menten curve to their uptake rates. They measured day and night uptake rates separately and combined. Their night and combined measurements were fairly poorly described by a Michaelis–Menten curve (R = 0.189 and 0.21 respectively) but daytime uptake was better (R<span class="math inline">\(^2\)</span> = 0.78). Since the model doesn’t have a day-night cycle anyway, I’ll be using the daytime parameters only.</p>
<div class="cell">
<details class="code-fold">
<summary>Plot curves based on Schuenhoff et al. (2006) data</summary>
<div class="sourceCode cell-code" id="cb6"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb6-1"><a href="#cb6-1" aria-hidden="true" tabindex="-1"></a>V_am <span class="ot">&lt;-</span> <span class="fu">set_units</span>(<span class="fl">14.85</span>, <span class="st">"umol gDW-1 h-1"</span>)</span>
<span id="cb6-2"><a href="#cb6-2" aria-hidden="true" tabindex="-1"></a>K_am <span class="ot">&lt;-</span> <span class="fu">set_units</span>(<span class="fl">10.01</span>, <span class="st">"umol L-1"</span>)</span>
<span id="cb6-3"><a href="#cb6-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb6-4"><a href="#cb6-4" aria-hidden="true" tabindex="-1"></a>df2 <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="at">S =</span> S_uM, <span class="at">uptake_rate =</span> <span class="fu">MM_uptake</span>(<span class="at">conc =</span> S_uM, <span class="at">V =</span> V_am, <span class="at">K =</span> K_am))</span>
<span id="cb6-5"><a href="#cb6-5" aria-hidden="true" tabindex="-1"></a>df2 <span class="ot">&lt;-</span> df2 <span class="sc">%&gt;%</span> </span>
<span id="cb6-6"><a href="#cb6-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">S =</span> <span class="fu">set_units</span>(S_uM, <span class="st">"mg m-3"</span>),</span>
<span id="cb6-7"><a href="#cb6-7" aria-hidden="true" tabindex="-1"></a>         <span class="at">uptake_rate =</span> <span class="fu">set_units</span>(uptake_rate, <span class="st">"mg gDW-1 d-1"</span>))</span>
<span id="cb6-8"><a href="#cb6-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb6-9"><a href="#cb6-9" aria-hidden="true" tabindex="-1"></a>int1 <span class="ot">&lt;-</span> <span class="fu">set_units</span>(<span class="dv">5</span>, <span class="st">"umol L-1"</span>)</span>
<span id="cb6-10"><a href="#cb6-10" aria-hidden="true" tabindex="-1"></a>int2 <span class="ot">&lt;-</span> <span class="fu">set_units</span>(<span class="dv">20</span>, <span class="st">"umol L-1"</span>)</span>
<span id="cb6-11"><a href="#cb6-11" aria-hidden="true" tabindex="-1"></a>int3 <span class="ot">&lt;-</span> <span class="fu">set_units</span>(<span class="dv">50</span>, <span class="st">"umol L-1"</span>)</span>
<span id="cb6-12"><a href="#cb6-12" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb6-13"><a href="#cb6-13" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>(dplyr<span class="sc">::</span><span class="fu">filter</span>(df2, S <span class="sc">&lt;=</span> <span class="fu">set_units</span>(<span class="dv">1000</span>, <span class="st">"mg m-3"</span>)), </span>
<span id="cb6-14"><a href="#cb6-14" aria-hidden="true" tabindex="-1"></a>       <span class="fu">aes</span>(<span class="at">x =</span> S, <span class="at">y =</span> uptake_rate)) <span class="sc">+</span></span>
<span id="cb6-15"><a href="#cb6-15" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_line</span>() <span class="sc">+</span></span>
<span id="cb6-16"><a href="#cb6-16" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_vline</span>(<span class="fu">aes</span>(<span class="at">xintercept =</span> <span class="fu">set_units</span>(int1, <span class="st">"mg m-3"</span>)), <span class="at">linetype =</span> <span class="st">"dashed"</span>) <span class="sc">+</span></span>
<span id="cb6-17"><a href="#cb6-17" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_vline</span>(<span class="fu">aes</span>(<span class="at">xintercept =</span> <span class="fu">set_units</span>(int2, <span class="st">"mg m-3"</span>)), <span class="at">linetype =</span> <span class="st">"dashed"</span>) <span class="sc">+</span></span>
<span id="cb6-18"><a href="#cb6-18" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_vline</span>(<span class="fu">aes</span>(<span class="at">xintercept =</span> <span class="fu">set_units</span>(int3, <span class="st">"mg m-3"</span>)), <span class="at">linetype =</span> <span class="st">"dashed"</span>) <span class="sc">+</span></span>
<span id="cb6-19"><a href="#cb6-19" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">y =</span> <span class="st">"Uptake rate"</span>, <span class="at">x =</span> <span class="st">"Substrate concentration"</span>) <span class="sc">+</span></span>
<span id="cb6-20"><a href="#cb6-20" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_classic</span>()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><img src="asparagopsis-parameterisation_files/figure-html/schuenhoff%20MM%20plot-1.png" class="img-fluid figure-img" width="672"></p>
<figcaption>Michaelis–Menten uptake of ammonium by <em>A. armata</em>. Dashed black lines show concentrations of 5 <span class="math inline">\(\mu\)</span>M, 20 <span class="math inline">\(\mu\)</span>M and 50 <span class="math inline">\(\mu\)</span>M for reference.</figcaption>
</figure>
</div>
</div>
</div>
<p>It’s immediately obvious that even though ammonium uptake by MM kinetics increases quickly at low concentrations, the linear uptake rate calculated by <span class="citation" data-cites="torres_nitrogen_2021">Torres et al. (<a href="#ref-torres_nitrogen_2021" role="doc-biblioref">2021</a>)</span> is higher than the MM uptake rate calculated by <span class="citation" data-cites="schuenhoff_tetrasporophyte_2006">Schuenhoff, Mata, and Santos (<a href="#ref-schuenhoff_tetrasporophyte_2006" role="doc-biblioref">2006</a>)</span>, especially at low (5 <span class="math inline">\(\mu\)</span>M) and very high (50 <span class="math inline">\(\mu\)</span>M) concentrations.</p>
<div class="cell">
<details class="code-fold">
<summary>Compare Michaelis–Menten uptake with linear uptake of ammonium</summary>
<div class="sourceCode cell-code" id="cb7"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb7-1"><a href="#cb7-1" aria-hidden="true" tabindex="-1"></a>both <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="at">S =</span> <span class="fu">set_units</span>(<span class="fu">seq</span>(<span class="dv">0</span>, <span class="dv">55</span>, <span class="fl">0.5</span>), <span class="st">"umol L-1"</span>))</span>
<span id="cb7-2"><a href="#cb7-2" aria-hidden="true" tabindex="-1"></a>both <span class="ot">&lt;-</span> both <span class="sc">%&gt;%</span> </span>
<span id="cb7-3"><a href="#cb7-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">MicMen =</span> <span class="fu">MM_uptake</span>(S, V_am, K_am),</span>
<span id="cb7-4"><a href="#cb7-4" aria-hidden="true" tabindex="-1"></a>         <span class="at">linear =</span> <span class="fu">lin_uptake</span>(S, M_am_combo, C_am_combo)) <span class="sc">%&gt;%</span> </span>
<span id="cb7-5"><a href="#cb7-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">pivot_longer</span>(<span class="at">names_to =</span> <span class="st">"shape"</span>, <span class="at">values_to =</span> <span class="st">"uptake_rate"</span>, <span class="at">cols =</span> <span class="fu">c</span>(MicMen, linear)) <span class="sc">%&gt;%</span> </span>
<span id="cb7-6"><a href="#cb7-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">S =</span> <span class="fu">set_units</span>(S, <span class="st">"mg m-3"</span>),</span>
<span id="cb7-7"><a href="#cb7-7" aria-hidden="true" tabindex="-1"></a>         <span class="at">uptake_rate =</span> <span class="fu">set_units</span>(uptake_rate, <span class="st">"mg gDW-1 d-1"</span>))</span>
<span id="cb7-8"><a href="#cb7-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-9"><a href="#cb7-9" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>(both, <span class="fu">aes</span>(<span class="at">x =</span> S, <span class="at">y =</span> uptake_rate, <span class="at">color =</span> shape)) <span class="sc">+</span></span>
<span id="cb7-10"><a href="#cb7-10" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_line</span>() <span class="sc">+</span></span>
<span id="cb7-11"><a href="#cb7-11" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_vline</span>(<span class="fu">aes</span>(<span class="at">xintercept =</span> <span class="fu">set_units</span>(int1, <span class="st">"mg m-3"</span>)), <span class="at">linetype =</span> <span class="st">"dashed"</span>) <span class="sc">+</span></span>
<span id="cb7-12"><a href="#cb7-12" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_vline</span>(<span class="fu">aes</span>(<span class="at">xintercept =</span> <span class="fu">set_units</span>(int2, <span class="st">"mg m-3"</span>)), <span class="at">linetype =</span> <span class="st">"dashed"</span>) <span class="sc">+</span></span>
<span id="cb7-13"><a href="#cb7-13" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_vline</span>(<span class="fu">aes</span>(<span class="at">xintercept =</span> <span class="fu">set_units</span>(int3, <span class="st">"mg m-3"</span>)), <span class="at">linetype =</span> <span class="st">"dashed"</span>) <span class="sc">+</span></span>
<span id="cb7-14"><a href="#cb7-14" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">y =</span> <span class="st">"Uptake rate"</span>, <span class="at">x =</span> <span class="st">"Substrate concentration"</span>) <span class="sc">+</span></span>
<span id="cb7-15"><a href="#cb7-15" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_classic</span>() <span class="sc">+</span></span>
<span id="cb7-16"><a href="#cb7-16" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">legend.position =</span> <span class="st">"none"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><img src="asparagopsis-parameterisation_files/figure-html/both%20uptake%20plot-1.png" class="img-fluid figure-img" width="672"></p>
<figcaption>Uptake of ammonium by <em>A. armata</em> via Michaelis–Menten kinetics (blue) or a linear rate (red) with increasing substrate concentration (maximum of 30<span class="math inline">\(\mu\)</span>M). Dashed lines show reference concentrations of 5, 20 and 50 <span class="math inline">\(\mu\)</span>M for reference.</figcaption>
</figure>
</div>
</div>
</div>
<div class="cell">
<details class="code-fold">
<summary>Convert units of N uptake parameters</summary>
<div class="sourceCode cell-code" id="cb8"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb8-1"><a href="#cb8-1" aria-hidden="true" tabindex="-1"></a>V_am <span class="ot">&lt;-</span> <span class="fu">drop_units</span>(<span class="fu">set_units</span>(V_am, <span class="st">"mg gDW-1 d-1"</span>))</span>
<span id="cb8-2"><a href="#cb8-2" aria-hidden="true" tabindex="-1"></a>K_am <span class="ot">&lt;-</span> <span class="fu">drop_units</span>(<span class="fu">set_units</span>(K_am, <span class="st">"mg m-3"</span>))</span>
<span id="cb8-3"><a href="#cb8-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb8-4"><a href="#cb8-4" aria-hidden="true" tabindex="-1"></a>M_am <span class="ot">&lt;-</span> <span class="fu">drop_units</span>(<span class="fu">set_units</span>(M_am_combo, <span class="st">"m3 gDW-1 d-1"</span>))</span>
<span id="cb8-5"><a href="#cb8-5" aria-hidden="true" tabindex="-1"></a>C_am <span class="ot">&lt;-</span> <span class="fu">drop_units</span>(<span class="fu">set_units</span>(C_am_combo, <span class="st">"mg gDW-1 d-1"</span>))</span>
<span id="cb8-6"><a href="#cb8-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb8-7"><a href="#cb8-7" aria-hidden="true" tabindex="-1"></a>M_ni <span class="ot">&lt;-</span> <span class="dv">0</span></span>
<span id="cb8-8"><a href="#cb8-8" aria-hidden="true" tabindex="-1"></a>C_ni <span class="ot">&lt;-</span> <span class="fu">drop_units</span>(<span class="fu">set_units</span>(C_ni_combo, <span class="st">"mg gDW-1 d-1"</span>))</span>
<span id="cb8-9"><a href="#cb8-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb8-10"><a href="#cb8-10" aria-hidden="true" tabindex="-1"></a>M_ot <span class="ot">&lt;-</span> <span class="fu">drop_units</span>(<span class="fu">set_units</span>(M_aa_combo, <span class="st">"m3 gDW-1 d-1"</span>))</span>
<span id="cb8-11"><a href="#cb8-11" aria-hidden="true" tabindex="-1"></a>C_ot <span class="ot">&lt;-</span> <span class="fu">drop_units</span>(<span class="fu">set_units</span>(C_aa_combo, <span class="st">"mg gDW-1 d-1"</span>))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
</section>
</section>
<section id="sec-internal-nitrogen-tetrasporophyte" class="level2">
<h2 class="anchored" data-anchor-id="sec-internal-nitrogen-tetrasporophyte">Internal nutrient state</h2>
<section id="sec-nitrogen-concentration-tetrasporophyte" class="level3">
<h3 class="anchored" data-anchor-id="sec-nitrogen-concentration-tetrasporophyte">Nitrogen concentration</h3>
<p>The model requires minimum and maximum internal nitrogen concentrations, so I’ll be taking the lowest and highest concentrations I can find.</p>
<p><span class="citation" data-cites="mata_direct_2010">Mata, Schuenhoff, and Santos (<a href="#ref-mata_direct_2010" role="doc-biblioref">2010</a>)</span> reported that <em>A. armata</em>’s N-content was 6.04% of DW at harvesting in December, up from 5.90% at the time of stocking (or 5.56% up to 5.62% in May).</p>
<div class="cell">
<details class="code-fold">
<summary>Clean up data from <span class="citation" data-cites="mata_direct_2010">Mata, Schuenhoff, and Santos (<a href="#ref-mata_direct_2010" role="doc-biblioref">2010</a>)</span></summary>
<div class="sourceCode cell-code" id="cb9"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb9-1"><a href="#cb9-1" aria-hidden="true" tabindex="-1"></a>matadata <span class="ot">&lt;-</span> <span class="fu">read.csv</span>(<span class="fu">file.path</span>(<span class="st">"data_raw"</span>, <span class="st">"sources"</span>, <span class="st">"mata_2010"</span>, <span class="st">"wpd_datasets.csv"</span>))</span>
<span id="cb9-2"><a href="#cb9-2" aria-hidden="true" tabindex="-1"></a>ls <span class="ot">&lt;-</span> <span class="fu">split.default</span>(matadata, <span class="fu">rep</span>(<span class="dv">1</span><span class="sc">:</span><span class="dv">12</span>, <span class="at">each =</span> <span class="dv">2</span>))</span>
<span id="cb9-3"><a href="#cb9-3" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> (i <span class="cf">in</span> <span class="dv">1</span><span class="sc">:</span><span class="fu">length</span>(ls)){</span>
<span id="cb9-4"><a href="#cb9-4" aria-hidden="true" tabindex="-1"></a>  ls[[i]] <span class="ot">&lt;-</span> ls[[i]] <span class="sc">%&gt;%</span> </span>
<span id="cb9-5"><a href="#cb9-5" aria-hidden="true" tabindex="-1"></a>    <span class="fu">mutate</span>(<span class="at">dataset =</span> <span class="fu">colnames</span>(ls[[i]][<span class="dv">1</span>]))</span>
<span id="cb9-6"><a href="#cb9-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">colnames</span>(ls[[i]]) <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">"X"</span>, <span class="st">"Y"</span>, <span class="st">"dataset"</span>)</span>
<span id="cb9-7"><a href="#cb9-7" aria-hidden="true" tabindex="-1"></a>  ls[[i]] <span class="ot">&lt;-</span> ls[[i]] <span class="sc">%&gt;%</span> </span>
<span id="cb9-8"><a href="#cb9-8" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">filter</span>(X <span class="sc">!=</span> <span class="st">"X"</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb9-9"><a href="#cb9-9" aria-hidden="true" tabindex="-1"></a>    <span class="fu">mutate</span>(<span class="at">X =</span> <span class="fu">as.numeric</span>(X),</span>
<span id="cb9-10"><a href="#cb9-10" aria-hidden="true" tabindex="-1"></a>           <span class="at">Y =</span> <span class="fu">as.numeric</span>(Y))</span>
<span id="cb9-11"><a href="#cb9-11" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb9-12"><a href="#cb9-12" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb9-13"><a href="#cb9-13" aria-hidden="true" tabindex="-1"></a>mata.N <span class="ot">&lt;-</span> <span class="fu">list</span>(ls[[<span class="dv">2</span>]], ls[[<span class="dv">3</span>]], ls[[<span class="dv">4</span>]], ls[[<span class="dv">5</span>]], ls[[<span class="dv">9</span>]], ls[[<span class="dv">10</span>]], ls[[<span class="dv">11</span>]], ls[[<span class="dv">12</span>]])</span>
<span id="cb9-14"><a href="#cb9-14" aria-hidden="true" tabindex="-1"></a>mata.N <span class="ot">&lt;-</span> <span class="fu">bind_rows</span>(mata.N) <span class="sc">%&gt;%</span> </span>
<span id="cb9-15"><a href="#cb9-15" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">species =</span> <span class="fu">case_when</span>(<span class="fu">str_detect</span>(dataset, <span class="at">pattern =</span> <span class="st">"U..rigidia"</span>) <span class="sc">~</span> <span class="st">"U. rigidia"</span>, <span class="cn">TRUE</span> <span class="sc">~</span> <span class="st">"A. armata"</span>),</span>
<span id="cb9-16"><a href="#cb9-16" aria-hidden="true" tabindex="-1"></a>         <span class="at">measure =</span> <span class="fu">case_when</span>(<span class="fu">str_detect</span>(dataset, <span class="at">pattern =</span> <span class="st">"N.removal"</span>) <span class="sc">~</span> <span class="st">"N removal"</span>, <span class="cn">TRUE</span> <span class="sc">~</span> <span class="st">"N yield"</span>),</span>
<span id="cb9-17"><a href="#cb9-17" aria-hidden="true" tabindex="-1"></a>         <span class="at">month =</span> <span class="fu">case_when</span>(<span class="fu">str_detect</span>(dataset, <span class="at">pattern =</span> <span class="st">"May"</span>) <span class="sc">~</span> <span class="st">"May"</span>, <span class="cn">TRUE</span> <span class="sc">~</span> <span class="st">"Dec"</span>),</span>
<span id="cb9-18"><a href="#cb9-18" aria-hidden="true" tabindex="-1"></a>         <span class="at">TAN_flux =</span> X, <span class="co"># umol L-1 hr-1</span></span>
<span id="cb9-19"><a href="#cb9-19" aria-hidden="true" tabindex="-1"></a>         <span class="at">Biofiltration =</span> Y <span class="co"># gDW m-2 d-1</span></span>
<span id="cb9-20"><a href="#cb9-20" aria-hidden="true" tabindex="-1"></a>         ) <span class="sc">%&gt;%</span> </span>
<span id="cb9-21"><a href="#cb9-21" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">select</span>(<span class="sc">-</span><span class="fu">c</span>(X, Y, dataset))</span>
<span id="cb9-22"><a href="#cb9-22" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb9-23"><a href="#cb9-23" aria-hidden="true" tabindex="-1"></a>mata.Y <span class="ot">&lt;-</span> <span class="fu">list</span>(ls[[<span class="dv">1</span>]], ls[[<span class="dv">6</span>]], ls[[<span class="dv">7</span>]], ls[[<span class="dv">8</span>]])</span>
<span id="cb9-24"><a href="#cb9-24" aria-hidden="true" tabindex="-1"></a>mata.Y <span class="ot">&lt;-</span> <span class="fu">bind_rows</span>(mata.Y) <span class="sc">%&gt;%</span> </span>
<span id="cb9-25"><a href="#cb9-25" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">species =</span> <span class="fu">case_when</span>(<span class="fu">str_detect</span>(dataset, <span class="at">pattern =</span> <span class="st">"U..rigidia"</span>) <span class="sc">~</span> <span class="st">"U. rigidia"</span>, <span class="cn">TRUE</span> <span class="sc">~</span> <span class="st">"A. armata"</span>),</span>
<span id="cb9-26"><a href="#cb9-26" aria-hidden="true" tabindex="-1"></a>         <span class="at">month =</span> <span class="fu">case_when</span>(<span class="fu">str_detect</span>(dataset, <span class="at">pattern =</span> <span class="st">"May"</span>) <span class="sc">~</span> <span class="st">"May"</span>, <span class="cn">TRUE</span> <span class="sc">~</span> <span class="st">"Dec"</span>),</span>
<span id="cb9-27"><a href="#cb9-27" aria-hidden="true" tabindex="-1"></a>         <span class="at">TAN_flux =</span> X, <span class="co"># umol L-1 hr-1</span></span>
<span id="cb9-28"><a href="#cb9-28" aria-hidden="true" tabindex="-1"></a>         <span class="at">Biomass_yield =</span> Y <span class="co"># gDW m-2 d-1</span></span>
<span id="cb9-29"><a href="#cb9-29" aria-hidden="true" tabindex="-1"></a>         ) <span class="sc">%&gt;%</span> </span>
<span id="cb9-30"><a href="#cb9-30" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">select</span>(<span class="sc">-</span><span class="fu">c</span>(X, Y, dataset))</span>
<span id="cb9-31"><a href="#cb9-31" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb9-32"><a href="#cb9-32" aria-hidden="true" tabindex="-1"></a><span class="fu">save</span>(mata.N, <span class="at">file =</span> <span class="fu">file.path</span>(<span class="st">"data_raw"</span>, <span class="st">"sources"</span>, <span class="st">"mata_2010"</span>, <span class="st">"mata.N.rda"</span>))</span>
<span id="cb9-33"><a href="#cb9-33" aria-hidden="true" tabindex="-1"></a><span class="fu">save</span>(mata.Y, <span class="at">file =</span> <span class="fu">file.path</span>(<span class="st">"data_raw"</span>, <span class="st">"sources"</span>, <span class="st">"mata_2010"</span>, <span class="st">"mata.Y.rda"</span>))</span>
<span id="cb9-34"><a href="#cb9-34" aria-hidden="true" tabindex="-1"></a><span class="fu">rm</span>(ls)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<p>Using a MM curve (V = Vmax/(Ks + [S])) <span class="citation" data-cites="mata_direct_2010">Mata, Schuenhoff, and Santos (<a href="#ref-mata_direct_2010" role="doc-biblioref">2010</a>)</span> estimated a maximum N-yield of 9.5 g N m-2 d-1 for <em>A. armata</em> in May, and a maximum biomass yield of 153.6 g DW m-2 d-1. This gives a maximum N-content of 6.18%. The minimum N% recorded in their study (at stocking) was 5.56% <span class="math inline">\(\pm\)</span> 0.13%.</p>
<p><span class="citation" data-cites="schuenhoff_tetrasporophyte_2006">Schuenhoff, Mata, and Santos (<a href="#ref-schuenhoff_tetrasporophyte_2006" role="doc-biblioref">2006</a>)</span> found that C- and N-content of their tank-grown <em>A. armata</em> remained consistent regardless of available TAN (minimum 0, maximum 400 <span class="math inline">\(\mu\)</span>mol l<span class="math inline">\(^{-1}\)</span> h<span class="math inline">\(^{-1}\)</span>). <!-- This doesn't give minimum and maximum values but do help calculate mg N from @schuenhoff_tetrasporophyte_2006's data. --></p>
<div class="cell">
<details class="code-fold">
<summary>Minimum and maximum N composition</summary>
<div class="sourceCode cell-code" id="cb10"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb10-1"><a href="#cb10-1" aria-hidden="true" tabindex="-1"></a>N_min <span class="ot">&lt;-</span> <span class="fu">set_units</span>(<span class="fl">0.056-0.0013</span>, <span class="st">"g gDW-1"</span>) <span class="co"># Mata</span></span>
<span id="cb10-2"><a href="#cb10-2" aria-hidden="true" tabindex="-1"></a>Q_min <span class="ot">&lt;-</span> <span class="fu">set_units</span>(N_min, <span class="st">"mg gDW-1"</span>)</span>
<span id="cb10-3"><a href="#cb10-3" aria-hidden="true" tabindex="-1"></a>N_max <span class="ot">&lt;-</span> <span class="fu">set_units</span>(<span class="fl">0.065</span>, <span class="st">"g gDW-1"</span>) <span class="co"># Shuenhoff</span></span>
<span id="cb10-4"><a href="#cb10-4" aria-hidden="true" tabindex="-1"></a>Q_max <span class="ot">&lt;-</span> <span class="fu">set_units</span>(N_max, <span class="st">"mg gDW-1"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<p>This gives a <span class="math inline">\(Q_{min}\)</span> of 54.7 mg gDW<span class="math inline">\(^{-1}\)</span> and a <span class="math inline">\(Q_{max}\)</span> of 65 mg gDW<span class="math inline">\(^{-1}\)</span>. This is a very narrow range.</p>
<div class="cell">
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb11"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb11-1"><a href="#cb11-1" aria-hidden="true" tabindex="-1"></a>mata <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="at">x1=</span><span class="fu">c</span>(<span class="st">'A. armata'</span>, <span class="dv">110</span>, <span class="dv">5</span>, <span class="fl">14.2</span>, <span class="fl">4.1</span>, <span class="dv">71</span>,<span class="dv">72</span>, <span class="fl">2.8</span>, <span class="fl">3.2</span>,<span class="fl">4.2</span>, <span class="fl">4.1</span>,<span class="st">'This study'</span>),</span>
<span id="cb11-2"><a href="#cb11-2" aria-hidden="true" tabindex="-1"></a>                   <span class="at">x2=</span><span class="fu">c</span>(<span class="st">'U. rigida'</span>, <span class="dv">110</span>, <span class="dv">4</span>, <span class="fl">14.2</span>, <span class="fl">4.1</span>, <span class="dv">44</span>,<span class="fl">51.7</span>, <span class="fl">2.7</span>, <span class="fl">2.9</span>,<span class="fl">2.7</span>,<span class="fl">3.2</span>, <span class="st">'This study'</span>),</span>
<span id="cb11-3"><a href="#cb11-3" aria-hidden="true" tabindex="-1"></a>                   <span class="at">x3=</span><span class="fu">c</span>(<span class="st">'A. armata'</span>, <span class="dv">110</span>, <span class="dv">5</span>, <span class="fl">21.5</span>, <span class="fl">3.8</span>, <span class="dv">125</span>,<span class="dv">153</span>, <span class="fl">6.5</span>, <span class="fl">8.2</span>,<span class="fl">7.4</span>,<span class="fl">9.5</span>, <span class="st">'This study'</span>),</span>
<span id="cb11-4"><a href="#cb11-4" aria-hidden="true" tabindex="-1"></a>                   <span class="at">x4=</span><span class="fu">c</span>(<span class="st">'U. rigida'</span>, <span class="dv">110</span>, <span class="dv">4</span>, <span class="fl">21.5</span>, <span class="fl">3.8</span>, <span class="dv">73</span>, <span class="fl">72.2</span>, <span class="fl">5.1</span>, <span class="fl">6.4</span>, <span class="fl">3.6</span>, <span class="fl">3.9</span>, <span class="st">'This study'</span>),</span>
<span id="cb11-5"><a href="#cb11-5" aria-hidden="true" tabindex="-1"></a>                   <span class="at">x5=</span><span class="fu">c</span>(<span class="st">'U. lactuca'</span>, <span class="dv">600</span>, <span class="fl">1.7</span>, <span class="dv">20</span>, <span class="fl">0.5</span>, <span class="dv">55</span>,<span class="fl">60.3</span>,<span class="fl">3.2</span>,<span class="fl">5.6</span>,<span class="fl">2.3</span>,<span class="cn">NA</span>, <span class="st">'Cohen and Neori 1991; Neori et al. 1991'</span>),</span>
<span id="cb11-6"><a href="#cb11-6" aria-hidden="true" tabindex="-1"></a>                   <span class="at">x6=</span><span class="fu">c</span>(<span class="st">'U. rigida'</span>, <span class="dv">750</span>, <span class="fl">2.5</span>, <span class="dv">24</span>, <span class="fl">0.5</span>, <span class="dv">40</span>,<span class="cn">NA</span>, <span class="dv">2</span>,<span class="fl">2.3</span>,<span class="fl">1.8</span>,<span class="cn">NA</span>,<span class="st">'Jimenez del Rio et al. 1994, 1996'</span>),</span>
<span id="cb11-7"><a href="#cb11-7" aria-hidden="true" tabindex="-1"></a>                   <span class="at">x7=</span><span class="fu">c</span>(<span class="st">'U. rigida'</span>, <span class="dv">1900</span>, <span class="dv">2</span>, <span class="dv">22</span>, <span class="fl">0.6</span>, <span class="dv">48</span>,<span class="fl">47.9</span>,<span class="fl">1.31</span>,<span class="fl">3.1</span>,<span class="fl">1.45</span>,<span class="cn">NA</span>, <span class="st">'Mata and Santos 2003'</span>))</span>
<span id="cb11-8"><a href="#cb11-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb11-9"><a href="#cb11-9" aria-hidden="true" tabindex="-1"></a>mata <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="fu">t</span>(mata))</span>
<span id="cb11-10"><a href="#cb11-10" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb11-11"><a href="#cb11-11" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(mata) <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">"Species"</span>, </span>
<span id="cb11-12"><a href="#cb11-12" aria-hidden="true" tabindex="-1"></a>         <span class="st">"Tank_vol_L"</span>, </span>
<span id="cb11-13"><a href="#cb11-13" aria-hidden="true" tabindex="-1"></a>         <span class="st">"Stocking_density_g.FW.L−1"</span>, </span>
<span id="cb11-14"><a href="#cb11-14" aria-hidden="true" tabindex="-1"></a>         <span class="st">"Mean_water_temp_C"</span>, </span>
<span id="cb11-15"><a href="#cb11-15" aria-hidden="true" tabindex="-1"></a>         <span class="st">"Max_water_exchange_vol.h−1"</span>, </span>
<span id="cb11-16"><a href="#cb11-16" aria-hidden="true" tabindex="-1"></a>         <span class="st">"Max_biomass_yield_g.DW.m−2.d−1"</span>, </span>
<span id="cb11-17"><a href="#cb11-17" aria-hidden="true" tabindex="-1"></a>         <span class="st">"Est_max_biomass_yield_g.DW.m−2.d−1"</span>, </span>
<span id="cb11-18"><a href="#cb11-18" aria-hidden="true" tabindex="-1"></a>         <span class="st">"TAN_removal_g.m−2.d−1"</span>, </span>
<span id="cb11-19"><a href="#cb11-19" aria-hidden="true" tabindex="-1"></a>         <span class="st">"Est_max_TAN_removal_g.m−2.d−1"</span>,</span>
<span id="cb11-20"><a href="#cb11-20" aria-hidden="true" tabindex="-1"></a>         <span class="st">"N_yield_g.m−2.d−1"</span>, </span>
<span id="cb11-21"><a href="#cb11-21" aria-hidden="true" tabindex="-1"></a>         <span class="st">"Est_max_N_yield_g.m−2.d−1"</span>,</span>
<span id="cb11-22"><a href="#cb11-22" aria-hidden="true" tabindex="-1"></a>         <span class="st">"Reference"</span>)</span>
<span id="cb11-23"><a href="#cb11-23" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb11-24"><a href="#cb11-24" aria-hidden="true" tabindex="-1"></a>mata <span class="ot">&lt;-</span> mata <span class="sc">%&gt;%</span> </span>
<span id="cb11-25"><a href="#cb11-25" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">Tank_vol =</span> <span class="fu">set_units</span>(<span class="fu">as.numeric</span>(Tank_vol_L), <span class="st">"L"</span>),</span>
<span id="cb11-26"><a href="#cb11-26" aria-hidden="true" tabindex="-1"></a>         <span class="at">Stocking_density =</span> <span class="fu">set_units</span>(<span class="fu">as.numeric</span>(<span class="st">`</span><span class="at">Stocking_density_g.FW.L−1</span><span class="st">`</span>), <span class="st">"gFW L-1"</span>),</span>
<span id="cb11-27"><a href="#cb11-27" aria-hidden="true" tabindex="-1"></a>         <span class="at">Mean_water_temp =</span> <span class="fu">set_units</span>(<span class="fu">as.numeric</span>(Mean_water_temp_C), <span class="st">"C"</span>),</span>
<span id="cb11-28"><a href="#cb11-28" aria-hidden="true" tabindex="-1"></a>         <span class="at">Max_water_exchange_vol =</span> <span class="fu">set_units</span>(<span class="fu">as.numeric</span>(<span class="st">`</span><span class="at">Max_water_exchange_vol.h−1</span><span class="st">`</span>), <span class="st">"h-1"</span>),</span>
<span id="cb11-29"><a href="#cb11-29" aria-hidden="true" tabindex="-1"></a>         <span class="at">Max_biomass_yield =</span> <span class="fu">set_units</span>(<span class="fu">as.numeric</span>(<span class="st">`</span><span class="at">Max_biomass_yield_g.DW.m−2.d−1</span><span class="st">`</span>), <span class="st">"gDW m-2 d-1"</span>),</span>
<span id="cb11-30"><a href="#cb11-30" aria-hidden="true" tabindex="-1"></a>         <span class="at">Est_max_biomass_yield =</span> <span class="fu">set_units</span>(<span class="fu">as.numeric</span>(<span class="st">`</span><span class="at">Est_max_biomass_yield_g.DW.m−2.d−1</span><span class="st">`</span>), <span class="st">"gDW m-2 d-1"</span>),</span>
<span id="cb11-31"><a href="#cb11-31" aria-hidden="true" tabindex="-1"></a>         <span class="at">TAN_removal =</span> <span class="fu">set_units</span>(<span class="fu">as.numeric</span>(<span class="st">`</span><span class="at">TAN_removal_g.m−2.d−1</span><span class="st">`</span>), <span class="st">"g m-2 d-1"</span>),</span>
<span id="cb11-32"><a href="#cb11-32" aria-hidden="true" tabindex="-1"></a>         <span class="at">Est_max_TAN_removal =</span> <span class="fu">set_units</span>(<span class="fu">as.numeric</span>(<span class="st">`</span><span class="at">Est_max_TAN_removal_g.m−2.d−1</span><span class="st">`</span>), <span class="st">"g m-2 d-1"</span>),</span>
<span id="cb11-33"><a href="#cb11-33" aria-hidden="true" tabindex="-1"></a>         <span class="at">N_yield =</span> <span class="fu">set_units</span>(<span class="fu">as.numeric</span>(<span class="st">`</span><span class="at">N_yield_g.m−2.d−1</span><span class="st">`</span>), <span class="st">"g m-2 d-1"</span>),</span>
<span id="cb11-34"><a href="#cb11-34" aria-hidden="true" tabindex="-1"></a>         <span class="at">Est_max_N_yield =</span> <span class="fu">set_units</span>(<span class="fu">as.numeric</span>(<span class="st">`</span><span class="at">Est_max_N_yield_g.m−2.d−1</span><span class="st">`</span>), <span class="st">"g m-2 d-1"</span>)</span>
<span id="cb11-35"><a href="#cb11-35" aria-hidden="true" tabindex="-1"></a>  ) <span class="sc">%&gt;%</span> </span>
<span id="cb11-36"><a href="#cb11-36" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(<span class="sc">-</span><span class="fu">c</span>(Tank_vol_L,<span class="st">`</span><span class="at">Stocking_density_g.FW.L−1</span><span class="st">`</span>,Mean_water_temp_C,<span class="st">`</span><span class="at">Max_water_exchange_vol.h−1</span><span class="st">`</span>,<span class="st">`</span><span class="at">Max_biomass_yield_g.DW.m−2.d−1</span><span class="st">`</span>,<span class="st">`</span><span class="at">TAN_removal_g.m−2.d−1</span><span class="st">`</span>,<span class="st">`</span><span class="at">Est_max_N_yield_g.m−2.d−1</span><span class="st">`</span>,<span class="st">`</span><span class="at">Est_max_TAN_removal_g.m−2.d−1</span><span class="st">`</span>,<span class="st">`</span><span class="at">Est_max_biomass_yield_g.DW.m−2.d−1</span><span class="st">`</span>,<span class="st">`</span><span class="at">N_yield_g.m−2.d−1</span><span class="st">`</span>))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<p>C% of tissue is not currently used within the model but might be useful later.</p>
<div class="cell">
<details class="code-fold">
<summary>Get Schuenhoff et al. (2006) compositional data</summary>
<div class="sourceCode cell-code" id="cb12"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb12-1"><a href="#cb12-1" aria-hidden="true" tabindex="-1"></a>N_perc <span class="ot">&lt;-</span> <span class="fu">set_units</span>(<span class="fl">0.065</span>, <span class="st">"g gDW-1"</span>)</span>
<span id="cb12-2"><a href="#cb12-2" aria-hidden="true" tabindex="-1"></a>C_perc <span class="ot">&lt;-</span> <span class="fu">set_units</span>(<span class="fl">0.355</span>, <span class="st">"g gDW-1"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
</section>
</section>
<section id="sec-salinity-tetrasporophyte" class="level2">
<h2 class="anchored" data-anchor-id="sec-salinity-tetrasporophyte">Salinity</h2>
<p>Not much is known about how <em>Asparagopsis</em> responds to salinity, but it is thought to be fairly intolerant of low salinity (Camille White, pers. comm.). <!-- Camille reckons salinity is a must, and that _A. armata_ has a minimum salinity tolerance of ~34ppm. --> Mean salinity in most Tasmanian coastal waters is fairly consistent at ~35.25 PSU, so I’m going to set that as the optimum for both species.</p>
<div class="cell">
<details class="code-fold">
<summary>Show salinity parameters</summary>
<div class="sourceCode cell-code" id="cb13"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb13-1"><a href="#cb13-1" aria-hidden="true" tabindex="-1"></a>S_opt <span class="ot">&lt;-</span> <span class="fl">35.25</span> </span>
<span id="cb13-2"><a href="#cb13-2" aria-hidden="true" tabindex="-1"></a>S_min <span class="ot">&lt;-</span> <span class="fl">21.15</span> <span class="co"># ecklonia minimum is 80% of mean below mean. This is 40% of mean below mean.</span></span>
<span id="cb13-3"><a href="#cb13-3" aria-hidden="true" tabindex="-1"></a>S_max <span class="ot">&lt;-</span> <span class="fl">41.65</span> <span class="co"># ecklonia minimum is 36% of mean below mean. This is 18% of mean below mean.</span></span>
<span id="cb13-4"><a href="#cb13-4" aria-hidden="true" tabindex="-1"></a>sal <span class="ot">&lt;-</span> <span class="fu">seq</span>(<span class="dv">0</span>,<span class="dv">50</span>,<span class="fl">0.5</span>)</span>
<span id="cb13-5"><a href="#cb13-5" aria-hidden="true" tabindex="-1"></a>Slim <span class="ot">&lt;-</span> <span class="fu">sapply</span>(<span class="at">X =</span> sal, <span class="at">FUN =</span> T_lim, <span class="at">spec_params =</span> <span class="fu">c</span>(<span class="at">T_opt =</span> S_opt, <span class="at">T_min =</span> S_min, <span class="at">T_max =</span> S_max))</span>
<span id="cb13-6"><a href="#cb13-6" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>(<span class="at">mapping =</span> <span class="fu">aes</span>(<span class="at">x =</span> sal, <span class="at">y =</span> Slim)) <span class="sc">+</span></span>
<span id="cb13-7"><a href="#cb13-7" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_line</span>(<span class="at">size =</span> <span class="dv">1</span>) <span class="sc">+</span></span>
<span id="cb13-8"><a href="#cb13-8" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_classic</span>() <span class="sc">+</span></span>
<span id="cb13-9"><a href="#cb13-9" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">legend.position =</span> <span class="st">"none"</span>, <span class="fu">element_text</span>(<span class="at">family =</span> <span class="st">"sans"</span>, <span class="at">colour =</span> <span class="st">"black"</span>, <span class="at">size =</span> <span class="dv">11</span>),</span>
<span id="cb13-10"><a href="#cb13-10" aria-hidden="true" tabindex="-1"></a>        <span class="at">axis.title.y =</span> <span class="fu">element_text</span>(<span class="at">vjust =</span> <span class="dv">2</span>)) <span class="sc">+</span></span>
<span id="cb13-11"><a href="#cb13-11" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">x =</span> <span class="fu">expression</span>(<span class="st">"Salinity (g L"</span><span class="sc">^-</span><span class="dv">1</span><span class="sc">*</span><span class="st">")"</span>), <span class="at">y =</span> <span class="st">"Relative growth"</span>) <span class="sc">+</span></span>
<span id="cb13-12"><a href="#cb13-12" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_y_continuous</span>(<span class="at">breaks =</span> <span class="fu">seq</span>(<span class="dv">0</span>, <span class="dv">1</span>, <span class="fl">0.2</span>), <span class="at">limits =</span> <span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">1</span>)) <span class="sc">+</span></span>
<span id="cb13-13"><a href="#cb13-13" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_x_continuous</span>(<span class="at">breaks =</span> <span class="fu">seq</span>(<span class="dv">33</span>, <span class="dv">37</span>, <span class="fl">0.5</span>))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="asparagopsis-parameterisation_files/figure-html/salinity-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
</section>
<section id="biomass-factors" class="level2">
<h2 class="anchored" data-anchor-id="biomass-factors">Biomass factors</h2>
<section id="sec-dry-weight-tetrasporophyte" class="level3">
<h3 class="anchored" data-anchor-id="sec-dry-weight-tetrasporophyte">Dry weight to fresh weight</h3>
<p><span class="citation" data-cites="schuenhoff_tetrasporophyte_2006">Schuenhoff, Mata, and Santos (<a href="#ref-schuenhoff_tetrasporophyte_2006" role="doc-biblioref">2006</a>)</span> found that the DW:FW ratio of <em>A. armata</em> remained constant at 0.25 <span class="math inline">\(\pm\)</span> 0.001 (mean <span class="math inline">\(\pm\)</span> SD) regardless of season or stocking density (tetrasporophytes sourced from southern Portugal, tank-grown in fish farm effluent at a water exchange rate of 2 Vol hr<span class="math inline">\(^{-1}\)</span>).</p>
<div class="cell">
<details class="code-fold">
<summary>Set DWWW units based on <span class="citation" data-cites="schuenhoff_tetrasporophyte_2006">Schuenhoff, Mata, and Santos (<a href="#ref-schuenhoff_tetrasporophyte_2006" role="doc-biblioref">2006</a>)</span></summary>
<div class="sourceCode cell-code" id="cb14"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb14-1"><a href="#cb14-1" aria-hidden="true" tabindex="-1"></a><span class="fu">remove_unit</span>(<span class="st">"gWW"</span>)</span>
<span id="cb14-2"><a href="#cb14-2" aria-hidden="true" tabindex="-1"></a><span class="fu">install_unit</span>(<span class="st">"gWW"</span>, <span class="st">"0.25 gDW"</span>)</span>
<span id="cb14-3"><a href="#cb14-3" aria-hidden="true" tabindex="-1"></a>DWWW <span class="ot">&lt;-</span> <span class="fu">set_units</span>(<span class="dv">1</span>, <span class="st">"gDW"</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb14-4"><a href="#cb14-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">set_units</span>(<span class="st">"gWW"</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb14-5"><a href="#cb14-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">drop_units</span>()</span>
<span id="cb14-6"><a href="#cb14-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb14-7"><a href="#cb14-7" aria-hidden="true" tabindex="-1"></a><span class="fu">remove_unit</span>(<span class="st">"gWW"</span>)</span>
<span id="cb14-8"><a href="#cb14-8" aria-hidden="true" tabindex="-1"></a><span class="fu">install_unit</span>(<span class="st">"gWW"</span>, <span class="st">"0.12 gDW"</span>)</span>
<span id="cb14-9"><a href="#cb14-9" aria-hidden="true" tabindex="-1"></a>DWWW2 <span class="ot">&lt;-</span> <span class="fu">set_units</span>(<span class="dv">1</span>, <span class="st">"gDW"</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb14-10"><a href="#cb14-10" aria-hidden="true" tabindex="-1"></a>  <span class="fu">set_units</span>(<span class="st">"gWW"</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb14-11"><a href="#cb14-11" aria-hidden="true" tabindex="-1"></a>  <span class="fu">drop_units</span>()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<p>By contrast, <span class="citation" data-cites="sainz-villegas_exploring_2024">Sainz-Villegas et al. (<a href="#ref-sainz-villegas_exploring_2024" role="doc-biblioref">2024</a>)</span> measured the change from FW to DW as 0.12, which would give a <span class="math inline">\(DWWW\)</span> value of 8.33.</p>
</section>
<section id="height" class="level3">
<h3 class="anchored" data-anchor-id="height">Height</h3>
<p>The tetrasporophyte of <em>Asparagopsis</em> is very small and has a consistent pom-pom structure. How this manifests will be dictated by culturing setup, if this is the cultured stage.</p>
<div class="cell">
<details class="code-fold">
<summary>Set some default height values</summary>
<div class="sourceCode cell-code" id="cb15"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb15-1"><a href="#cb15-1" aria-hidden="true" tabindex="-1"></a>h_a <span class="ot">&lt;-</span> <span class="cn">NA</span></span>
<span id="cb15-2"><a href="#cb15-2" aria-hidden="true" tabindex="-1"></a>h_b <span class="ot">&lt;-</span> <span class="cn">NA</span></span>
<span id="cb15-3"><a href="#cb15-3" aria-hidden="true" tabindex="-1"></a>h_c <span class="ot">&lt;-</span> <span class="cn">NA</span></span>
<span id="cb15-4"><a href="#cb15-4" aria-hidden="true" tabindex="-1"></a>h_max <span class="ot">&lt;-</span> <span class="dv">1</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
</section>
<section id="growth-rate-stocking-density" class="level3">
<h3 class="anchored" data-anchor-id="growth-rate-stocking-density">Growth rate, stocking density</h3>
<p><span class="citation" data-cites="schuenhoff_tetrasporophyte_2006">Schuenhoff, Mata, and Santos (<a href="#ref-schuenhoff_tetrasporophyte_2006" role="doc-biblioref">2006</a>)</span> investigated the optimal stocking density for <em>A. armata</em> in spring and winter at a high water exchange rate (2 Vol hr<span class="math inline">\(^{-1}\)</span>). Yield was determined via:</p>
<p><span class="math display">\[
Y(g DW m^{−2} week^{−1}) = \frac{N_t (g FW) − N_0 (g FW)}{t (weeks) \times DW:FW \times Area (m^{2})}
\]</span></p>
<p>The area of each tank was 0.23 m<span class="math inline">\(^{-2}\)</span> (with a depth of 0.48 m).</p>
<p>Maximum yields in <span class="citation" data-cites="schuenhoff_tetrasporophyte_2006">Schuenhoff, Mata, and Santos (<a href="#ref-schuenhoff_tetrasporophyte_2006" role="doc-biblioref">2006</a>)</span> in both seasons was achieved at a stocking density of 5 g FW l<span class="math inline">\(^{-1}\)</span>, but the yield in spring was higher than in winter (~700 vs ~520 g DW m<span class="math inline">\(^{-2}\)</span> week<span class="math inline">\(^{-1}\)</span>). These can be converted to mg N m<span class="math inline">\(^{-2}\)</span> d<span class="math inline">\(^{-1}\)</span>:</p>
<div class="cell">
<details class="code-fold">
<summary>Get Schuenhoff et al. (2006) starting condition and yield data</summary>
<div class="sourceCode cell-code" id="cb16"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb16-1"><a href="#cb16-1" aria-hidden="true" tabindex="-1"></a>spring <span class="ot">&lt;-</span> <span class="fu">read.csv</span>(<span class="fu">file.path</span>(<span class="st">"data_raw"</span>, <span class="st">"sources"</span>, <span class="st">"schuenhoff_2006"</span>, <span class="st">"schuenhoff_2006_Fig1_spring-data.csv"</span>), <span class="at">header =</span> F) <span class="sc">%&gt;%</span></span>
<span id="cb16-2"><a href="#cb16-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">season =</span> <span class="st">"spring"</span>)</span>
<span id="cb16-3"><a href="#cb16-3" aria-hidden="true" tabindex="-1"></a>winter <span class="ot">&lt;-</span> <span class="fu">read.csv</span>(<span class="fu">file.path</span>(<span class="st">"data_raw"</span>, <span class="st">"sources"</span>, <span class="st">"schuenhoff_2006"</span>, <span class="st">"schuenhoff_2006_Fig1_winter-data.csv"</span>), <span class="at">header =</span> F) <span class="sc">%&gt;%</span></span>
<span id="cb16-4"><a href="#cb16-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">season =</span> <span class="st">"winter"</span>)</span>
<span id="cb16-5"><a href="#cb16-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb16-6"><a href="#cb16-6" aria-hidden="true" tabindex="-1"></a><span class="co"># initial starting value</span></span>
<span id="cb16-7"><a href="#cb16-7" aria-hidden="true" tabindex="-1"></a>stock_dens <span class="ot">&lt;-</span> <span class="fu">set_units</span>(<span class="dv">5</span>, <span class="st">"gWW L-1"</span>)</span>
<span id="cb16-8"><a href="#cb16-8" aria-hidden="true" tabindex="-1"></a>stock_dens <span class="ot">&lt;-</span> <span class="fu">set_units</span>(stock_dens, <span class="st">"gDW m-3"</span>)</span>
<span id="cb16-9"><a href="#cb16-9" aria-hidden="true" tabindex="-1"></a>stock_dens <span class="ot">&lt;-</span> <span class="fu">set_units</span>(stock_dens<span class="sc">*</span>N_perc, <span class="st">"mg m-3"</span>)</span>
<span id="cb16-10"><a href="#cb16-10" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb16-11"><a href="#cb16-11" aria-hidden="true" tabindex="-1"></a>yield_spring <span class="ot">&lt;-</span> <span class="fu">set_units</span>(<span class="dv">700</span>, <span class="st">"gDW m-2 week-1"</span>)</span>
<span id="cb16-12"><a href="#cb16-12" aria-hidden="true" tabindex="-1"></a>yield_spring <span class="ot">&lt;-</span> yield_spring<span class="sc">/</span><span class="fu">set_units</span>(<span class="fl">0.48</span>, <span class="st">"m"</span>)</span>
<span id="cb16-13"><a href="#cb16-13" aria-hidden="true" tabindex="-1"></a>yield_spring <span class="ot">&lt;-</span> <span class="fu">set_units</span>(yield_spring<span class="sc">*</span>N_perc, <span class="st">"mg m-3 d-1"</span>)</span>
<span id="cb16-14"><a href="#cb16-14" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb16-15"><a href="#cb16-15" aria-hidden="true" tabindex="-1"></a>yield_winter <span class="ot">&lt;-</span> <span class="fu">set_units</span>(<span class="dv">520</span>, <span class="st">"gDW m-2 week-1"</span>)</span>
<span id="cb16-16"><a href="#cb16-16" aria-hidden="true" tabindex="-1"></a>yield_winter <span class="ot">&lt;-</span> yield_winter<span class="sc">/</span><span class="fu">set_units</span>(<span class="fl">0.48</span>, <span class="st">"m"</span>)</span>
<span id="cb16-17"><a href="#cb16-17" aria-hidden="true" tabindex="-1"></a>yield_winter <span class="ot">&lt;-</span> <span class="fu">set_units</span>(yield_winter<span class="sc">*</span>N_perc, <span class="st">"mg m-3 d-1"</span>)</span>
<span id="cb16-18"><a href="#cb16-18" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb16-19"><a href="#cb16-19" aria-hidden="true" tabindex="-1"></a>max_growth <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="at">spring =</span> yield_spring<span class="sc">/</span>stock_dens, <span class="at">winter =</span> yield_winter<span class="sc">/</span>stock_dens)</span>
<span id="cb16-20"><a href="#cb16-20" aria-hidden="true" tabindex="-1"></a>mu <span class="ot">&lt;-</span> <span class="fu">drop_units</span>(<span class="fu">max</span>(max_growth))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<p>The maximum growth rate (<span class="math inline">\(\mu\)</span>) for <em>A. armata</em> reported in <span class="citation" data-cites="schuenhoff_tetrasporophyte_2006">Schuenhoff, Mata, and Santos (<a href="#ref-schuenhoff_tetrasporophyte_2006" role="doc-biblioref">2006</a>)</span> was 34.72% in spring and 25.79% in winter. Hard to know if these are being limited by factors other than temperature/light, so I am going to use the maximum $=$34.7222222% for growth in all seasons.</p>
</section>
</section>
<section id="sec-temperature-tetrasporophyte" class="level2">
<h2 class="anchored" data-anchor-id="sec-temperature-tetrasporophyte">Temperature response</h2>
<p>Previous studies on the temperature tolerance of <em>Asparagopsis</em> tetrasporophytes were reviewed by <span class="citation" data-cites="zanolla_concise_2022">Zanolla et al. (<a href="#ref-zanolla_concise_2022" role="doc-biblioref">2022</a>)</span>, and <span class="citation" data-cites="chualain_invasive_2004">Chualáin et al. (<a href="#ref-chualain_invasive_2004" role="doc-biblioref">2004</a>)</span> collated previous studies’ observations on growth minima and maxima. They found that <em>A. armata</em> had a minimum growth temperature of 5-7<span class="math inline">\(^\circ\)</span>, an optimum growth temperature of 15 or 20<span class="math inline">\(^\circ\)</span>, and a maximum growth temperature of 25<span class="math inline">\(^\circ\)</span>C. However, it should be noted that within <span class="citation" data-cites="chualain_invasive_2004">Chualáin et al. (<a href="#ref-chualain_invasive_2004" role="doc-biblioref">2004</a>)</span> the only sample collected from Australia (Sorrento Back Beach, Victoria, from the lower eulittoral zone) had an optimum growth temperature of 15<span class="math inline">\(^\circ\)</span>C.</p>
<div class="cell">
<details class="code-fold">
<summary>Get temperature/irradiance data from <span class="citation" data-cites="zanolla_photosynthetic_2015">Zanolla et al. (<a href="#ref-zanolla_photosynthetic_2015" role="doc-biblioref">2015</a>)</span>, Figure 1</summary>
<div class="sourceCode cell-code" id="cb17"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb17-1"><a href="#cb17-1" aria-hidden="true" tabindex="-1"></a>zanolla1 <span class="ot">&lt;-</span> <span class="fu">list</span>(<span class="fu">read.csv</span>(<span class="fu">file.path</span>(<span class="st">"data_raw"</span>, <span class="st">"sources"</span>, <span class="st">"zanolla_2015"</span>, <span class="st">"12 degrees.csv"</span>), <span class="at">header =</span> F) <span class="sc">%&gt;%</span> <span class="fu">mutate</span>(<span class="at">temp =</span> <span class="dv">12</span>),</span>
<span id="cb17-2"><a href="#cb17-2" aria-hidden="true" tabindex="-1"></a>                 <span class="fu">read.csv</span>(<span class="fu">file.path</span>(<span class="st">"data_raw"</span>, <span class="st">"sources"</span>, <span class="st">"zanolla_2015"</span>, <span class="st">"15 degrees.csv"</span>), <span class="at">header =</span> F) <span class="sc">%&gt;%</span> <span class="fu">mutate</span>(<span class="at">temp =</span> <span class="dv">15</span>),</span>
<span id="cb17-3"><a href="#cb17-3" aria-hidden="true" tabindex="-1"></a>                 <span class="fu">read.csv</span>(<span class="fu">file.path</span>(<span class="st">"data_raw"</span>, <span class="st">"sources"</span>, <span class="st">"zanolla_2015"</span>, <span class="st">"18 degrees.csv"</span>), <span class="at">header =</span> F) <span class="sc">%&gt;%</span> <span class="fu">mutate</span>(<span class="at">temp =</span> <span class="dv">18</span>),</span>
<span id="cb17-4"><a href="#cb17-4" aria-hidden="true" tabindex="-1"></a>                 <span class="fu">read.csv</span>(<span class="fu">file.path</span>(<span class="st">"data_raw"</span>, <span class="st">"sources"</span>, <span class="st">"zanolla_2015"</span>, <span class="st">"22 degrees.csv"</span>), <span class="at">header =</span> F) <span class="sc">%&gt;%</span> <span class="fu">mutate</span>(<span class="at">temp =</span> <span class="dv">22</span>),</span>
<span id="cb17-5"><a href="#cb17-5" aria-hidden="true" tabindex="-1"></a>                 <span class="fu">read.csv</span>(<span class="fu">file.path</span>(<span class="st">"data_raw"</span>, <span class="st">"sources"</span>, <span class="st">"zanolla_2015"</span>, <span class="st">"26 degrees.csv"</span>), <span class="at">header =</span> F) <span class="sc">%&gt;%</span> <span class="fu">mutate</span>(<span class="at">temp =</span> <span class="dv">26</span>))</span>
<span id="cb17-6"><a href="#cb17-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-7"><a href="#cb17-7" aria-hidden="true" tabindex="-1"></a>zanolla1 <span class="ot">&lt;-</span> <span class="fu">bind_rows</span>(zanolla1)</span>
<span id="cb17-8"><a href="#cb17-8" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(zanolla1) <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">"Irradiance"</span>, <span class="st">"NPR"</span>, <span class="st">"Temperature"</span>)</span>
<span id="cb17-9"><a href="#cb17-9" aria-hidden="true" tabindex="-1"></a>zanolla1<span class="sc">$</span>NPR_rel <span class="ot">&lt;-</span> zanolla1<span class="sc">$</span>NPR<span class="sc">/</span><span class="fu">max</span>(zanolla1<span class="sc">$</span>NPR)</span>
<span id="cb17-10"><a href="#cb17-10" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-11"><a href="#cb17-11" aria-hidden="true" tabindex="-1"></a><span class="co"># # For when I need units</span></span>
<span id="cb17-12"><a href="#cb17-12" aria-hidden="true" tabindex="-1"></a><span class="co"># zanolla1$Irradiance &lt;- set_units(zanolla1$Irradiance, "umol m-2 s-1")</span></span>
<span id="cb17-13"><a href="#cb17-13" aria-hidden="true" tabindex="-1"></a><span class="co"># zanolla1$NPR &lt;- set_units(zanolla1$NPR, "mg gWW-1 h-1")</span></span>
<span id="cb17-14"><a href="#cb17-14" aria-hidden="true" tabindex="-1"></a><span class="co"># zanolla1$Temperature &lt;- set_units(zanolla1$Temperature, "C")</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<div class="cell">
<details class="code-fold">
<summary>Get temperature/irradiance data from <span class="citation" data-cites="zanolla_photosynthetic_2015">Zanolla et al. (<a href="#ref-zanolla_photosynthetic_2015" role="doc-biblioref">2015</a>)</span></summary>
<div class="sourceCode cell-code" id="cb18"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb18-1"><a href="#cb18-1" aria-hidden="true" tabindex="-1"></a>zanolla1 <span class="ot">&lt;-</span> zanolla1 <span class="sc">%&gt;%</span> </span>
<span id="cb18-2"><a href="#cb18-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">Irr_group =</span> <span class="fu">case_when</span>(</span>
<span id="cb18-3"><a href="#cb18-3" aria-hidden="true" tabindex="-1"></a>    Irradiance <span class="sc">&gt;</span> <span class="dv">0</span> <span class="sc">&amp;</span> Irradiance <span class="sc">&lt;</span> <span class="dv">10</span> <span class="sc">~</span> <span class="st">"~5.7"</span>, Irradiance <span class="sc">&gt;</span> <span class="dv">10</span> <span class="sc">&amp;</span> Irradiance <span class="sc">&lt;</span> <span class="dv">20</span> <span class="sc">~</span> <span class="st">"~19.8"</span>,</span>
<span id="cb18-4"><a href="#cb18-4" aria-hidden="true" tabindex="-1"></a>    Irradiance <span class="sc">&gt;</span> <span class="dv">20</span> <span class="sc">&amp;</span> Irradiance <span class="sc">&lt;</span> <span class="dv">35</span> <span class="sc">~</span> <span class="st">"~32.3"</span>, Irradiance <span class="sc">&gt;</span> <span class="dv">35</span> <span class="sc">&amp;</span> Irradiance <span class="sc">&lt;</span> <span class="dv">65</span> <span class="sc">~</span> <span class="st">"~55.3"</span>,</span>
<span id="cb18-5"><a href="#cb18-5" aria-hidden="true" tabindex="-1"></a>    Irradiance <span class="sc">&gt;</span> <span class="dv">65</span> <span class="sc">&amp;</span> Irradiance <span class="sc">&lt;</span> <span class="dv">130</span> <span class="sc">~</span> <span class="st">"~124.1"</span>, Irradiance <span class="sc">&gt;</span> <span class="dv">130</span> <span class="sc">&amp;</span> Irradiance <span class="sc">&lt;</span> <span class="dv">260</span> <span class="sc">~</span> <span class="st">"~254.4"</span>,</span>
<span id="cb18-6"><a href="#cb18-6" aria-hidden="true" tabindex="-1"></a>    Irradiance <span class="sc">&gt;</span> <span class="dv">260</span> <span class="sc">&amp;</span> Irradiance <span class="sc">&lt;</span> <span class="dv">370</span> <span class="sc">~</span> <span class="st">"~365.9"</span>, Irradiance <span class="sc">&gt;</span> <span class="dv">370</span> <span class="sc">&amp;</span> Irradiance <span class="sc">&lt;</span> <span class="dv">510</span> <span class="sc">~</span> <span class="st">"~504.5"</span>,</span>
<span id="cb18-7"><a href="#cb18-7" aria-hidden="true" tabindex="-1"></a>    Irradiance <span class="sc">&gt;</span> <span class="dv">510</span> <span class="sc">~</span> <span class="st">"~598.5"</span>, <span class="cn">TRUE</span> <span class="sc">~</span> <span class="cn">NA</span>),</span>
<span id="cb18-8"><a href="#cb18-8" aria-hidden="true" tabindex="-1"></a>    <span class="at">Irr_group =</span> <span class="fu">factor</span>(Irr_group, </span>
<span id="cb18-9"><a href="#cb18-9" aria-hidden="true" tabindex="-1"></a>                       <span class="at">levels =</span> <span class="fu">c</span>(<span class="st">"~5.7"</span>, <span class="st">"~19.8"</span>, <span class="st">"~32.3"</span>, <span class="st">"~55.3"</span>, <span class="st">"~124.1"</span>, <span class="st">"~254.4"</span>, <span class="st">"~365.9"</span>, <span class="st">"~504.5"</span>, <span class="st">"~598.5"</span>)))</span>
<span id="cb18-10"><a href="#cb18-10" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb18-11"><a href="#cb18-11" aria-hidden="true" tabindex="-1"></a>pT<span class="fl">.1</span> <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(dplyr<span class="sc">::</span><span class="fu">filter</span>(zanolla1, Irradiance <span class="sc">&gt;</span> <span class="dv">0</span>),</span>
<span id="cb18-12"><a href="#cb18-12" aria-hidden="true" tabindex="-1"></a>       <span class="fu">aes</span>(<span class="at">x =</span> Temperature, <span class="at">y =</span> NPR_rel, <span class="at">color =</span> Irr_group)) <span class="sc">+</span></span>
<span id="cb18-13"><a href="#cb18-13" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_line</span>() <span class="sc">+</span></span>
<span id="cb18-14"><a href="#cb18-14" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_classic</span>()</span>
<span id="cb18-15"><a href="#cb18-15" aria-hidden="true" tabindex="-1"></a><span class="co"># pT.1</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<div class="cell">
<details class="code-fold">
<summary>Setting temperature params</summary>
<div class="sourceCode cell-code" id="cb19"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb19-1"><a href="#cb19-1" aria-hidden="true" tabindex="-1"></a>both <span class="ot">&lt;-</span> <span class="fu">bind_rows</span>(</span>
<span id="cb19-2"><a href="#cb19-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">list</span>(</span>
<span id="cb19-3"><a href="#cb19-3" aria-hidden="true" tabindex="-1"></a>    <span class="fu">read.csv</span>(<span class="fu">file.path</span>(<span class="st">"data_raw"</span>, <span class="st">"sources"</span>, <span class="st">"chualain_2004"</span>, <span class="st">"armata.csv"</span>), <span class="at">header =</span> F) <span class="sc">%&gt;%</span> <span class="fu">mutate</span>(<span class="at">species =</span> <span class="st">"armata"</span>),</span>
<span id="cb19-4"><a href="#cb19-4" aria-hidden="true" tabindex="-1"></a>    <span class="fu">read.csv</span>(<span class="fu">file.path</span>(<span class="st">"data_raw"</span>, <span class="st">"sources"</span>, <span class="st">"chualain_2004"</span>, <span class="st">"taxiformis.csv"</span>), <span class="at">header =</span> F) <span class="sc">%&gt;%</span> <span class="fu">mutate</span>(<span class="at">species =</span> <span class="st">"taxiformis"</span>)</span>
<span id="cb19-5"><a href="#cb19-5" aria-hidden="true" tabindex="-1"></a>  )) <span class="sc">%&gt;%</span> </span>
<span id="cb19-6"><a href="#cb19-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">sample =</span> V3, </span>
<span id="cb19-7"><a href="#cb19-7" aria-hidden="true" tabindex="-1"></a>         <span class="at">temperature =</span> V2,</span>
<span id="cb19-8"><a href="#cb19-8" aria-hidden="true" tabindex="-1"></a>         <span class="at">variable =</span> <span class="fu">factor</span>(V4, <span class="at">levels =</span> <span class="fu">c</span>(<span class="st">" grow min"</span>, <span class="st">" grow max"</span>), <span class="at">labels =</span> <span class="fu">c</span>(<span class="st">"grow_min"</span>, <span class="st">"grow_max"</span>))) <span class="sc">%&gt;%</span> </span>
<span id="cb19-9"><a href="#cb19-9" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(species, variable) <span class="sc">%&gt;%</span> </span>
<span id="cb19-10"><a href="#cb19-10" aria-hidden="true" tabindex="-1"></a>  <span class="fu">reframe</span>(<span class="at">temperature =</span> <span class="fu">mean</span>(temperature))</span>
<span id="cb19-11"><a href="#cb19-11" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-12"><a href="#cb19-12" aria-hidden="true" tabindex="-1"></a>T_max_arma <span class="ot">&lt;-</span> both<span class="sc">$</span>temperature[both<span class="sc">$</span>species <span class="sc">==</span> <span class="st">"armata"</span> <span class="sc">&amp;</span> both<span class="sc">$</span>variable <span class="sc">==</span> <span class="st">"grow_max"</span>]</span>
<span id="cb19-13"><a href="#cb19-13" aria-hidden="true" tabindex="-1"></a>T_min_arma <span class="ot">&lt;-</span> both<span class="sc">$</span>temperature[both<span class="sc">$</span>species <span class="sc">==</span> <span class="st">"armata"</span> <span class="sc">&amp;</span> both<span class="sc">$</span>variable <span class="sc">==</span> <span class="st">"grow_min"</span>]</span>
<span id="cb19-14"><a href="#cb19-14" aria-hidden="true" tabindex="-1"></a>T_max_taxi <span class="ot">&lt;-</span> both<span class="sc">$</span>temperature[both<span class="sc">$</span>species <span class="sc">==</span> <span class="st">"taxiformis"</span> <span class="sc">&amp;</span> both<span class="sc">$</span>variable <span class="sc">==</span> <span class="st">"grow_max"</span>]</span>
<span id="cb19-15"><a href="#cb19-15" aria-hidden="true" tabindex="-1"></a>T_min_taxi <span class="ot">&lt;-</span> both<span class="sc">$</span>temperature[both<span class="sc">$</span>species <span class="sc">==</span> <span class="st">"taxiformis"</span> <span class="sc">&amp;</span> both<span class="sc">$</span>variable <span class="sc">==</span> <span class="st">"grow_min"</span>]</span>
<span id="cb19-16"><a href="#cb19-16" aria-hidden="true" tabindex="-1"></a>T_opt_arma <span class="ot">&lt;-</span> <span class="fu">mean</span>(<span class="fu">c</span>(<span class="dv">20</span>,<span class="dv">20</span>,<span class="dv">15</span>,<span class="dv">15</span>,<span class="dv">20</span>,<span class="dv">20</span>,<span class="dv">20</span>,<span class="dv">20</span>,<span class="dv">20</span>,<span class="dv">20</span>,<span class="dv">20</span>))</span>
<span id="cb19-17"><a href="#cb19-17" aria-hidden="true" tabindex="-1"></a>T_opt_taxi <span class="ot">&lt;-</span> <span class="fu">mean</span>(<span class="fu">c</span>(<span class="dv">25</span>,<span class="dv">25</span>,<span class="dv">25</span>,<span class="dv">25</span>,<span class="dv">25</span>,<span class="dv">25</span>,<span class="dv">25</span>,<span class="dv">25</span>,<span class="dv">25</span>,<span class="dv">20</span>,<span class="dv">25</span>,<span class="dv">25</span>,<span class="dv">25</span>))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<div class="cell">
<details class="code-fold">
<summary>Add a <span class="math inline">\(T_{lim}\)</span> function to plot of temperature/irradiance data from <span class="citation" data-cites="zanolla_photosynthetic_2015">Zanolla et al. (<a href="#ref-zanolla_photosynthetic_2015" role="doc-biblioref">2015</a>)</span>, Figure 1</summary>
<div class="sourceCode cell-code" id="cb20"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb20-1"><a href="#cb20-1" aria-hidden="true" tabindex="-1"></a>test_T <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="at">Tc =</span> <span class="fu">seq</span>(<span class="dv">5</span>, <span class="dv">28</span>, <span class="fl">0.5</span>))</span>
<span id="cb20-2"><a href="#cb20-2" aria-hidden="true" tabindex="-1"></a>test_params <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="at">T_opt =</span> T_opt_arma, <span class="at">T_max =</span> T_max_arma, <span class="at">T_min =</span> T_min_arma)</span>
<span id="cb20-3"><a href="#cb20-3" aria-hidden="true" tabindex="-1"></a>test_T<span class="sc">$</span>arma <span class="ot">&lt;-</span> <span class="fu">sapply</span>(<span class="at">X =</span> test_T<span class="sc">$</span>Tc, <span class="at">FUN =</span> T_lim, <span class="at">spec_params =</span> test_params)</span>
<span id="cb20-4"><a href="#cb20-4" aria-hidden="true" tabindex="-1"></a>test_params <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="at">T_opt =</span> T_opt_taxi, <span class="at">T_max =</span> T_max_taxi, <span class="at">T_min =</span> T_min_taxi)</span>
<span id="cb20-5"><a href="#cb20-5" aria-hidden="true" tabindex="-1"></a>test_T<span class="sc">$</span>taxi <span class="ot">&lt;-</span> <span class="fu">sapply</span>(<span class="at">X =</span> test_T<span class="sc">$</span>Tc, <span class="at">FUN =</span> T_lim, <span class="at">spec_params =</span> test_params)</span>
<span id="cb20-6"><a href="#cb20-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb20-7"><a href="#cb20-7" aria-hidden="true" tabindex="-1"></a>pT<span class="fl">.2</span> <span class="ot">&lt;-</span> pT<span class="fl">.1</span> <span class="sc">+</span></span>
<span id="cb20-8"><a href="#cb20-8" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_line</span>(<span class="at">data =</span> test_T, <span class="fu">aes</span>(<span class="at">x =</span> Tc, <span class="at">y =</span> arma), <span class="at">color =</span> <span class="st">"black"</span>) <span class="sc">+</span></span>
<span id="cb20-9"><a href="#cb20-9" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_line</span>(<span class="at">data =</span> test_T, <span class="fu">aes</span>(<span class="at">x =</span> Tc, <span class="at">y =</span> taxi), <span class="at">color =</span> <span class="st">"black"</span>, <span class="at">linetype =</span> <span class="st">"dashed"</span>) </span>
<span id="cb20-10"><a href="#cb20-10" aria-hidden="true" tabindex="-1"></a><span class="co"># pT.2</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<p>It seems that <em>A. taxiformis</em> has a slightly higher optimum temperature than <em>A. armata</em> * The optimum temperature for <em>A. armata</em> in <span class="citation" data-cites="zanolla_photosynthetic_2015">Zanolla et al. (<a href="#ref-zanolla_photosynthetic_2015" role="doc-biblioref">2015</a>)</span> was ~22<span class="math inline">\(^\circ\)</span>C, although they only tested 5 temperatures (12, 15, 18, 22, and 26<span class="math inline">\(^\circ\)</span>C). * <span class="citation" data-cites="sainz-villegas_exploring_2024">Sainz-Villegas et al. (<a href="#ref-sainz-villegas_exploring_2024" role="doc-biblioref">2024</a>)</span> also only tested three temperatures (15, 20, and 25<span class="math inline">\(^\circ\)</span>C). There was no positive growth at 25<span class="math inline">\(^\circ\)</span>C regardless of irradiance and 20<span class="math inline">\(^\circ\)</span>C showed mixed results. However, their irradiance was kept between 55-60 <span class="math inline">\(\mu\)</span>mol photons m<span class="math inline">\(^{-2}\)</span> s<span class="math inline">\(^{-1}\)</span>, which seems very low to me. * <span class="citation" data-cites="mata_effects_2006">Mata et al. (<a href="#ref-mata_effects_2006" role="doc-biblioref">2006</a>)</span> found that net growth was significantly decreased above 29<span class="math inline">\(^\circ\)</span>C due to an increase in respiration * <span class="citation" data-cites="mata_within-species_2017">Mata et al. (<a href="#ref-mata_within-species_2017" role="doc-biblioref">2017</a>)</span> investigated temperature limits of <em>A. taxiformis</em>, but like <span class="citation" data-cites="zanolla_photosynthetic_2015">Zanolla et al. (<a href="#ref-zanolla_photosynthetic_2015" role="doc-biblioref">2015</a>)</span> they didn’t actually do a full temperature range experiment and only tested three temperatures. Still, growth was best at 20.2<span class="math inline">\(^\circ\)</span>C.</p>
</section>
<section id="sec-light-tetrasporophyte" class="level2">
<h2 class="anchored" data-anchor-id="sec-light-tetrasporophyte">Light response</h2>
<p><span class="citation" data-cites="figueroa_use_2006">Figueroa et al. (<a href="#ref-figueroa_use_2006" role="doc-biblioref">2006</a>)</span> grew <em>A. armata</em> in fish pond effluent and measured <span class="math inline">\(K_d\)</span> at different macroalgae densities. From this I can work out an <span class="math inline">\(a_cs\)</span> (self-shading) value.</p>
<!-- $$ -->
<!-- K_d^{PAR} = ln(Ed_{z_2}/Ed_{z_1}) \times (z_1-z_2)-1 -->
<!-- $$ -->
<p>They expressed ETR in g DW but didn’t provide details of how g FW was converted to g DW. I’ll just use my previously calculated ratio of 4.</p>
<div class="cell">
<details class="code-fold">
<summary>Conditions in <span class="citation" data-cites="figueroa_use_2006">Figueroa et al. (<a href="#ref-figueroa_use_2006" role="doc-biblioref">2006</a>)</span></summary>
<div class="sourceCode cell-code" id="cb21"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb21-1"><a href="#cb21-1" aria-hidden="true" tabindex="-1"></a>get_Kd <span class="ot">&lt;-</span> <span class="cf">function</span>(<span class="at">z0 =</span> <span class="dv">0</span>, E0, z1, E1) {</span>
<span id="cb21-2"><a href="#cb21-2" aria-hidden="true" tabindex="-1"></a>  Kd <span class="ot">&lt;-</span> <span class="co">#log(E2/E1) * (z1-z2)-1</span></span>
<span id="cb21-3"><a href="#cb21-3" aria-hidden="true" tabindex="-1"></a>  <span class="dv">1</span><span class="sc">/</span>z1<span class="sc">*</span><span class="fu">log</span>(E0<span class="sc">/</span>E1)</span>
<span id="cb21-4"><a href="#cb21-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">return</span>(<span class="fu">unname</span>(Kd))</span>
<span id="cb21-5"><a href="#cb21-5" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb21-6"><a href="#cb21-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb21-7"><a href="#cb21-7" aria-hidden="true" tabindex="-1"></a>cond <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="at">E_0 =</span> <span class="fu">c</span>(<span class="dv">2112</span>, <span class="dv">2112</span>, <span class="dv">2112</span>, <span class="dv">2112</span>), <span class="co"># umol photons m-2 s-1</span></span>
<span id="cb21-8"><a href="#cb21-8" aria-hidden="true" tabindex="-1"></a>                  <span class="at">dens =</span> <span class="fu">c</span>(<span class="dv">0</span>, <span class="dv">4</span>, <span class="dv">6</span>, <span class="dv">8</span>), <span class="co"># g (FW?) L-1</span></span>
<span id="cb21-9"><a href="#cb21-9" aria-hidden="true" tabindex="-1"></a>                  <span class="at">E_1 =</span> <span class="fu">c</span>(<span class="dv">2112</span><span class="sc">*</span><span class="fl">0.9</span>, <span class="dv">264</span>, <span class="dv">220</span>, <span class="dv">110</span>))</span>
<span id="cb21-10"><a href="#cb21-10" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb21-11"><a href="#cb21-11" aria-hidden="true" tabindex="-1"></a>cond<span class="sc">$</span>Kd <span class="ot">&lt;-</span> <span class="fu">sapply</span>(<span class="at">X =</span> cond<span class="sc">$</span>E_1, <span class="at">FUN =</span> get_Kd, <span class="at">E0 =</span> <span class="dv">2112</span>, <span class="at">z1 =</span> <span class="sc">-</span><span class="fl">0.1</span>, <span class="at">z0 =</span> <span class="dv">0</span>) <span class="sc">*</span> <span class="sc">-</span><span class="dv">1</span></span>
<span id="cb21-12"><a href="#cb21-12" aria-hidden="true" tabindex="-1"></a>cond<span class="sc">$</span>Kma <span class="ot">&lt;-</span> cond<span class="sc">$</span>Kd <span class="sc">-</span> cond<span class="sc">$</span>Kd[cond<span class="sc">$</span>dens <span class="sc">==</span> <span class="dv">0</span>]</span>
<span id="cb21-13"><a href="#cb21-13" aria-hidden="true" tabindex="-1"></a>cond<span class="sc">$</span>dens.DW <span class="ot">&lt;-</span> <span class="fu">set_units</span>(cond<span class="sc">$</span>dens<span class="sc">*</span>DWWW, <span class="st">"gDW L-1"</span>) <span class="co"># g WW L-1</span></span>
<span id="cb21-14"><a href="#cb21-14" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb21-15"><a href="#cb21-15" aria-hidden="true" tabindex="-1"></a><span class="co"># Figueroa 2006 did not measure N percentage, but we can take a minimum and maximum of 0.055 and 0.065 for starters</span></span>
<span id="cb21-16"><a href="#cb21-16" aria-hidden="true" tabindex="-1"></a>cond<span class="sc">$</span>dens.N.hi <span class="ot">&lt;-</span> cond<span class="sc">$</span>dens.DW <span class="sc">*</span> N_perc</span>
<span id="cb21-17"><a href="#cb21-17" aria-hidden="true" tabindex="-1"></a>cond<span class="sc">$</span>dens.N.lo <span class="ot">&lt;-</span> cond<span class="sc">$</span>dens.DW <span class="sc">*</span> <span class="fu">set_units</span>(N_min, <span class="st">"g gDW-1"</span>)</span>
<span id="cb21-18"><a href="#cb21-18" aria-hidden="true" tabindex="-1"></a>cond<span class="sc">$</span>dens.N.hi <span class="ot">&lt;-</span> <span class="fu">set_units</span>(cond<span class="sc">$</span>dens.N.hi, <span class="st">"mg m-3"</span>)</span>
<span id="cb21-19"><a href="#cb21-19" aria-hidden="true" tabindex="-1"></a>cond<span class="sc">$</span>dens.N.lo <span class="ot">&lt;-</span> <span class="fu">set_units</span>(cond<span class="sc">$</span>dens.N.lo, <span class="st">"mg m-3"</span>)</span>
<span id="cb21-20"><a href="#cb21-20" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb21-21"><a href="#cb21-21" aria-hidden="true" tabindex="-1"></a><span class="co"># Within the model, k_ma = Nf * h_m * a_cs * max(h_m/d_top, 1) * 1/(min(h_m, d_top)), d_top =/= 0</span></span>
<span id="cb21-22"><a href="#cb21-22" aria-hidden="true" tabindex="-1"></a><span class="co"># Assuming a maximum height of 26cm because the algae are so dense</span></span>
<span id="cb21-23"><a href="#cb21-23" aria-hidden="true" tabindex="-1"></a>cond<span class="sc">$</span>a_cs.lo <span class="ot">&lt;-</span> <span class="dv">1</span><span class="sc">/</span>(cond<span class="sc">$</span>dens.N.lo <span class="sc">*</span> </span>
<span id="cb21-24"><a href="#cb21-24" aria-hidden="true" tabindex="-1"></a>                     <span class="fu">set_units</span>(<span class="fl">0.26</span>, <span class="st">"m"</span>) <span class="sc">*</span> </span>
<span id="cb21-25"><a href="#cb21-25" aria-hidden="true" tabindex="-1"></a>                     <span class="fu">set_units</span>(<span class="fu">pmax</span>(<span class="fl">0.26</span><span class="sc">/</span><span class="fl">0.1</span>, <span class="dv">1</span>), <span class="st">"m"</span>) <span class="sc">*</span> </span>
<span id="cb21-26"><a href="#cb21-26" aria-hidden="true" tabindex="-1"></a>                     <span class="fu">set_units</span>(<span class="dv">1</span><span class="sc">/</span>(<span class="fu">pmin</span>(<span class="fl">0.26</span>, <span class="fl">0.1</span>)), <span class="st">"m"</span>) <span class="sc">*</span> </span>
<span id="cb21-27"><a href="#cb21-27" aria-hidden="true" tabindex="-1"></a>                     <span class="fu">set_units</span>(<span class="dv">1</span><span class="sc">/</span>cond<span class="sc">$</span>Kma, <span class="st">"m-1"</span>))</span>
<span id="cb21-28"><a href="#cb21-28" aria-hidden="true" tabindex="-1"></a>cond<span class="sc">$</span>a_cs.hi <span class="ot">&lt;-</span> <span class="dv">1</span><span class="sc">/</span>(cond<span class="sc">$</span>dens.N.hi <span class="sc">*</span> </span>
<span id="cb21-29"><a href="#cb21-29" aria-hidden="true" tabindex="-1"></a>                     <span class="fu">set_units</span>(<span class="fl">0.26</span>, <span class="st">"m"</span>) <span class="sc">*</span> </span>
<span id="cb21-30"><a href="#cb21-30" aria-hidden="true" tabindex="-1"></a>                     <span class="fu">set_units</span>(<span class="fu">pmax</span>(<span class="fl">0.26</span><span class="sc">/</span><span class="fl">0.1</span>, <span class="dv">1</span>), <span class="st">"m"</span>) <span class="sc">*</span> </span>
<span id="cb21-31"><a href="#cb21-31" aria-hidden="true" tabindex="-1"></a>                     <span class="fu">set_units</span>(<span class="dv">1</span><span class="sc">/</span>(<span class="fu">pmin</span>(<span class="fl">0.26</span>, <span class="fl">0.1</span>)), <span class="st">"m"</span>) <span class="sc">*</span> </span>
<span id="cb21-32"><a href="#cb21-32" aria-hidden="true" tabindex="-1"></a>                     <span class="fu">set_units</span>(<span class="dv">1</span><span class="sc">/</span>cond<span class="sc">$</span>Kma, <span class="st">"m-1"</span>))</span>
<span id="cb21-33"><a href="#cb21-33" aria-hidden="true" tabindex="-1"></a>all_acs <span class="ot">&lt;-</span> <span class="fu">c</span>(cond<span class="sc">$</span>a_cs.lo[<span class="sc">!</span><span class="fu">is.na</span>(cond<span class="sc">$</span>a_cs.lo)], cond<span class="sc">$</span>a_cs.hi[<span class="sc">!</span><span class="fu">is.na</span>(cond<span class="sc">$</span>a_cs.hi)])</span>
<span id="cb21-34"><a href="#cb21-34" aria-hidden="true" tabindex="-1"></a>a_cs <span class="ot">&lt;-</span> <span class="fu">mean</span>(all_acs)</span>
<span id="cb21-35"><a href="#cb21-35" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb21-36"><a href="#cb21-36" aria-hidden="true" tabindex="-1"></a>all_acs <span class="ot">&lt;-</span> <span class="fu">drop_units</span>(all_acs)</span>
<span id="cb21-37"><a href="#cb21-37" aria-hidden="true" tabindex="-1"></a>a_cs <span class="ot">&lt;-</span> <span class="fu">drop_units</span>(a_cs)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<p>The above (from <span class="citation" data-cites="figueroa_use_2006">Figueroa et al. (<a href="#ref-figueroa_use_2006" role="doc-biblioref">2006</a>)</span>) actually gave a fairly narrow range of <span class="math inline">\(a_{cs}\)</span> values to use, from <span class="math inline">\(2.027 \times 10^{-6}\)</span> to <span class="math inline">\(3.337 \times 10^{-6}\)</span>. Taking an average of those values gives <span class="math inline">\(a_{cs}=\)</span><span class="math inline">\(2.509 \times 10^{-6}\)</span>.</p>
<p>The optimum irradiance in <span class="citation" data-cites="zanolla_photosynthetic_2015">Zanolla et al. (<a href="#ref-zanolla_photosynthetic_2015" role="doc-biblioref">2015</a>)</span> was ~150 <span class="math inline">\(\mu\)</span>mol photons m<span class="math inline">\(^{-2}\)</span> s<span class="math inline">\(^{-1}\)</span> but this was only obvious in the 22<span class="math inline">\(^\circ\)</span> trials.</p>
<div class="cell">
<details class="code-fold">
<summary>Plot of temperature/irradiance data from <span class="citation" data-cites="zanolla_photosynthetic_2015">Zanolla et al. (<a href="#ref-zanolla_photosynthetic_2015" role="doc-biblioref">2015</a>)</span> Figure 1</summary>
<div class="sourceCode cell-code" id="cb22"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb22-1"><a href="#cb22-1" aria-hidden="true" tabindex="-1"></a>pI<span class="fl">.1</span> <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(zanolla1, <span class="fu">aes</span>(<span class="at">x =</span> Irradiance, <span class="at">y =</span> NPR_rel, <span class="at">colour =</span> <span class="fu">as.factor</span>(Temperature))) <span class="sc">+</span></span>
<span id="cb22-2"><a href="#cb22-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_line</span>() <span class="sc">+</span></span>
<span id="cb22-3"><a href="#cb22-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_y_continuous</span>(<span class="at">limits =</span> <span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">1</span>)) <span class="sc">+</span></span>
<span id="cb22-4"><a href="#cb22-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_classic</span>()</span>
<span id="cb22-5"><a href="#cb22-5" aria-hidden="true" tabindex="-1"></a>pI<span class="fl">.1</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><img src="asparagopsis-parameterisation_files/figure-html/zanolla%202015%20irradiance%20plot-1.png" class="img-fluid figure-img" width="672"></p>
<figcaption>Relative NPR<span class="math inline">\(_{max}\)</span> with changing irradiance.</figcaption>
</figure>
</div>
</div>
</div>
<p>Using <span class="math inline">\(I_o=150\)</span> in the <span class="math inline">\(I_{lim}\)</span> function while trying to approximate the culture conditions (clear water, algae grown close to the surface) specified in <span class="citation" data-cites="zanolla_photosynthetic_2015">Zanolla et al. (<a href="#ref-zanolla_photosynthetic_2015" role="doc-biblioref">2015</a>)</span> produces a good approximation (black line, below). The self-shading constant used below is the one calculated from <span class="citation" data-cites="figueroa_use_2006">Figueroa et al. (<a href="#ref-figueroa_use_2006" role="doc-biblioref">2006</a>)</span> above.</p>
<div class="cell">
<details class="code-fold">
<summary>Add an <span class="math inline">\(I_{lim}\)</span> function to plot of temperature/irradiance data from <span class="citation" data-cites="zanolla_photosynthetic_2015">Zanolla et al. (<a href="#ref-zanolla_photosynthetic_2015" role="doc-biblioref">2015</a>)</span> Figure 1</summary>
<div class="sourceCode cell-code" id="cb23"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb23-1"><a href="#cb23-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Convert density to N</span></span>
<span id="cb23-2"><a href="#cb23-2" aria-hidden="true" tabindex="-1"></a>dens<span class="fl">.1</span> <span class="ot">&lt;-</span> <span class="fu">set_units</span>(<span class="fl">0.04</span><span class="sc">/</span>DWWW, <span class="st">"gDW L-1"</span>) <span class="co"># Zanolla tested 1 individual in lab settings</span></span>
<span id="cb23-3"><a href="#cb23-3" aria-hidden="true" tabindex="-1"></a>dens<span class="fl">.1</span> <span class="ot">&lt;-</span> dens<span class="fl">.1</span><span class="sc">*</span>N_perc</span>
<span id="cb23-4"><a href="#cb23-4" aria-hidden="true" tabindex="-1"></a>dens<span class="fl">.1</span> <span class="ot">&lt;-</span> <span class="fu">set_units</span>(dens<span class="fl">.1</span>, <span class="st">"mg m-3"</span>)</span>
<span id="cb23-5"><a href="#cb23-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb23-6"><a href="#cb23-6" aria-hidden="true" tabindex="-1"></a>I_o <span class="ot">&lt;-</span> <span class="dv">150</span></span>
<span id="cb23-7"><a href="#cb23-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb23-8"><a href="#cb23-8" aria-hidden="true" tabindex="-1"></a>test_I <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="at">Ic =</span> <span class="fu">seq</span>(<span class="dv">0</span>, <span class="dv">800</span>, <span class="dv">20</span>))</span>
<span id="cb23-9"><a href="#cb23-9" aria-hidden="true" tabindex="-1"></a>site_params_light <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="at">d_top =</span> <span class="fl">0.05</span>, <span class="at">hc =</span> <span class="dv">1</span>, <span class="at">kW =</span> <span class="dv">1</span><span class="sc">/</span><span class="dv">100</span>)</span>
<span id="cb23-10"><a href="#cb23-10" aria-hidden="true" tabindex="-1"></a>spec_params_light <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="at">a_cs =</span> a_cs, <span class="co"># This is the a_cs value from Figueroa 2006</span></span>
<span id="cb23-11"><a href="#cb23-11" aria-hidden="true" tabindex="-1"></a>                       <span class="at">I_o =</span> I_o, </span>
<span id="cb23-12"><a href="#cb23-12" aria-hidden="true" tabindex="-1"></a>                       <span class="at">h_max =</span> <span class="fl">0.1</span>) <span class="co"># height is constant at full culture depth range</span></span>
<span id="cb23-13"><a href="#cb23-13" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb23-14"><a href="#cb23-14" aria-hidden="true" tabindex="-1"></a>test_I<span class="sc">$</span>I_lim <span class="ot">&lt;-</span> <span class="fu">sapply</span>(<span class="at">X =</span> test_I<span class="sc">$</span>Ic, <span class="at">FUN =</span> I_lim, <span class="at">Nf =</span> <span class="fu">drop_units</span>(dens<span class="fl">.1</span>), </span>
<span id="cb23-15"><a href="#cb23-15" aria-hidden="true" tabindex="-1"></a>                       <span class="at">spec_params =</span> spec_params_light, <span class="at">site_params =</span> site_params_light)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<div class="cell">
<details class="code-fold">
<summary>Plot <span class="math inline">\(I_{lim}\)</span> function test</summary>
<div class="sourceCode cell-code" id="cb24"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb24-1"><a href="#cb24-1" aria-hidden="true" tabindex="-1"></a>pI<span class="fl">.1</span> <span class="sc">+</span></span>
<span id="cb24-2"><a href="#cb24-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_line</span>(<span class="at">data =</span> test_I, <span class="fu">aes</span>(<span class="at">x =</span> Ic, <span class="at">y =</span> I_lim), <span class="at">color =</span> <span class="st">"black"</span>) <span class="sc">+</span></span>
<span id="cb24-3"><a href="#cb24-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_vline</span>(<span class="at">xintercept =</span> <span class="dv">150</span>, <span class="at">color =</span> <span class="st">"grey"</span>, <span class="at">linetype =</span> <span class="st">"dashed"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><img src="asparagopsis-parameterisation_files/figure-html/Ilim%20test%20plot-1.png" class="img-fluid figure-img" width="672"></p>
<figcaption>Relative NPR<span class="math inline">\(_{max}\)</span> with changing irradiance from Zanolla et al.&nbsp;(2015) (colors) and from the <span class="math inline">\(I_{lim}\)</span> function (black line). The grey dashed line shows the peak of <span class="math inline">\(I_o=150\)</span> <span class="math inline">\(\mu\)</span>mol photons m<span class="math inline">\(^{-2}\)</span> s<span class="math inline">\(^{-1}\)</span>.</figcaption>
</figure>
</div>
</div>
</div>
</section>
<section id="fragmentation-with-exposure" class="level2">
<h2 class="anchored" data-anchor-id="fragmentation-with-exposure">Fragmentation with exposure</h2>
<p>“The highest percentage of biomass lost from any single tetrasporophyte in the experiment was 19.34% in the high turbulence treatment, and the lowest was 0.95% biomass loss in the static treatment.” <span class="citation" data-cites="hall_investigating_2023">(<a href="#ref-hall_investigating_2023" role="doc-biblioref">Hall 2023, 37</a>)</span></p>
<p>Since the tetrasporophyte is not being grown in open water, the biomass lost due to velocity will be 0 and they will lose biomass from turbulence only.</p>
<div class="cell">
<details class="code-fold">
<summary>Get <span class="citation" data-cites="hall_investigating_2023">Hall (<a href="#ref-hall_investigating_2023" role="doc-biblioref">2023</a>)</span> turbulence data for tetrasporophytes</summary>
<div class="sourceCode cell-code" id="cb25"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb25-1"><a href="#cb25-1" aria-hidden="true" tabindex="-1"></a>hall <span class="ot">&lt;-</span> <span class="fu">read.csv</span>(<span class="fu">file.path</span>(<span class="st">"data_raw"</span>, <span class="st">"sources"</span>, <span class="st">"hall_2023"</span>, <span class="st">"hall_2023.csv"</span>))</span>
<span id="cb25-2"><a href="#cb25-2" aria-hidden="true" tabindex="-1"></a>ls <span class="ot">&lt;-</span> <span class="fu">split.default</span>(hall, <span class="fu">rep</span>(<span class="dv">1</span><span class="sc">:</span><span class="dv">12</span>, <span class="at">each =</span> <span class="dv">2</span>))</span>
<span id="cb25-3"><a href="#cb25-3" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> (i <span class="cf">in</span> <span class="dv">1</span><span class="sc">:</span><span class="fu">length</span>(ls)){</span>
<span id="cb25-4"><a href="#cb25-4" aria-hidden="true" tabindex="-1"></a>  ls[[i]] <span class="ot">&lt;-</span> ls[[i]] <span class="sc">%&gt;%</span> </span>
<span id="cb25-5"><a href="#cb25-5" aria-hidden="true" tabindex="-1"></a>    <span class="fu">mutate</span>(<span class="at">dataset =</span> <span class="fu">colnames</span>(ls[[i]][<span class="dv">1</span>]))</span>
<span id="cb25-6"><a href="#cb25-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">colnames</span>(ls[[i]]) <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">"Label"</span>, <span class="st">"perc_biomass_lost"</span>, <span class="st">"dataset"</span>)</span>
<span id="cb25-7"><a href="#cb25-7" aria-hidden="true" tabindex="-1"></a>  ls[[i]] <span class="ot">&lt;-</span> ls[[i]] <span class="sc">%&gt;%</span> </span>
<span id="cb25-8"><a href="#cb25-8" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">filter</span>(Label <span class="sc">!=</span> <span class="st">"Label"</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb25-9"><a href="#cb25-9" aria-hidden="true" tabindex="-1"></a>    <span class="fu">mutate</span>(<span class="at">perc_biomass_lost =</span> <span class="fu">as.numeric</span>(perc_biomass_lost)) <span class="sc">%&gt;%</span> </span>
<span id="cb25-10"><a href="#cb25-10" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">filter</span>(<span class="sc">!</span><span class="fu">is.na</span>(perc_biomass_lost))</span>
<span id="cb25-11"><a href="#cb25-11" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb25-12"><a href="#cb25-12" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb25-13"><a href="#cb25-13" aria-hidden="true" tabindex="-1"></a>turbulence_levels <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">'static'</span>, <span class="st">'low'</span>, <span class="st">'mid'</span>, <span class="st">'high'</span>)</span>
<span id="cb25-14"><a href="#cb25-14" aria-hidden="true" tabindex="-1"></a>velocity_levels <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">'low'</span>, <span class="st">'low.mid'</span>, <span class="st">'mid'</span>, <span class="st">'mid.high'</span>, <span class="st">'high'</span>, <span class="st">'max.high'</span>)</span>
<span id="cb25-15"><a href="#cb25-15" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb25-16"><a href="#cb25-16" aria-hidden="true" tabindex="-1"></a>hall_2<span class="fl">.9</span> <span class="ot">&lt;-</span> <span class="fu">bind_rows</span>(<span class="fu">list</span>(ls[[<span class="dv">1</span>]], ls[[<span class="dv">2</span>]])) <span class="sc">%&gt;%</span> </span>
<span id="cb25-17"><a href="#cb25-17" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">Label =</span> <span class="fu">rep</span>(<span class="fu">c</span>(<span class="st">'static'</span>, <span class="st">'low'</span>, <span class="st">'mid'</span>, <span class="st">'high'</span>), <span class="dv">2</span>),</span>
<span id="cb25-18"><a href="#cb25-18" aria-hidden="true" tabindex="-1"></a>         <span class="at">Exp =</span> <span class="st">"Turbulence"</span>,</span>
<span id="cb25-19"><a href="#cb25-19" aria-hidden="true" tabindex="-1"></a>         <span class="at">Form =</span> <span class="st">"Tetrasporophyte"</span>,</span>
<span id="cb25-20"><a href="#cb25-20" aria-hidden="true" tabindex="-1"></a>         <span class="at">Data =</span> <span class="fu">c</span>(<span class="fu">rep</span>(<span class="st">"Mean"</span>, <span class="dv">4</span>), <span class="fu">rep</span>(<span class="st">"Mean+SE"</span>, <span class="dv">4</span>))) <span class="sc">%&gt;%</span> </span>
<span id="cb25-21"><a href="#cb25-21" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">select</span>(<span class="sc">-</span>dataset) <span class="sc">%&gt;%</span> </span>
<span id="cb25-22"><a href="#cb25-22" aria-hidden="true" tabindex="-1"></a>  <span class="fu">pivot_wider</span>(<span class="at">names_from =</span> Data, <span class="at">values_from =</span> perc_biomass_lost) <span class="sc">%&gt;%</span> </span>
<span id="cb25-23"><a href="#cb25-23" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">SE =</span> <span class="st">`</span><span class="at">Mean+SE</span><span class="st">`</span> <span class="sc">-</span> Mean) <span class="sc">%&gt;%</span> </span>
<span id="cb25-24"><a href="#cb25-24" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">select</span>(<span class="sc">-</span><span class="st">`</span><span class="at">Mean+SE</span><span class="st">`</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb25-25"><a href="#cb25-25" aria-hidden="true" tabindex="-1"></a>  <span class="fu">pivot_longer</span>(<span class="at">names_to =</span> <span class="st">"Data"</span>, <span class="at">values_to =</span> <span class="st">"perc_biomass_lost"</span>, <span class="at">cols =</span> <span class="fu">c</span>(Mean, SE)) <span class="sc">%&gt;%</span> </span>
<span id="cb25-26"><a href="#cb25-26" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">Label =</span> <span class="fu">factor</span>(Label, <span class="at">levels =</span> turbulence_levels))</span>
<span id="cb25-27"><a href="#cb25-27" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb25-28"><a href="#cb25-28" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>(<span class="fu">filter</span>(hall_2<span class="fl">.9</span>, Data <span class="sc">==</span> <span class="st">"Mean"</span>)) <span class="sc">+</span></span>
<span id="cb25-29"><a href="#cb25-29" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_col</span>(<span class="fu">aes</span>(<span class="at">x =</span> Label, <span class="at">y =</span> perc_biomass_lost<span class="sc">*</span><span class="dv">100</span>)) <span class="sc">+</span></span>
<span id="cb25-30"><a href="#cb25-30" aria-hidden="true" tabindex="-1"></a>  prettyplot <span class="sc">+</span></span>
<span id="cb25-31"><a href="#cb25-31" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">x =</span> <span class="st">"Level of turbulence"</span>, <span class="at">y =</span> <span class="st">"Mean % of biomass loss"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><img src="asparagopsis-parameterisation_files/figure-html/get%20hall%202023%20data-1.png" class="img-fluid figure-img" width="672"></p>
<figcaption>Data from Figure 2.9 of <span class="citation" data-cites="hall_investigating_2023">Hall (<a href="#ref-hall_investigating_2023" role="doc-biblioref">2023</a>)</span></figcaption>
</figure>
</div>
</div>
</div>
<div class="cell">
<details class="code-fold">
<summary>Get turbulence data ready for vector</summary>
<div class="sourceCode cell-code" id="cb26"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb26-1"><a href="#cb26-1" aria-hidden="true" tabindex="-1"></a>D_st <span class="ot">&lt;-</span> hall_2<span class="fl">.9</span><span class="sc">$</span>perc_biomass_lost[hall_2<span class="fl">.9</span><span class="sc">$</span>Data <span class="sc">==</span> <span class="st">"Mean"</span> <span class="sc">&amp;</span> hall_2<span class="fl">.9</span><span class="sc">$</span>Label <span class="sc">==</span> <span class="st">"static"</span>]</span>
<span id="cb26-2"><a href="#cb26-2" aria-hidden="true" tabindex="-1"></a>D_lo <span class="ot">&lt;-</span> hall_2<span class="fl">.9</span><span class="sc">$</span>perc_biomass_lost[hall_2<span class="fl">.9</span><span class="sc">$</span>Data <span class="sc">==</span> <span class="st">"Mean"</span> <span class="sc">&amp;</span> hall_2<span class="fl">.9</span><span class="sc">$</span>Label <span class="sc">==</span> <span class="st">"low"</span>]</span>
<span id="cb26-3"><a href="#cb26-3" aria-hidden="true" tabindex="-1"></a>D_mi <span class="ot">&lt;-</span> hall_2<span class="fl">.9</span><span class="sc">$</span>perc_biomass_lost[hall_2<span class="fl">.9</span><span class="sc">$</span>Data <span class="sc">==</span> <span class="st">"Mean"</span> <span class="sc">&amp;</span> hall_2<span class="fl">.9</span><span class="sc">$</span>Label <span class="sc">==</span> <span class="st">"mid"</span>]</span>
<span id="cb26-4"><a href="#cb26-4" aria-hidden="true" tabindex="-1"></a>D_hi <span class="ot">&lt;-</span> hall_2<span class="fl">.9</span><span class="sc">$</span>perc_biomass_lost[hall_2<span class="fl">.9</span><span class="sc">$</span>Data <span class="sc">==</span> <span class="st">"Mean"</span> <span class="sc">&amp;</span> hall_2<span class="fl">.9</span><span class="sc">$</span>Label <span class="sc">==</span> <span class="st">"high"</span>]</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
</section>
</section>
<section id="finishing-up" class="level1">
<h1>Finishing up</h1>
<section id="sec-final-params-tetrasporophyte" class="level2">
<h2 class="anchored" data-anchor-id="sec-final-params-tetrasporophyte">All parameters and sources</h2>
<p>This is exactly what’s going into the species vectors.</p>
<table class="caption-top table">
<caption>Final values for species parameters before saving.</caption>
<colgroup>
<col style="width: 25%">
<col style="width: 24%">
<col style="width: 50%">
</colgroup>
<thead>
<tr class="header">
<th>Parameter</th>
<th style="text-align: center;">Value</th>
<th style="text-align: center;">Source</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td><span class="math inline">\(V_{am}\)</span>, <span class="math inline">\(K_{am}\)</span></td>
<td style="text-align: center;">4.9919879, 140.207067</td>
<td style="text-align: center;"></td>
</tr>
<tr class="even">
<td><span class="math inline">\(M_{am}\)</span>, <span class="math inline">\(C_{am}\)</span></td>
<td style="text-align: center;">0.00336, 2.9548534</td>
<td style="text-align: center;"></td>
</tr>
<tr class="odd">
<td><span class="math inline">\(V_{ni}\)</span>, <span class="math inline">\(K_{ni}\)</span></td>
<td style="text-align: center;">NA, NA</td>
<td style="text-align: center;"></td>
</tr>
<tr class="even">
<td><span class="math inline">\(M_{ni}\)</span>, <span class="math inline">\(C_{ni}\)</span></td>
<td style="text-align: center;">0, 0.1138183</td>
<td style="text-align: center;"></td>
</tr>
<tr class="odd">
<td><span class="math inline">\(V_{ot}\)</span>, <span class="math inline">\(K_{ot}\)</span></td>
<td style="text-align: center;">NA, NA</td>
<td style="text-align: center;"></td>
</tr>
<tr class="even">
<td><span class="math inline">\(M_{ot}\)</span>, <span class="math inline">\(C_{ot}\)</span></td>
<td style="text-align: center;">1.44^{-4}, 0.2958215</td>
<td style="text-align: center;"></td>
</tr>
<tr class="odd">
<td><span class="math inline">\(K_{c}\)</span></td>
<td style="text-align: center;">6</td>
<td style="text-align: center;"></td>
</tr>
<tr class="even">
<td><span class="math inline">\(Q_{min}\)</span>, <span class="math inline">\(Q_{max}\)</span></td>
<td style="text-align: center;">54.7, 65</td>
<td style="text-align: center;"></td>
</tr>
<tr class="odd">
<td><span class="math inline">\(N_{min}\)</span>, <span class="math inline">\(N_{max}\)</span></td>
<td style="text-align: center;">0.0547, 0.065</td>
<td style="text-align: center;"></td>
</tr>
<tr class="even">
<td><span class="math inline">\(DWWW\)</span></td>
<td style="text-align: center;">4</td>
<td style="text-align: center;"></td>
</tr>
<tr class="odd">
<td><span class="math inline">\(\mu\)</span></td>
<td style="text-align: center;">0.3472222</td>
<td style="text-align: center;"></td>
</tr>
<tr class="even">
<td><span class="math inline">\(D_{m}\)</span>, <span class="math inline">\(D_{v}\)</span></td>
<td style="text-align: center;">NA, NA</td>
<td style="text-align: center;"></td>
</tr>
<tr class="odd">
<td><span class="math inline">\(D_{lo}\)</span>, <span class="math inline">\(D_{mi}\)</span>, <span class="math inline">\(D_{hi}\)</span></td>
<td style="text-align: center;">0.0487879, 0.0769697, 0.1137879</td>
<td style="text-align: center;"></td>
</tr>
<tr class="even">
<td><span class="math inline">\(I_o\)</span>, <span class="math inline">\(a_{cs}\)</span></td>
<td style="text-align: center;">150, 2.5090131^{-6}</td>
<td style="text-align: center;"></td>
</tr>
<tr class="odd">
<td><span class="math inline">\(T_{opt}\)</span>, <span class="math inline">\(T_{min}\)</span>, <span class="math inline">\(T_{max}\)</span> (<em>A. armata</em>)</td>
<td style="text-align: center;">19.0909091, 8.8954869, 22.4292359</td>
<td style="text-align: center;"></td>
</tr>
<tr class="even">
<td><span class="math inline">\(T_{opt}\)</span>, <span class="math inline">\(T_{min}\)</span>, <span class="math inline">\(T_{max}\)</span> (<em>A. taxiformis</em>)</td>
<td style="text-align: center;">24.6153846, 16.0326603, 29.226247</td>
<td style="text-align: center;"></td>
</tr>
<tr class="odd">
<td><span class="math inline">\(S_{opt}\)</span>, <span class="math inline">\(S_{min}\)</span>, <span class="math inline">\(S_{max}\)</span></td>
<td style="text-align: center;">35.25, 21.15, 41.65</td>
<td style="text-align: center;"></td>
</tr>
<tr class="even">
<td><span class="math inline">\(h_a\)</span>, <span class="math inline">\(h_b\)</span>, <span class="math inline">\(h_c\)</span>, <span class="math inline">\(h_{max}\)</span></td>
<td style="text-align: center;">NA, NA, NA, 1</td>
<td style="text-align: center;"></td>
</tr>
</tbody>
</table>
</section>
<section id="sec-save-tetrasporophyte" class="level2">
<h2 class="anchored" data-anchor-id="sec-save-tetrasporophyte">Saving data</h2>
<p>The model uses a named vector, which can be loaded straight from Rdata files, but I’m saving as a .csv file because that’s easier for human reading.</p>
<div class="cell">
<details class="code-fold">
<summary>Insert values and units into named vector</summary>
<div class="sourceCode cell-code" id="cb27"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb27-1"><a href="#cb27-1" aria-hidden="true" tabindex="-1"></a>a_armata_tetrasporophyte <span class="ot">&lt;-</span> <span class="fu">c</span>(</span>
<span id="cb27-2"><a href="#cb27-2" aria-hidden="true" tabindex="-1"></a>  <span class="at">V_am =</span> V_am,</span>
<span id="cb27-3"><a href="#cb27-3" aria-hidden="true" tabindex="-1"></a>  <span class="at">K_am =</span> K_am,</span>
<span id="cb27-4"><a href="#cb27-4" aria-hidden="true" tabindex="-1"></a>  <span class="at">M_am =</span> M_am,</span>
<span id="cb27-5"><a href="#cb27-5" aria-hidden="true" tabindex="-1"></a>  <span class="at">C_am =</span> C_am,</span>
<span id="cb27-6"><a href="#cb27-6" aria-hidden="true" tabindex="-1"></a>  <span class="at">V_ni =</span> V_ni,</span>
<span id="cb27-7"><a href="#cb27-7" aria-hidden="true" tabindex="-1"></a>  <span class="at">K_ni =</span> K_ni,</span>
<span id="cb27-8"><a href="#cb27-8" aria-hidden="true" tabindex="-1"></a>  <span class="at">M_ni =</span> M_ni,</span>
<span id="cb27-9"><a href="#cb27-9" aria-hidden="true" tabindex="-1"></a>  <span class="at">C_ni =</span> C_ni,</span>
<span id="cb27-10"><a href="#cb27-10" aria-hidden="true" tabindex="-1"></a>  <span class="at">V_ot =</span> V_ot,</span>
<span id="cb27-11"><a href="#cb27-11" aria-hidden="true" tabindex="-1"></a>  <span class="at">K_ot =</span> K_ot,</span>
<span id="cb27-12"><a href="#cb27-12" aria-hidden="true" tabindex="-1"></a>  <span class="at">M_ot =</span> M_ot,</span>
<span id="cb27-13"><a href="#cb27-13" aria-hidden="true" tabindex="-1"></a>  <span class="at">C_ot =</span> C_ot,</span>
<span id="cb27-14"><a href="#cb27-14" aria-hidden="true" tabindex="-1"></a>  <span class="at">Q_min =</span> Q_min, </span>
<span id="cb27-15"><a href="#cb27-15" aria-hidden="true" tabindex="-1"></a>  <span class="at">Q_max =</span> Q_max, </span>
<span id="cb27-16"><a href="#cb27-16" aria-hidden="true" tabindex="-1"></a>  <span class="at">K_c =</span> K_c,</span>
<span id="cb27-17"><a href="#cb27-17" aria-hidden="true" tabindex="-1"></a>  <span class="at">mu =</span> mu,</span>
<span id="cb27-18"><a href="#cb27-18" aria-hidden="true" tabindex="-1"></a>  <span class="co"># N_min = N_min,</span></span>
<span id="cb27-19"><a href="#cb27-19" aria-hidden="true" tabindex="-1"></a>  <span class="co"># N_max = N_max,</span></span>
<span id="cb27-20"><a href="#cb27-20" aria-hidden="true" tabindex="-1"></a>  <span class="at">D_m =</span> D_m,</span>
<span id="cb27-21"><a href="#cb27-21" aria-hidden="true" tabindex="-1"></a>  <span class="at">D_ve =</span> D_ve, </span>
<span id="cb27-22"><a href="#cb27-22" aria-hidden="true" tabindex="-1"></a>  <span class="at">D_lo =</span> D_lo,</span>
<span id="cb27-23"><a href="#cb27-23" aria-hidden="true" tabindex="-1"></a>  <span class="at">D_mi =</span> D_mi,</span>
<span id="cb27-24"><a href="#cb27-24" aria-hidden="true" tabindex="-1"></a>  <span class="at">D_hi =</span> D_hi,</span>
<span id="cb27-25"><a href="#cb27-25" aria-hidden="true" tabindex="-1"></a>  <span class="at">a_cs =</span> a_cs,                  <span class="co"># or a_cs from figueroa 2006?</span></span>
<span id="cb27-26"><a href="#cb27-26" aria-hidden="true" tabindex="-1"></a>  <span class="at">I_o =</span> I_o,</span>
<span id="cb27-27"><a href="#cb27-27" aria-hidden="true" tabindex="-1"></a>  <span class="at">T_opt =</span> T_opt_arma,</span>
<span id="cb27-28"><a href="#cb27-28" aria-hidden="true" tabindex="-1"></a>  <span class="at">T_min =</span> T_min_arma,</span>
<span id="cb27-29"><a href="#cb27-29" aria-hidden="true" tabindex="-1"></a>  <span class="at">T_max =</span> T_max_arma,</span>
<span id="cb27-30"><a href="#cb27-30" aria-hidden="true" tabindex="-1"></a>  <span class="at">S_opt =</span> S_opt,</span>
<span id="cb27-31"><a href="#cb27-31" aria-hidden="true" tabindex="-1"></a>  <span class="at">S_min =</span> S_min,</span>
<span id="cb27-32"><a href="#cb27-32" aria-hidden="true" tabindex="-1"></a>  <span class="at">S_max =</span> S_max,</span>
<span id="cb27-33"><a href="#cb27-33" aria-hidden="true" tabindex="-1"></a>  <span class="at">h_a =</span> h_a,</span>
<span id="cb27-34"><a href="#cb27-34" aria-hidden="true" tabindex="-1"></a>  <span class="at">h_b =</span> h_b,</span>
<span id="cb27-35"><a href="#cb27-35" aria-hidden="true" tabindex="-1"></a>  <span class="at">h_c =</span> h_c,</span>
<span id="cb27-36"><a href="#cb27-36" aria-hidden="true" tabindex="-1"></a>  <span class="at">h_max =</span> h_max,</span>
<span id="cb27-37"><a href="#cb27-37" aria-hidden="true" tabindex="-1"></a>  <span class="at">DWWW =</span> DWWW </span>
<span id="cb27-38"><a href="#cb27-38" aria-hidden="true" tabindex="-1"></a>  )</span>
<span id="cb27-39"><a href="#cb27-39" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-40"><a href="#cb27-40" aria-hidden="true" tabindex="-1"></a>param_units <span class="ot">&lt;-</span> <span class="fu">c</span>(</span>
<span id="cb27-41"><a href="#cb27-41" aria-hidden="true" tabindex="-1"></a>  <span class="st">"mg gDW-1 d-1"</span>,                                              <span class="co"># V_am</span></span>
<span id="cb27-42"><a href="#cb27-42" aria-hidden="true" tabindex="-1"></a>  <span class="st">"mg m-3"</span>,                                                    <span class="co"># K_am</span></span>
<span id="cb27-43"><a href="#cb27-43" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rep</span>(<span class="st">"m3 gDW-1 d-1"</span>, <span class="dv">2</span>),                                      <span class="co"># M_am, C_am</span></span>
<span id="cb27-44"><a href="#cb27-44" aria-hidden="true" tabindex="-1"></a>  <span class="st">"mg gDW-1 d-1"</span>,                                              <span class="co"># V_ni</span></span>
<span id="cb27-45"><a href="#cb27-45" aria-hidden="true" tabindex="-1"></a>  <span class="st">"mg m-3"</span>,                                                    <span class="co"># K_ni</span></span>
<span id="cb27-46"><a href="#cb27-46" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rep</span>(<span class="st">"m3 gDW-1 d-1"</span>, <span class="dv">2</span>),                                      <span class="co"># M_ni, C_ni</span></span>
<span id="cb27-47"><a href="#cb27-47" aria-hidden="true" tabindex="-1"></a>  <span class="st">"mg gDW-1 d-1"</span>,                                              <span class="co"># V_ot</span></span>
<span id="cb27-48"><a href="#cb27-48" aria-hidden="true" tabindex="-1"></a>  <span class="st">"mg m-3"</span>,                                                    <span class="co"># K_ot</span></span>
<span id="cb27-49"><a href="#cb27-49" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rep</span>(<span class="st">"m3 gDW-1 d-1"</span>, <span class="dv">2</span>),                                      <span class="co"># M_ot, C_ot</span></span>
<span id="cb27-50"><a href="#cb27-50" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rep</span>(<span class="st">"mg gDW-1"</span>, <span class="dv">3</span>),                                          <span class="co"># Q_min, Q_max, K_c</span></span>
<span id="cb27-51"><a href="#cb27-51" aria-hidden="true" tabindex="-1"></a>  <span class="st">"d-1"</span>,                                                       <span class="co"># mu</span></span>
<span id="cb27-52"><a href="#cb27-52" aria-hidden="true" tabindex="-1"></a>  <span class="co"># rep("g gDW-1", 2),                                           # N_min, N_max</span></span>
<span id="cb27-53"><a href="#cb27-53" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rep</span>(<span class="st">"d-1"</span>, <span class="dv">5</span>),                                               <span class="co"># D_m</span></span>
<span id="cb27-54"><a href="#cb27-54" aria-hidden="true" tabindex="-1"></a>  <span class="st">"m mg-1"</span>,                                                    <span class="co"># a_cs</span></span>
<span id="cb27-55"><a href="#cb27-55" aria-hidden="true" tabindex="-1"></a>  <span class="st">"umol photons m-2 s-1"</span>,                                      <span class="co"># I_o</span></span>
<span id="cb27-56"><a href="#cb27-56" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rep</span>(<span class="st">"degrees C"</span>, <span class="dv">3</span>),                                         <span class="co"># T_opt, T_min, T_max</span></span>
<span id="cb27-57"><a href="#cb27-57" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rep</span>(<span class="st">"g L-1"</span>, <span class="dv">3</span>),                                             <span class="co"># S_opt, S_min, S_max</span></span>
<span id="cb27-58"><a href="#cb27-58" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rep</span>(<span class="st">"d-1"</span>, <span class="dv">3</span>),                                               <span class="co"># h_a, h_b, h_c</span></span>
<span id="cb27-59"><a href="#cb27-59" aria-hidden="true" tabindex="-1"></a>  <span class="st">"m"</span>,                                                         <span class="co"># h_max</span></span>
<span id="cb27-60"><a href="#cb27-60" aria-hidden="true" tabindex="-1"></a>  <span class="st">"gWW gDW-1"</span>                                                  <span class="co"># DWWW</span></span>
<span id="cb27-61"><a href="#cb27-61" aria-hidden="true" tabindex="-1"></a>  )</span>
<span id="cb27-62"><a href="#cb27-62" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-63"><a href="#cb27-63" aria-hidden="true" tabindex="-1"></a>a_taxiformis_tetrasporophyte <span class="ot">&lt;-</span> a_armata_tetrasporophyte</span>
<span id="cb27-64"><a href="#cb27-64" aria-hidden="true" tabindex="-1"></a>a_taxiformis_tetrasporophyte[<span class="st">'T_opt'</span>] <span class="ot">&lt;-</span> T_opt_taxi</span>
<span id="cb27-65"><a href="#cb27-65" aria-hidden="true" tabindex="-1"></a>a_taxiformis_tetrasporophyte[<span class="st">'T_max'</span>] <span class="ot">&lt;-</span> T_max_taxi</span>
<span id="cb27-66"><a href="#cb27-66" aria-hidden="true" tabindex="-1"></a>a_taxiformis_tetrasporophyte[<span class="st">'T_min'</span>] <span class="ot">&lt;-</span> T_min_taxi</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<div class="cell">
<details class="code-fold">
<summary>Save named vector and .csv file</summary>
<div class="sourceCode cell-code" id="cb28"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb28-1"><a href="#cb28-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Save data as a named vector - model uses this</span></span>
<span id="cb28-2"><a href="#cb28-2" aria-hidden="true" tabindex="-1"></a><span class="co"># save(a_armata_tetrasporophyte, file = file.path(base_path, "data", "asparagopsis_armata_tetrasporophyte.rda"))</span></span>
<span id="cb28-3"><a href="#cb28-3" aria-hidden="true" tabindex="-1"></a><span class="co"># load(file.path(base_path, "data", "asparagopsis_armata_tetrasporophyte.rda")) # use this to load data from file</span></span>
<span id="cb28-4"><a href="#cb28-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb28-5"><a href="#cb28-5" aria-hidden="true" tabindex="-1"></a><span class="co"># Save duplicate as CSV - easier for humans to read</span></span>
<span id="cb28-6"><a href="#cb28-6" aria-hidden="true" tabindex="-1"></a>df <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="at">parameter =</span> <span class="fu">names</span>(a_armata_tetrasporophyte),</span>
<span id="cb28-7"><a href="#cb28-7" aria-hidden="true" tabindex="-1"></a>                              <span class="at">unit =</span> <span class="fu">as.character</span>(param_units),</span>
<span id="cb28-8"><a href="#cb28-8" aria-hidden="true" tabindex="-1"></a>                              <span class="at">value =</span> (<span class="fu">unname</span>(a_armata_tetrasporophyte)))</span>
<span id="cb28-9"><a href="#cb28-9" aria-hidden="true" tabindex="-1"></a><span class="fu">write.csv</span>(df, <span class="fu">file.path</span>(base_path, <span class="st">"data"</span>, <span class="st">"asparagopsis_armata_tetrasporophyte.csv"</span>))</span>
<span id="cb28-10"><a href="#cb28-10" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb28-11"><a href="#cb28-11" aria-hidden="true" tabindex="-1"></a><span class="co"># save(a_taxiformis_tetrasporophyte, file = file.path(base_path, "data", "asparagopsis_taxiformis_tetrasporophyte.rda"))</span></span>
<span id="cb28-12"><a href="#cb28-12" aria-hidden="true" tabindex="-1"></a>df <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="at">parameter =</span> <span class="fu">names</span>(a_taxiformis_tetrasporophyte),</span>
<span id="cb28-13"><a href="#cb28-13" aria-hidden="true" tabindex="-1"></a>                              <span class="at">unit =</span> <span class="fu">as.character</span>(param_units),</span>
<span id="cb28-14"><a href="#cb28-14" aria-hidden="true" tabindex="-1"></a>                              <span class="at">value =</span> (<span class="fu">unname</span>(a_taxiformis_tetrasporophyte)))</span>
<span id="cb28-15"><a href="#cb28-15" aria-hidden="true" tabindex="-1"></a><span class="fu">write.csv</span>(df, <span class="fu">file.path</span>(base_path, <span class="st">"data"</span>, <span class="st">"asparagopsis_taxiformis_tetrasporophyte.csv"</span>))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
</section>
</section>
<section id="gametophytes" class="level1 page-columns page-full">
<h1>Gametophytes</h1>
<section id="sec-nitrogen-uptake-gametophyte" class="level2 page-columns page-full">
<h2 class="anchored" data-anchor-id="sec-nitrogen-uptake-gametophyte">Nitrogen uptake</h2>
<p>It’s possible that the gametophyte and tetrasporophyte will have different uptake rates of ammonium and nitrate, and in the wild they do not appear to be affected by fluctuations in nutrients <span class="citation" data-cites="zanolla_concise_2022 zanolla_structure_2018">Zanolla, Altamirano, Carmona, et al. (<a href="#ref-zanolla_assessing_2018" role="doc-biblioref">2018</a>)</span>. However, if we assume that the gametophyte has a similar preference for ammonium to the sporophyte then it’s not surprising that it would show little response to nutrients in the environment, as natural nitrogen is mostly in nitrate form with ammonium only being present at constant low levels generally.</p>

<div class="no-row-height column-margin column-container"><div class="callout callout-style-default callout-important callout-titled">
<div class="callout-header d-flex align-content-center">
<div class="callout-icon-container">
<i class="callout-icon"></i>
</div>
<div class="callout-title-container flex-fill">
Important
</div>
</div>
<div class="callout-body-container callout-body">
<p><span class="citation" data-cites="zanolla_assessing_2018">Zanolla, Altamirano, Carmona, et al. (<a href="#ref-zanolla_assessing_2018" role="doc-biblioref">2018</a>)</span> didn’t directly measure nutrients(?) but found that temperature was highly correlated with species distribution. Check <span class="citation" data-cites="zanolla_structure_2018">Zanolla, Carmona, et al. (<a href="#ref-zanolla_structure_2018" role="doc-biblioref">2018</a>)</span> - did they measure nutrients?</p>
</div>
</div></div><p>As far as I can tell, no-one has investigated nitrogen uptake directly in <em>Asparagopsis</em> gametophytes. I will therefore be using the uptake rates calculated for the tetrasporophyte.</p>
</section>
<section id="sec-internal-nitrogen-gametophyte" class="level2">
<h2 class="anchored" data-anchor-id="sec-internal-nitrogen-gametophyte">Internal nutrient state</h2>
<p>Maximum (or, at least, very high) internal nitrogen concentrations for <em>Asparagopsis</em> are fairly easy to find. <em>A. taxiformis</em> had a protein content of 22.69 %DM in <span class="citation" data-cites="nunes_evaluation_2024">Nunes et al. (<a href="#ref-nunes_evaluation_2024" role="doc-biblioref">2024</a>)</span> and 23.76 %DM in <span class="citation" data-cites="pacheco_invasive_2020">Pacheco et al. (<a href="#ref-pacheco_invasive_2020" role="doc-biblioref">2020</a>)</span>. <span class="citation" data-cites="nunes_evaluation_2024">Nunes et al. (<a href="#ref-nunes_evaluation_2024" role="doc-biblioref">2024</a>)</span> also found that <em>A. armata</em> had a protein concentration of 24.23 %DM. However, the <em>A. taxiformis</em> spp. gametophytes eaten in Hawaii have also been reported having protein concentrations &lt;10 %DW <span class="citation" data-cites="mcdermid_nutritional_2003">(<a href="#ref-mcdermid_nutritional_2003" role="doc-biblioref">McDermid and Stuercke 2003</a>)</span>, and all of these measurements came from the invasive range of <em>Asparagopsis</em>.</p>
<p>Samples of <em>A. armata</em> collected from New Zealand showed a protein concentrations ranging from of 10.94 to 15.2 %DW <span class="citation" data-cites="mihaila_new_2022 zemke-white_chlorophyte_1999">(<a href="#ref-mihaila_new_2022" role="doc-biblioref">Mihaila et al. 2022</a>; <a href="#ref-zemke-white_chlorophyte_1999" role="doc-biblioref">Zemke-White and Clements 1999</a>)</span>, while Australian samples of <em>A. taxiformis</em> had mean a protein concentration of 18.2 %DW <span class="citation" data-cites="brooke_methane_2020">(<a href="#ref-brooke_methane_2020" role="doc-biblioref">Brooke et al. 2020</a>)</span>.</p>
<p>There are two observations made from this. First, <em>A. armata</em> appears to have a slightly higher average protein concentration than <em>A. taxiformis</em>, although this might be due to small sample sizes and a lack of direct comparison. Second, the samples collected from Australia and New Zealand had a much lower protein concentration than those collected elsewhere <span class="citation" data-cites="nunes_evaluation_2024 mihaila_new_2022">(<a href="#ref-nunes_evaluation_2024" role="doc-biblioref">Nunes et al. 2024</a>; <a href="#ref-mihaila_new_2022" role="doc-biblioref">Mihaila et al. 2022</a>)</span>, possibly reflecting the region’s naturally low nitrogen levels. Luckily, these results can still be useful as there appear to be no significant differences between the protein content of cultured and wild macroalgae, so intra-species variation is probably the largest factor determining the protein content of most macroalgae <span class="citation" data-cites="angell_protein_2016">(<a href="#ref-angell_protein_2016" role="doc-biblioref">Angell et al. 2016</a>)</span>.</p>
<div class="cell">
<details class="code-fold">
<summary>Range of protein concentrations</summary>
<div class="sourceCode cell-code" id="cb29"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb29-1"><a href="#cb29-1" aria-hidden="true" tabindex="-1"></a>P_range_taxi <span class="ot">&lt;-</span> <span class="fu">sort</span>(<span class="fu">c</span>(<span class="fl">22.69</span>, <span class="fl">23.76</span>, <span class="fl">18.2</span>, <span class="fl">7.5</span>))<span class="sc">/</span><span class="dv">100</span> <span class="co"># Asparagopsis taxiformis</span></span>
<span id="cb29-2"><a href="#cb29-2" aria-hidden="true" tabindex="-1"></a>P_range_arma <span class="ot">&lt;-</span> <span class="fu">sort</span>(<span class="fu">c</span>(<span class="fl">24.23</span>, <span class="fl">15.2</span>, <span class="fl">10.94</span>))<span class="sc">/</span><span class="dv">100</span> <span class="co"># Asparagopsis armata</span></span>
<span id="cb29-3"><a href="#cb29-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb29-4"><a href="#cb29-4" aria-hidden="true" tabindex="-1"></a>N_conv_aspara <span class="ot">&lt;-</span> <span class="fl">5.63</span></span>
<span id="cb29-5"><a href="#cb29-5" aria-hidden="true" tabindex="-1"></a>N_conv_reds <span class="ot">&lt;-</span> <span class="fl">5.10</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<p>There are also different possible ways of converting protein to nitrogen: <span class="citation" data-cites="angell_protein_2016">Angell et al. (<a href="#ref-angell_protein_2016" role="doc-biblioref">2016</a>)</span> suggested that a factor 5.10 is most appropriate for red algae, although they also reported <em>Asparagopsis</em> specifically having a conversion factor of 5.63 which was used by <span class="citation" data-cites="mihaila_new_2022">Mihaila et al. (<a href="#ref-mihaila_new_2022" role="doc-biblioref">2022</a>)</span> in their calculations. These two conversion factors produce nitrogen concentrations ranging from 2.15 to 4.75 and 1.94 to 4.3 % DW for <em>A. armata</em> respectively.</p>
<div class="cell">
<details class="code-fold">
<summary>Choosing <span class="math inline">\(Q_{min}\)</span> and <span class="math inline">\(Q_{max}\)</span> values</summary>
<div class="sourceCode cell-code" id="cb30"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb30-1"><a href="#cb30-1" aria-hidden="true" tabindex="-1"></a>N_min <span class="ot">&lt;-</span> <span class="fu">set_units</span>(<span class="fu">min</span>(P_range_arma)<span class="sc">/</span>N_conv_aspara, <span class="st">"g gDW-1"</span>)</span>
<span id="cb30-2"><a href="#cb30-2" aria-hidden="true" tabindex="-1"></a>Q_min <span class="ot">&lt;-</span> <span class="fu">drop_units</span>(<span class="fu">set_units</span>(N_min, <span class="st">"mg gDW-1"</span>))</span>
<span id="cb30-3"><a href="#cb30-3" aria-hidden="true" tabindex="-1"></a>N_max <span class="ot">&lt;-</span> <span class="fu">set_units</span>(<span class="fu">max</span>(P_range_arma)<span class="sc">/</span>N_conv_aspara, <span class="st">"g gDW-1"</span>)</span>
<span id="cb30-4"><a href="#cb30-4" aria-hidden="true" tabindex="-1"></a>Q_max <span class="ot">&lt;-</span> <span class="fu">drop_units</span>(<span class="fu">set_units</span>(N_max, <span class="st">"mg gDW-1"</span>))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<p>This gives a <span class="math inline">\(Q_{min}\)</span> of 19.43 mg gDW<span class="math inline">\(^{-1}\)</span> and a <span class="math inline">\(Q_{max}\)</span> of 43.04 mg gDW<span class="math inline">\(^{-1}\)</span>.</p>
</section>
<section id="sec-salinity-gametophyte" class="level2">
<h2 class="anchored" data-anchor-id="sec-salinity-gametophyte">Salinity</h2>
<p>There’s really not enough information to do this seperately, so we’ll have to go with the tetrasporophyte estimate.</p>
</section>
<section id="biomass-factors-1" class="level2">
<h2 class="anchored" data-anchor-id="biomass-factors-1">Biomass factors</h2>
<section id="sec-dry-weight-gametophyte" class="level3">
<h3 class="anchored" data-anchor-id="sec-dry-weight-gametophyte">Dry weight to fresh weight</h3>
<p><span class="citation" data-cites="nunes_evaluation_2024">Nunes et al. (<a href="#ref-nunes_evaluation_2024" role="doc-biblioref">2024</a>)</span> found a dry mass of 6.55 <span class="math inline">\(\pm\)</span> 1.25% for <em>A. taxiformis</em> and 7.68 <span class="math inline">\(\pm\)</span> 1.07% for <em>A. armata</em>, and reported that similar values were found by <span class="citation" data-cites="roque_effect_2019">Roque et al. (<a href="#ref-roque_effect_2019" role="doc-biblioref">2019</a>)</span>.</p>
<div class="cell">
<details class="code-fold">
<summary>Set DWWW units based on <span class="citation" data-cites="nunes_evaluation_2024">Nunes et al. (<a href="#ref-nunes_evaluation_2024" role="doc-biblioref">2024</a>)</span></summary>
<div class="sourceCode cell-code" id="cb31"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb31-1"><a href="#cb31-1" aria-hidden="true" tabindex="-1"></a><span class="fu">remove_unit</span>(<span class="st">"gWW"</span>)</span>
<span id="cb31-2"><a href="#cb31-2" aria-hidden="true" tabindex="-1"></a><span class="fu">install_unit</span>(<span class="st">"gWW"</span>, <span class="st">"1/7.68 gDW"</span>) <span class="co"># A. armata</span></span>
<span id="cb31-3"><a href="#cb31-3" aria-hidden="true" tabindex="-1"></a>DWWW <span class="ot">&lt;-</span> <span class="fu">set_units</span>(<span class="dv">1</span>, <span class="st">"gDW"</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb31-4"><a href="#cb31-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">set_units</span>(<span class="st">"gWW"</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb31-5"><a href="#cb31-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">drop_units</span>()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
</section>
<section id="sec-height-gametophyte" class="level3">
<h3 class="anchored" data-anchor-id="sec-height-gametophyte">Height</h3>
<p>Within the model, “height” does not indicate height exactly, rather it indicates how much of the available space is being taken up by the macroalgae. This combines with density (mg m<span class="math inline">\(^{-3}\)</span>, i.e.&nbsp;growth) to indicate how much light and nutrients the macroalgae has access to. For example:</p>
<ul>
<li>a macroalgae with a high growth rate but a low height-increase rate (e.g.&nbsp;an <em>Asparagopsis</em> tetrasporophyte) will be more dense in a smaller volume, potentially having higher self-shading rates and decreased access to nutrients</li>
<li>a macroalgae with a low growth rate and a high height-increase rate (e.g.&nbsp;<em>Ecklonia</em>) will be less dense in a larger volume, potentially lower self-shading rates and increased access to nutrients</li>
</ul>
<p><span class="citation" data-cites="zanolla_size_2018">Zanolla, Altamirano, De La Rosa, et al. (<a href="#ref-zanolla_size_2018" role="doc-biblioref">2018</a>)</span> reported that <em>A, taxiformis</em> gametophyte branches had a maximum length of ~26 cm at ~115 g DW m<span class="math inline">\(^{-2}\)</span>. This is equivalent to approximately 29.9 g DW m<span class="math inline">\(^{-3}\)</span>. Assuming a nitrogen concentration of 6%, this translates to 1794 g N mg<span class="math inline">\(^{-3}\)</span>. <span class="citation" data-cites="zanolla_size_2018">Zanolla, Altamirano, De La Rosa, et al. (<a href="#ref-zanolla_size_2018" role="doc-biblioref">2018</a>)</span> found that smaller size classes were more abundant than higher classes, indicating that the gametophyte prioritises new shoots (lateral growth) over increasing height (vertical growth). Within the model this is captured by a very slow overall increase in height.</p>
<div class="cell">
<details class="code-fold">
<summary>Create a realistic height curve</summary>
<div class="sourceCode cell-code" id="cb32"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb32-1"><a href="#cb32-1" aria-hidden="true" tabindex="-1"></a>h_a <span class="ot">&lt;-</span> <span class="dv">3850</span></span>
<span id="cb32-2"><a href="#cb32-2" aria-hidden="true" tabindex="-1"></a>h_b <span class="ot">&lt;-</span> <span class="fl">1.75</span></span>
<span id="cb32-3"><a href="#cb32-3" aria-hidden="true" tabindex="-1"></a>h_c <span class="ot">&lt;-</span> <span class="fl">0.001</span></span>
<span id="cb32-4"><a href="#cb32-4" aria-hidden="true" tabindex="-1"></a>h_max <span class="ot">&lt;-</span> <span class="fl">0.26</span></span>
<span id="cb32-5"><a href="#cb32-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb32-6"><a href="#cb32-6" aria-hidden="true" tabindex="-1"></a>dens <span class="ot">&lt;-</span> <span class="fu">seq</span>(<span class="dv">10</span>, <span class="fu">drop_units</span>(sizedisN)<span class="sc">+</span><span class="dv">500</span>, <span class="dv">10</span>)</span>
<span id="cb32-7"><a href="#cb32-7" aria-hidden="true" tabindex="-1"></a>h <span class="ot">&lt;-</span> <span class="fu">sapply</span>(<span class="at">X =</span> dens, <span class="at">FUN =</span> algae_height, <span class="at">spec_params =</span> <span class="fu">c</span>(<span class="at">h_a =</span> h_a, <span class="at">h_b =</span> h_b, <span class="at">h_c =</span> h_c, <span class="at">h_max =</span> h_max))</span>
<span id="cb32-8"><a href="#cb32-8" aria-hidden="true" tabindex="-1"></a>height <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="at">dens =</span> <span class="fu">set_units</span>(dens, <span class="st">"mg m-3"</span>), <span class="at">h =</span> <span class="fu">set_units</span>(h, <span class="st">"m"</span>))</span>
<span id="cb32-9"><a href="#cb32-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb32-10"><a href="#cb32-10" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>(height) <span class="sc">+</span></span>
<span id="cb32-11"><a href="#cb32-11" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_line</span>(<span class="fu">aes</span>(<span class="at">x =</span> dens, <span class="at">y =</span> h)) <span class="sc">+</span></span>
<span id="cb32-12"><a href="#cb32-12" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_vline</span>(<span class="at">xintercept =</span> sizedisN, <span class="at">linetype =</span> <span class="st">"dashed"</span>) <span class="sc">+</span></span>
<span id="cb32-13"><a href="#cb32-13" aria-hidden="true" tabindex="-1"></a>  prettyplot <span class="sc">+</span></span>
<span id="cb32-14"><a href="#cb32-14" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">x =</span> <span class="st">"Nf"</span>, <span class="at">y =</span> <span class="st">"Height"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><img src="asparagopsis-parameterisation_files/figure-html/height-1.png" class="img-fluid figure-img" width="672"></p>
<figcaption>Algae height with Nf using spec_params = c(h_a = 3850, h_b = 1.75, h_c = 0.001, h_max = 0.26). The dashed line shows the point at which max height is reached (1794 g N m<span class="math inline">\(^{-3}\)</span>)</figcaption>
</figure>
</div>
</div>
</div>
</section>
<section id="sec-growth-rate-gametophyte" class="level3">
<h3 class="anchored" data-anchor-id="sec-growth-rate-gametophyte">Growth rate, stocking density</h3>
<p><span class="citation" data-cites="zanolla_size_2018">Zanolla, Altamirano, De La Rosa, et al. (<a href="#ref-zanolla_size_2018" role="doc-biblioref">2018</a>)</span>? <span class="citation" data-cites="wright_asexual_2022">Wright et al. (<a href="#ref-wright_asexual_2022" role="doc-biblioref">2022</a>)</span>?</p>
</section>
</section>
<section id="sec-temperature-gametophyte" class="level2">
<h2 class="anchored" data-anchor-id="sec-temperature-gametophyte">Temperature response</h2>
<p>By default, the gametophyte temperature parameters for <em>Asparagopsis</em> in the model will be identical to the tetrasporophyte. This is because much less research has been done on the gametophyte phases. However, the gametophyte and tetrasporophyte phase might have similar temperature optima. In <span class="citation" data-cites="chualain_invasive_2004">Chualáin et al. (<a href="#ref-chualain_invasive_2004" role="doc-biblioref">2004</a>)</span> <em>A.taxiformis</em> tetrasporophytes had an optimum growth temperature of 25<span class="math inline">\(^{\circ}\)</span>, and this same growth optimum was observed in <em>A.taxiformis</em> gametophytes by <span class="citation" data-cites="padilla-gamino_seasonal_2007">Padilla-Gamiño and Carpenter (<a href="#ref-padilla-gamino_seasonal_2007" role="doc-biblioref">2007</a>)</span>.</p>
<p>However, <span class="citation" data-cites="mihaila_moderate_2024">Mihaila et al. (<a href="#ref-mihaila_moderate_2024" role="doc-biblioref">2024</a>)</span> tested the growth of juvenile (microscopic) gametophytes at 15, 18 and 21<span class="math inline">\(^{\circ}\)</span> and found that growth was highest at 18<span class="math inline">\(^{\circ}\)</span>, somewhat reduced at 15<span class="math inline">\(^{\circ}\)</span> and substantially reduced at 21<span class="math inline">\(^{\circ}\)</span>. This temperature optimum is lower than the NA<span class="math inline">\(^{\circ}\)</span> optimum found for the tetrasporophyte in <a href="#sec-temperature-tetrasporophyte" class="quarto-xref">Section&nbsp;2.5</a>, and the maximum is much lower. This might indicate that the gametophyte stage has a narrower temperature tolerance than the tetrasporophyte stage.</p>
<div id="fig-seasonality" class="quarto-float quarto-figure quarto-figure-center anchored">
<figure class="quarto-float quarto-float-fig figure">
<div aria-describedby="fig-seasonality-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<img src="images/seasonality.jpg" class="img-fluid figure-img" style="width:80.0%">
</div>
<figcaption class="quarto-float-caption-bottom quarto-float-caption quarto-float-fig" id="fig-seasonality-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
Figure&nbsp;2: Timing of sea-cultured gametophytes and their appearance in the wild. Dark red indicates when the gametophyte is present on the reef, and lighter red indicates when gametophytes are just starting to appear or disappear, so less abundant. Tetrasporophytes have thus far been very difficult to find in the wild (Margie Rule, pers. comm.).
</figcaption>
</figure>
</div>
</section>
<section id="light-response" class="level2">
<h2 class="anchored" data-anchor-id="light-response">Light response</h2>
<p>There isn’t really enough data to calculate a good <span class="math inline">\(a_{cs}\)</span> value for the <em>Asparagopsis</em> gametophyte, so I’ll be using the tetrasporophyte <span class="math inline">\(a_{cs}\)</span> value calculated in <a href="#sec-light-tetrasporophyte" class="quarto-xref">Section&nbsp;2.6</a>.</p>
</section>
<section id="fragmentation-with-exposure-1" class="level2 page-columns page-full">
<h2 class="anchored" data-anchor-id="fragmentation-with-exposure-1">Fragmentation with exposure</h2>
<p><span class="citation" data-cites="hall_investigating_2023">Hall (<a href="#ref-hall_investigating_2023" role="doc-biblioref">2023</a>)</span> counducted three (seemingly identical?) velocity treatments on gametophytes and measured % biomass lost. Experiments 1 and 2 showed no differences between treatments, but experiment 3 did. I’m going to use experiment 3’s results to get a linear rate with velocity, and set base and maximum rates based on observations from all three experiments.</p>
<ul>
<li>It has been suggested that “this species prefer semi-exposed areas as biomass production decreased with wave exposure” <span class="citation" data-cites="sainz-villegas_modelling_2024">(<a href="#ref-sainz-villegas_modelling_2024" role="doc-biblioref">Sainz-Villegas 2024</a> pg 66)</span> but this was not tested in that study.</li>
<li>May be worth including a dynamic “breakage” term, perhaps modifying <span class="math inline">\(D_m\)</span> such that it has a minimum value and increases with <span class="math inline">\(U_0\)</span> (or <span class="math inline">\(U_0^2\)</span>?).</li>
</ul>
<p>In Tasmania, <em>A. armata</em> are harvested after only 6 weeks because the shedding rate means they don’t get any additional biomass. There are two ways to implement this:</p>
<ol type="1">
<li>Manually, i.e.&nbsp;the model is only ever run for 6 weeks at a time with <em>Asparagopsis</em> (Camille White, pers comm).</li>
</ol>
<ul>
<li>Disadvantage: no mechanistic recreation of shedding, that 6-week time frame will be fixed</li>
<li>Advantage: fast and easy</li>
</ul>
<ol start="2" type="1">
<li>Mechanistically, i.e.&nbsp;implementing a value of <span class="math inline">\(D_m\)</span> that increases over time.</li>
</ol>
<ul>
<li>Disadvantage: additional uncertainty that comes with needing to create/parameterise a new function</li>
<li>Advantages: mechanistic recreation of realistic conditions and culture practices, and a possible synergies with an exposure-based <span class="math inline">\(D_m\)</span> parameters</li>
</ul>
<p>A loss term might end up looking like: <span class="math display">\[
D_m = f(growdays) + f(h_m/h_{max}) + f(V)
\]</span></p>
<p>Where <span class="math inline">\(D_m\)</span> is the % of biomass lost d<span class="math inline">\(^{-1}\)</span>.</p>

<div class="no-row-height column-margin column-container"><div class="callout callout-style-default callout-important callout-titled">
<div class="callout-header d-flex align-content-center">
<div class="callout-icon-container">
<i class="callout-icon"></i>
</div>
<div class="callout-title-container flex-fill">
Important
</div>
</div>
<div class="callout-body-container callout-body">
<p>Is fragmentation maybe a function of age (maturity) or length of thalli, in addition to exposure?</p>
</div>
</div></div><section id="velocity" class="level3">
<h3 class="anchored" data-anchor-id="velocity">Velocity</h3>
<p><span class="citation" data-cites="hall_investigating_2023">Hall (<a href="#ref-hall_investigating_2023" role="doc-biblioref">2023</a>)</span> investigated tetrasporophyte and gametophyte fragmentation at different levels of turbulence and water velocities. Focus of the study was on how fragmentation can be used in nurseries to enhance stock production. The laminar water velocities tested were:</p>
<ul>
<li>Low ( 0.25 m s<span class="math inline">\(^{-1}\)</span>)</li>
<li>Low-mid ( 0.37 m s<span class="math inline">\(^{-1}\)</span>)</li>
<li>Mid ( 0.52 m s<span class="math inline">\(^{-1}\)</span>)</li>
<li>Mid-high ( 0.81 m s<span class="math inline">\(^{-1}\)</span>)</li>
<li>High ( 1.14 m s<span class="math inline">\(^{-1}\)</span>)</li>
<li>Max-high ( 1.42 m s<span class="math inline">\(^{-1}\)</span>)</li>
</ul>
<p>These results are summarised in Figures 3.17, 3.18 and 3.19.</p>
<div class="cell">
<details class="code-fold">
<summary>Get <span class="citation" data-cites="hall_investigating_2023">Hall (<a href="#ref-hall_investigating_2023" role="doc-biblioref">2023</a>)</span> velocity data for gametophytes</summary>
<div class="sourceCode cell-code" id="cb33"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb33-1"><a href="#cb33-1" aria-hidden="true" tabindex="-1"></a>hall_3<span class="fl">.17</span> <span class="ot">&lt;-</span> <span class="fu">bind_rows</span>(<span class="fu">list</span>(ls[[<span class="dv">3</span>]], ls[[<span class="dv">4</span>]])) <span class="sc">%&gt;%</span> </span>
<span id="cb33-2"><a href="#cb33-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">Label =</span> <span class="fu">rep</span>(<span class="fu">c</span>(<span class="st">'low'</span>, <span class="st">'low.mid'</span>, <span class="st">'mid'</span>, <span class="st">'mid.high'</span>, <span class="st">'high'</span>), <span class="dv">2</span>),</span>
<span id="cb33-3"><a href="#cb33-3" aria-hidden="true" tabindex="-1"></a>         <span class="at">Exp =</span> <span class="st">"Velocity 1"</span>,</span>
<span id="cb33-4"><a href="#cb33-4" aria-hidden="true" tabindex="-1"></a>         <span class="at">Form =</span> <span class="st">"Gametophyte"</span>,</span>
<span id="cb33-5"><a href="#cb33-5" aria-hidden="true" tabindex="-1"></a>         <span class="at">Data =</span> <span class="fu">c</span>(<span class="fu">rep</span>(<span class="st">"Mean"</span>, <span class="dv">5</span>), <span class="fu">rep</span>(<span class="st">"Mean+SE"</span>, <span class="dv">5</span>))) <span class="sc">%&gt;%</span> </span>
<span id="cb33-6"><a href="#cb33-6" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">select</span>(<span class="sc">-</span>dataset) <span class="sc">%&gt;%</span> </span>
<span id="cb33-7"><a href="#cb33-7" aria-hidden="true" tabindex="-1"></a>  <span class="fu">pivot_wider</span>(<span class="at">names_from =</span> Data, <span class="at">values_from =</span> perc_biomass_lost) <span class="sc">%&gt;%</span> </span>
<span id="cb33-8"><a href="#cb33-8" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">SE =</span> <span class="st">`</span><span class="at">Mean+SE</span><span class="st">`</span> <span class="sc">-</span> Mean) <span class="sc">%&gt;%</span> </span>
<span id="cb33-9"><a href="#cb33-9" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">select</span>(<span class="sc">-</span><span class="st">`</span><span class="at">Mean+SE</span><span class="st">`</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb33-10"><a href="#cb33-10" aria-hidden="true" tabindex="-1"></a>  <span class="fu">pivot_longer</span>(<span class="at">names_to =</span> <span class="st">"Data"</span>, <span class="at">values_to =</span> <span class="st">"perc_biomass_lost"</span>, <span class="at">cols =</span> <span class="fu">c</span>(Mean, SE)) <span class="sc">%&gt;%</span> </span>
<span id="cb33-11"><a href="#cb33-11" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">Label =</span> <span class="fu">factor</span>(Label, <span class="at">levels =</span> velocity_levels, <span class="at">labels =</span> Vlevels))</span>
<span id="cb33-12"><a href="#cb33-12" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb33-13"><a href="#cb33-13" aria-hidden="true" tabindex="-1"></a>hall_3<span class="fl">.18</span> <span class="ot">&lt;-</span> <span class="fu">bind_rows</span>(<span class="fu">list</span>(ls[[<span class="dv">5</span>]], ls[[<span class="dv">6</span>]])) <span class="sc">%&gt;%</span> </span>
<span id="cb33-14"><a href="#cb33-14" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">Label =</span> <span class="fu">rep</span>(<span class="fu">c</span>(<span class="st">'low'</span>, <span class="st">'low.mid'</span>, <span class="st">'mid'</span>, <span class="st">'mid.high'</span>, <span class="st">'high'</span>, <span class="st">'max.high'</span>), <span class="dv">2</span>),</span>
<span id="cb33-15"><a href="#cb33-15" aria-hidden="true" tabindex="-1"></a>         <span class="at">Exp =</span> <span class="st">"Velocity 2"</span>,</span>
<span id="cb33-16"><a href="#cb33-16" aria-hidden="true" tabindex="-1"></a>         <span class="at">Form =</span> <span class="st">"Gametophyte"</span>,</span>
<span id="cb33-17"><a href="#cb33-17" aria-hidden="true" tabindex="-1"></a>         <span class="at">Data =</span> <span class="fu">c</span>(<span class="fu">rep</span>(<span class="st">"Mean"</span>, <span class="dv">6</span>), <span class="fu">rep</span>(<span class="st">"Mean+SE"</span>, <span class="dv">6</span>))) <span class="sc">%&gt;%</span> </span>
<span id="cb33-18"><a href="#cb33-18" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">select</span>(<span class="sc">-</span>dataset) <span class="sc">%&gt;%</span> </span>
<span id="cb33-19"><a href="#cb33-19" aria-hidden="true" tabindex="-1"></a>  <span class="fu">pivot_wider</span>(<span class="at">names_from =</span> Data, <span class="at">values_from =</span> perc_biomass_lost) <span class="sc">%&gt;%</span> </span>
<span id="cb33-20"><a href="#cb33-20" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">SE =</span> <span class="st">`</span><span class="at">Mean+SE</span><span class="st">`</span> <span class="sc">-</span> Mean) <span class="sc">%&gt;%</span> </span>
<span id="cb33-21"><a href="#cb33-21" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">select</span>(<span class="sc">-</span><span class="st">`</span><span class="at">Mean+SE</span><span class="st">`</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb33-22"><a href="#cb33-22" aria-hidden="true" tabindex="-1"></a>  <span class="fu">pivot_longer</span>(<span class="at">names_to =</span> <span class="st">"Data"</span>, <span class="at">values_to =</span> <span class="st">"perc_biomass_lost"</span>, <span class="at">cols =</span> <span class="fu">c</span>(Mean, SE)) <span class="sc">%&gt;%</span> </span>
<span id="cb33-23"><a href="#cb33-23" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">Label =</span> <span class="fu">factor</span>(Label, <span class="at">levels =</span> velocity_levels, <span class="at">labels =</span> Vlevels))</span>
<span id="cb33-24"><a href="#cb33-24" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb33-25"><a href="#cb33-25" aria-hidden="true" tabindex="-1"></a>hall_3<span class="fl">.19</span> <span class="ot">&lt;-</span> <span class="fu">bind_rows</span>(<span class="fu">list</span>(ls[[<span class="dv">7</span>]], ls[[<span class="dv">8</span>]])) <span class="sc">%&gt;%</span> </span>
<span id="cb33-26"><a href="#cb33-26" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">Label =</span> <span class="fu">rep</span>(<span class="fu">c</span>(<span class="st">'low'</span>, <span class="st">'low.mid'</span>, <span class="st">'mid'</span>, <span class="st">'mid.high'</span>, <span class="st">'high'</span>, <span class="st">'max.high'</span>), <span class="dv">2</span>),</span>
<span id="cb33-27"><a href="#cb33-27" aria-hidden="true" tabindex="-1"></a>         <span class="at">Exp =</span> <span class="st">"Velocity 3"</span>,</span>
<span id="cb33-28"><a href="#cb33-28" aria-hidden="true" tabindex="-1"></a>         <span class="at">Form =</span> <span class="st">"Gametophyte"</span>,</span>
<span id="cb33-29"><a href="#cb33-29" aria-hidden="true" tabindex="-1"></a>         <span class="at">Data =</span> <span class="fu">c</span>(<span class="fu">rep</span>(<span class="st">"Mean"</span>, <span class="dv">6</span>), <span class="fu">rep</span>(<span class="st">"Mean+SE"</span>, <span class="dv">6</span>))) <span class="sc">%&gt;%</span> </span>
<span id="cb33-30"><a href="#cb33-30" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">select</span>(<span class="sc">-</span>dataset) <span class="sc">%&gt;%</span> </span>
<span id="cb33-31"><a href="#cb33-31" aria-hidden="true" tabindex="-1"></a>  <span class="fu">pivot_wider</span>(<span class="at">names_from =</span> Data, <span class="at">values_from =</span> perc_biomass_lost) <span class="sc">%&gt;%</span> </span>
<span id="cb33-32"><a href="#cb33-32" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">SE =</span> <span class="st">`</span><span class="at">Mean+SE</span><span class="st">`</span> <span class="sc">-</span> Mean) <span class="sc">%&gt;%</span> </span>
<span id="cb33-33"><a href="#cb33-33" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">select</span>(<span class="sc">-</span><span class="st">`</span><span class="at">Mean+SE</span><span class="st">`</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb33-34"><a href="#cb33-34" aria-hidden="true" tabindex="-1"></a>  <span class="fu">pivot_longer</span>(<span class="at">names_to =</span> <span class="st">"Data"</span>, <span class="at">values_to =</span> <span class="st">"perc_biomass_lost"</span>, <span class="at">cols =</span> <span class="fu">c</span>(Mean, SE)) <span class="sc">%&gt;%</span> </span>
<span id="cb33-35"><a href="#cb33-35" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">Label =</span> <span class="fu">factor</span>(Label, <span class="at">levels =</span> velocity_levels, <span class="at">labels =</span> Vlevels))</span>
<span id="cb33-36"><a href="#cb33-36" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb33-37"><a href="#cb33-37" aria-hidden="true" tabindex="-1"></a>hall_gam_vel <span class="ot">&lt;-</span> <span class="fu">bind_rows</span>(<span class="fu">list</span>(hall_3<span class="fl">.17</span>, hall_3<span class="fl">.18</span>, hall_3<span class="fl">.19</span>)) <span class="sc">%&gt;%</span> </span>
<span id="cb33-38"><a href="#cb33-38" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">Exp =</span> <span class="fu">as.factor</span>(Exp),</span>
<span id="cb33-39"><a href="#cb33-39" aria-hidden="true" tabindex="-1"></a>         <span class="at">Data =</span> <span class="fu">as.factor</span>(Data))</span>
<span id="cb33-40"><a href="#cb33-40" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb33-41"><a href="#cb33-41" aria-hidden="true" tabindex="-1"></a>p1 <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(<span class="fu">filter</span>(hall_gam_vel, Data <span class="sc">==</span> <span class="st">"Mean"</span>), </span>
<span id="cb33-42"><a href="#cb33-42" aria-hidden="true" tabindex="-1"></a>       <span class="fu">aes</span>(<span class="at">x =</span> <span class="fu">as.numeric</span>(Label), <span class="at">y =</span> perc_biomass_lost<span class="sc">*</span><span class="dv">100</span>, <span class="at">colour =</span> Exp)) <span class="sc">+</span></span>
<span id="cb33-43"><a href="#cb33-43" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_point</span>() <span class="sc">+</span></span>
<span id="cb33-44"><a href="#cb33-44" aria-hidden="true" tabindex="-1"></a>  prettyplot <span class="sc">+</span></span>
<span id="cb33-45"><a href="#cb33-45" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">x =</span> <span class="st">"Velocity [m/s]"</span>, <span class="at">y =</span> <span class="st">"Mean % biomass lost"</span>)</span>
<span id="cb33-46"><a href="#cb33-46" aria-hidden="true" tabindex="-1"></a>p1</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><img src="asparagopsis-parameterisation_files/figure-html/hall%202023%20gametophyte%20velocity-1.png" class="img-fluid figure-img" width="672"></p>
<figcaption>Combined data from Figures 3.17 (red), 3.18 (green) and 3.19 (blue) from <span class="citation" data-cites="hall_investigating_2023">Hall (<a href="#ref-hall_investigating_2023" role="doc-biblioref">2023</a>)</span> as they appear to be three different experiments with the same treatments.</figcaption>
</figure>
</div>
</div>
</div>
<p>I’m not sure what’s going on with Velocity Experiment 2 (in green, above) but the other two experiments show a clear trend of biomass lost with increasing velocity.</p>
<div class="cell">
<details class="code-fold">
<summary>Create linear relationship for velocity</summary>
<div class="sourceCode cell-code" id="cb34"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb34-1"><a href="#cb34-1" aria-hidden="true" tabindex="-1"></a>hall_gam_vel2 <span class="ot">&lt;-</span> hall_gam_vel <span class="sc">%&gt;%</span> </span>
<span id="cb34-2"><a href="#cb34-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(Exp <span class="sc">!=</span> <span class="st">"Velocity 2"</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb34-3"><a href="#cb34-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(Data <span class="sc">==</span> <span class="st">"Mean"</span>)</span>
<span id="cb34-4"><a href="#cb34-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb34-5"><a href="#cb34-5" aria-hidden="true" tabindex="-1"></a>line_a <span class="ot">&lt;-</span> <span class="fu">nls</span>(<span class="at">formula =</span> <span class="fu">as.formula</span>(perc_biomass_lost <span class="sc">~</span> a <span class="sc">*</span> <span class="fu">as.numeric</span>(Label)),</span>
<span id="cb34-6"><a href="#cb34-6" aria-hidden="true" tabindex="-1"></a>    <span class="at">start =</span> <span class="fu">c</span>(<span class="at">a =</span> <span class="fl">0.1</span>),</span>
<span id="cb34-7"><a href="#cb34-7" aria-hidden="true" tabindex="-1"></a>    <span class="at">data =</span> hall_gam_vel2)</span>
<span id="cb34-8"><a href="#cb34-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb34-9"><a href="#cb34-9" aria-hidden="true" tabindex="-1"></a>D_ve <span class="ot">&lt;-</span> <span class="fu">unname</span>(<span class="fu">coef</span>(line_a))</span>
<span id="cb34-10"><a href="#cb34-10" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb34-11"><a href="#cb34-11" aria-hidden="true" tabindex="-1"></a>p2 <span class="ot">&lt;-</span> p1 <span class="sc">+</span></span>
<span id="cb34-12"><a href="#cb34-12" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_line</span>(<span class="at">data =</span> hall_gam_vel, <span class="fu">aes</span>(<span class="at">x =</span> <span class="fu">as.numeric</span>(Label), <span class="at">y =</span> <span class="dv">100</span><span class="sc">*</span>D_ve<span class="sc">*</span><span class="fu">as.numeric</span>(Label)))</span>
<span id="cb34-13"><a href="#cb34-13" aria-hidden="true" tabindex="-1"></a>p2</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><img src="asparagopsis-parameterisation_files/figure-html/gametophyte%20velocity%20trend-1.png" class="img-fluid figure-img" width="672"></p>
<figcaption>Add linear relationship to above plot</figcaption>
</figure>
</div>
</div>
</div>
</section>
<section id="turbulence" class="level3">
<h3 class="anchored" data-anchor-id="turbulence">Turbulence</h3>
<p><span class="citation" data-cites="hall_investigating_2023">Hall (<a href="#ref-hall_investigating_2023" role="doc-biblioref">2023</a>)</span> also investigated fragmentation due to turbulence on the gametophyte, which is summarised in Figures 3.28 and 3.29.</p>
<div class="cell">
<details class="code-fold">
<summary>Get <span class="citation" data-cites="hall_investigating_2023">Hall (<a href="#ref-hall_investigating_2023" role="doc-biblioref">2023</a>)</span> turbulence data for gametophytes</summary>
<div class="sourceCode cell-code" id="cb35"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb35-1"><a href="#cb35-1" aria-hidden="true" tabindex="-1"></a>hall_3<span class="fl">.28</span> <span class="ot">&lt;-</span> <span class="fu">bind_rows</span>(<span class="fu">list</span>(ls[[<span class="dv">9</span>]], ls[[<span class="dv">10</span>]])) <span class="sc">%&gt;%</span> </span>
<span id="cb35-2"><a href="#cb35-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">Label =</span> <span class="fu">rep</span>(<span class="fu">c</span>(<span class="st">'static'</span>, <span class="st">'low'</span>, <span class="st">'mid'</span>, <span class="st">'high'</span>), <span class="dv">2</span>),</span>
<span id="cb35-3"><a href="#cb35-3" aria-hidden="true" tabindex="-1"></a>         <span class="at">Exp =</span> <span class="st">"Turbulence 1"</span>,</span>
<span id="cb35-4"><a href="#cb35-4" aria-hidden="true" tabindex="-1"></a>         <span class="at">Form =</span> <span class="st">"Gametophyte"</span>,</span>
<span id="cb35-5"><a href="#cb35-5" aria-hidden="true" tabindex="-1"></a>         <span class="at">Data =</span> <span class="fu">c</span>(<span class="fu">rep</span>(<span class="st">"Mean"</span>, <span class="dv">4</span>), <span class="fu">rep</span>(<span class="st">"Mean+SE"</span>, <span class="dv">4</span>))) <span class="sc">%&gt;%</span> </span>
<span id="cb35-6"><a href="#cb35-6" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">select</span>(<span class="sc">-</span>dataset) <span class="sc">%&gt;%</span> </span>
<span id="cb35-7"><a href="#cb35-7" aria-hidden="true" tabindex="-1"></a>  <span class="fu">pivot_wider</span>(<span class="at">names_from =</span> Data, <span class="at">values_from =</span> perc_biomass_lost) <span class="sc">%&gt;%</span> </span>
<span id="cb35-8"><a href="#cb35-8" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">SE =</span> <span class="st">`</span><span class="at">Mean+SE</span><span class="st">`</span> <span class="sc">-</span> Mean) <span class="sc">%&gt;%</span> </span>
<span id="cb35-9"><a href="#cb35-9" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">select</span>(<span class="sc">-</span><span class="st">`</span><span class="at">Mean+SE</span><span class="st">`</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb35-10"><a href="#cb35-10" aria-hidden="true" tabindex="-1"></a>  <span class="fu">pivot_longer</span>(<span class="at">names_to =</span> <span class="st">"Data"</span>, <span class="at">values_to =</span> <span class="st">"perc_biomass_lost"</span>, <span class="at">cols =</span> <span class="fu">c</span>(Mean, SE)) <span class="sc">%&gt;%</span> </span>
<span id="cb35-11"><a href="#cb35-11" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">Label =</span> <span class="fu">factor</span>(Label, <span class="at">levels =</span> turbulence_levels))</span>
<span id="cb35-12"><a href="#cb35-12" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb35-13"><a href="#cb35-13" aria-hidden="true" tabindex="-1"></a>hall_3<span class="fl">.29</span> <span class="ot">&lt;-</span> <span class="fu">bind_rows</span>(<span class="fu">list</span>(ls[[<span class="dv">11</span>]], ls[[<span class="dv">12</span>]])) <span class="sc">%&gt;%</span> </span>
<span id="cb35-14"><a href="#cb35-14" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">Label =</span> <span class="fu">rep</span>(<span class="fu">c</span>(<span class="st">'static'</span>, <span class="st">'low'</span>, <span class="st">'mid'</span>, <span class="st">'high'</span>), <span class="dv">2</span>),</span>
<span id="cb35-15"><a href="#cb35-15" aria-hidden="true" tabindex="-1"></a>         <span class="at">Exp =</span> <span class="st">"Turbulence 2"</span>,</span>
<span id="cb35-16"><a href="#cb35-16" aria-hidden="true" tabindex="-1"></a>         <span class="at">Form =</span> <span class="st">"Gametophyte"</span>,</span>
<span id="cb35-17"><a href="#cb35-17" aria-hidden="true" tabindex="-1"></a>         <span class="at">Data =</span> <span class="fu">c</span>(<span class="fu">rep</span>(<span class="st">"Mean"</span>, <span class="dv">4</span>), <span class="fu">rep</span>(<span class="st">"Mean+SE"</span>, <span class="dv">4</span>))) <span class="sc">%&gt;%</span> </span>
<span id="cb35-18"><a href="#cb35-18" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">select</span>(<span class="sc">-</span>dataset) <span class="sc">%&gt;%</span> </span>
<span id="cb35-19"><a href="#cb35-19" aria-hidden="true" tabindex="-1"></a>  <span class="fu">pivot_wider</span>(<span class="at">names_from =</span> Data, <span class="at">values_from =</span> perc_biomass_lost) <span class="sc">%&gt;%</span> </span>
<span id="cb35-20"><a href="#cb35-20" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">SE =</span> <span class="st">`</span><span class="at">Mean+SE</span><span class="st">`</span> <span class="sc">-</span> Mean) <span class="sc">%&gt;%</span> </span>
<span id="cb35-21"><a href="#cb35-21" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">select</span>(<span class="sc">-</span><span class="st">`</span><span class="at">Mean+SE</span><span class="st">`</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb35-22"><a href="#cb35-22" aria-hidden="true" tabindex="-1"></a>  <span class="fu">pivot_longer</span>(<span class="at">names_to =</span> <span class="st">"Data"</span>, <span class="at">values_to =</span> <span class="st">"perc_biomass_lost"</span>, <span class="at">cols =</span> <span class="fu">c</span>(Mean, SE)) <span class="sc">%&gt;%</span> </span>
<span id="cb35-23"><a href="#cb35-23" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">Label =</span> <span class="fu">factor</span>(Label, <span class="at">levels =</span> turbulence_levels))</span>
<span id="cb35-24"><a href="#cb35-24" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb35-25"><a href="#cb35-25" aria-hidden="true" tabindex="-1"></a>hall_gam_turb <span class="ot">&lt;-</span> <span class="fu">bind_rows</span>(<span class="fu">list</span>(hall_3<span class="fl">.28</span>, hall_3<span class="fl">.29</span>)) <span class="sc">%&gt;%</span> </span>
<span id="cb35-26"><a href="#cb35-26" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">Exp =</span> <span class="fu">as.factor</span>(Exp),</span>
<span id="cb35-27"><a href="#cb35-27" aria-hidden="true" tabindex="-1"></a>         <span class="at">Data =</span> <span class="fu">as.factor</span>(Data))</span>
<span id="cb35-28"><a href="#cb35-28" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb35-29"><a href="#cb35-29" aria-hidden="true" tabindex="-1"></a>p1 <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(<span class="fu">filter</span>(hall_gam_turb, Data <span class="sc">==</span> <span class="st">"Mean"</span>), </span>
<span id="cb35-30"><a href="#cb35-30" aria-hidden="true" tabindex="-1"></a>       <span class="fu">aes</span>(<span class="at">x =</span> Label, <span class="at">y =</span> perc_biomass_lost<span class="sc">*</span><span class="dv">100</span>, <span class="at">fill =</span> Exp)) <span class="sc">+</span></span>
<span id="cb35-31"><a href="#cb35-31" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_col</span>(<span class="at">position =</span> <span class="st">"dodge"</span>) <span class="sc">+</span></span>
<span id="cb35-32"><a href="#cb35-32" aria-hidden="true" tabindex="-1"></a>  prettyplot <span class="sc">+</span></span>
<span id="cb35-33"><a href="#cb35-33" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">x =</span> <span class="st">"Level of turbulence"</span>, <span class="at">y =</span> <span class="st">"Mean % biomass lost"</span>)</span>
<span id="cb35-34"><a href="#cb35-34" aria-hidden="true" tabindex="-1"></a>p1</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><img src="asparagopsis-parameterisation_files/figure-html/hall%202023%20gametophyte%20turbulence-1.png" class="img-fluid figure-img" width="672"></p>
<figcaption>Combined data from Figures 3.28 (red) and 3.29 (blue) from <span class="citation" data-cites="hall_investigating_2023">Hall (<a href="#ref-hall_investigating_2023" role="doc-biblioref">2023</a>)</span> as they appear to be two different experiments with the same treatments.</figcaption>
</figure>
</div>
</div>
</div>
<p>Looks very different between the two treatments, but I’m going to go with the one that shows a clear trend.</p>
<div class="cell">
<details class="code-fold">
<summary>Get turbulence data ready for vector</summary>
<div class="sourceCode cell-code" id="cb36"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb36-1"><a href="#cb36-1" aria-hidden="true" tabindex="-1"></a>D_st <span class="ot">&lt;-</span> hall_3<span class="fl">.28</span><span class="sc">$</span>perc_biomass_lost[hall_3<span class="fl">.28</span><span class="sc">$</span>Data <span class="sc">==</span> <span class="st">"Mean"</span> <span class="sc">&amp;</span> hall_3<span class="fl">.28</span><span class="sc">$</span>Label <span class="sc">==</span> <span class="st">"static"</span>]</span>
<span id="cb36-2"><a href="#cb36-2" aria-hidden="true" tabindex="-1"></a>D_lo <span class="ot">&lt;-</span> hall_3<span class="fl">.28</span><span class="sc">$</span>perc_biomass_lost[hall_3<span class="fl">.28</span><span class="sc">$</span>Data <span class="sc">==</span> <span class="st">"Mean"</span> <span class="sc">&amp;</span> hall_3<span class="fl">.28</span><span class="sc">$</span>Label <span class="sc">==</span> <span class="st">"low"</span>]</span>
<span id="cb36-3"><a href="#cb36-3" aria-hidden="true" tabindex="-1"></a>D_mi <span class="ot">&lt;-</span> hall_3<span class="fl">.28</span><span class="sc">$</span>perc_biomass_lost[hall_3<span class="fl">.28</span><span class="sc">$</span>Data <span class="sc">==</span> <span class="st">"Mean"</span> <span class="sc">&amp;</span> hall_3<span class="fl">.28</span><span class="sc">$</span>Label <span class="sc">==</span> <span class="st">"mid"</span>]</span>
<span id="cb36-4"><a href="#cb36-4" aria-hidden="true" tabindex="-1"></a>D_hi <span class="ot">&lt;-</span> hall_3<span class="fl">.28</span><span class="sc">$</span>perc_biomass_lost[hall_3<span class="fl">.28</span><span class="sc">$</span>Data <span class="sc">==</span> <span class="st">"Mean"</span> <span class="sc">&amp;</span> hall_3<span class="fl">.28</span><span class="sc">$</span>Label <span class="sc">==</span> <span class="st">"high"</span>]</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
</section>
</section>
</section>
<section id="finishing-up-1" class="level1">
<h1>Finishing up</h1>
<section id="sec-final-params-gametophyte" class="level2">
<h2 class="anchored" data-anchor-id="sec-final-params-gametophyte">All parameters and sources</h2>
<table class="caption-top table">
<caption>Final values for species parameters before saving.</caption>
<colgroup>
<col style="width: 25%">
<col style="width: 24%">
<col style="width: 50%">
</colgroup>
<thead>
<tr class="header">
<th>Parameter</th>
<th style="text-align: center;">Value</th>
<th style="text-align: center;">Source</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td><span class="math inline">\(V_{am}\)</span>, <span class="math inline">\(K_{am}\)</span></td>
<td style="text-align: center;">4.9919879, 140.207067</td>
<td style="text-align: center;"><span class="citation" data-cites="schuenhoff_tetrasporophyte_2006">Schuenhoff, Mata, and Santos (<a href="#ref-schuenhoff_tetrasporophyte_2006" role="doc-biblioref">2006</a>)</span></td>
</tr>
<tr class="even">
<td><span class="math inline">\(M_{am}\)</span>, <span class="math inline">\(C_{am}\)</span></td>
<td style="text-align: center;">0.00336, 2.9548534</td>
<td style="text-align: center;"><span class="citation" data-cites="torres_nitrogen_2021">Torres et al. (<a href="#ref-torres_nitrogen_2021" role="doc-biblioref">2021</a>)</span></td>
</tr>
<tr class="odd">
<td><span class="math inline">\(V_{ni}\)</span>, <span class="math inline">\(K_{ni}\)</span></td>
<td style="text-align: center;">NA, NA</td>
<td style="text-align: center;"></td>
</tr>
<tr class="even">
<td><span class="math inline">\(M_{ni}\)</span>, <span class="math inline">\(C_{ni}\)</span></td>
<td style="text-align: center;">0, 0.1138183</td>
<td style="text-align: center;"></td>
</tr>
<tr class="odd">
<td><span class="math inline">\(V_{ot}\)</span>, <span class="math inline">\(K_{ot}\)</span></td>
<td style="text-align: center;">NA, NA</td>
<td style="text-align: center;"></td>
</tr>
<tr class="even">
<td><span class="math inline">\(M_{ot}\)</span>, <span class="math inline">\(C_{ot}\)</span></td>
<td style="text-align: center;">1.44^{-4}, 0.2958215</td>
<td style="text-align: center;"></td>
</tr>
<tr class="odd">
<td><span class="math inline">\(K_{c}\)</span></td>
<td style="text-align: center;">6</td>
<td style="text-align: center;"></td>
</tr>
<tr class="even">
<td><span class="math inline">\(Q_{min}\)</span>, <span class="math inline">\(Q_{max}\)</span></td>
<td style="text-align: center;">19.4316163, 43.0373002</td>
<td style="text-align: center;"></td>
</tr>
<tr class="odd">
<td><span class="math inline">\(N_{min}\)</span>, <span class="math inline">\(N_{max}\)</span></td>
<td style="text-align: center;">0.0194316163410302, 0.0430373001776199</td>
<td style="text-align: center;"></td>
</tr>
<tr class="even">
<td><span class="math inline">\(DWWW\)</span></td>
<td style="text-align: center;">7.68</td>
<td style="text-align: center;"></td>
</tr>
<tr class="odd">
<td><span class="math inline">\(\mu\)</span></td>
<td style="text-align: center;">0.3472222</td>
<td style="text-align: center;"></td>
</tr>
<tr class="even">
<td><span class="math inline">\(D_{m}\)</span>, <span class="math inline">\(D_{v}\)</span></td>
<td style="text-align: center;">NA, 0.0153363</td>
<td style="text-align: center;"></td>
</tr>
<tr class="odd">
<td><span class="math inline">\(D_{lo}\)</span>, <span class="math inline">\(D_{mi}\)</span>, <span class="math inline">\(D_{hi}\)</span></td>
<td style="text-align: center;">0.0538993, 0.1270484, 0.2567621</td>
<td style="text-align: center;"></td>
</tr>
<tr class="even">
<td><span class="math inline">\(I_o\)</span>, <span class="math inline">\(a_{cs}\)</span></td>
<td style="text-align: center;">150, 2.5090131^{-6}</td>
<td style="text-align: center;"></td>
</tr>
<tr class="odd">
<td><span class="math inline">\(T_{opt}\)</span>, <span class="math inline">\(T_{min}\)</span>, <span class="math inline">\(T_{max}\)</span> (<em>A. armata</em>)</td>
<td style="text-align: center;">19.0909091, 8.8954869, 22.4292359</td>
<td style="text-align: center;"></td>
</tr>
<tr class="even">
<td><span class="math inline">\(T_{opt}\)</span>, <span class="math inline">\(T_{min}\)</span>, <span class="math inline">\(T_{max}\)</span> (<em>A. taxiformis</em>)</td>
<td style="text-align: center;">24.6153846, 16.0326603, 29.226247</td>
<td style="text-align: center;"></td>
</tr>
<tr class="odd">
<td><span class="math inline">\(S_{opt}\)</span>, <span class="math inline">\(S_{min}\)</span>, <span class="math inline">\(S_{max}\)</span></td>
<td style="text-align: center;">35.25, 21.15, 41.65</td>
<td style="text-align: center;"></td>
</tr>
<tr class="even">
<td><span class="math inline">\(h_a\)</span>, <span class="math inline">\(h_b\)</span>, <span class="math inline">\(h_c\)</span>, <span class="math inline">\(h_{max}\)</span></td>
<td style="text-align: center;">3850, 1.75, 0.001, 0.26</td>
<td style="text-align: center;"></td>
</tr>
</tbody>
</table>
</section>
<section id="sec-save-gametophyte" class="level2">
<h2 class="anchored" data-anchor-id="sec-save-gametophyte">Saving data</h2>
<div class="cell">
<details class="code-fold">
<summary>Insert values and units into named vector</summary>
<div class="sourceCode cell-code" id="cb37"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb37-1"><a href="#cb37-1" aria-hidden="true" tabindex="-1"></a>a_armata_gametophyte <span class="ot">&lt;-</span> <span class="fu">c</span>(</span>
<span id="cb37-2"><a href="#cb37-2" aria-hidden="true" tabindex="-1"></a>  <span class="at">V_am =</span> V_am,</span>
<span id="cb37-3"><a href="#cb37-3" aria-hidden="true" tabindex="-1"></a>  <span class="at">K_am =</span> K_am,</span>
<span id="cb37-4"><a href="#cb37-4" aria-hidden="true" tabindex="-1"></a>  <span class="at">M_am =</span> M_am,</span>
<span id="cb37-5"><a href="#cb37-5" aria-hidden="true" tabindex="-1"></a>  <span class="at">C_am =</span> C_am,</span>
<span id="cb37-6"><a href="#cb37-6" aria-hidden="true" tabindex="-1"></a>  <span class="at">V_ni =</span> V_ni,</span>
<span id="cb37-7"><a href="#cb37-7" aria-hidden="true" tabindex="-1"></a>  <span class="at">K_ni =</span> K_ni,</span>
<span id="cb37-8"><a href="#cb37-8" aria-hidden="true" tabindex="-1"></a>  <span class="at">M_ni =</span> M_ni,</span>
<span id="cb37-9"><a href="#cb37-9" aria-hidden="true" tabindex="-1"></a>  <span class="at">C_ni =</span> C_ni,</span>
<span id="cb37-10"><a href="#cb37-10" aria-hidden="true" tabindex="-1"></a>  <span class="at">V_ot =</span> V_ot,</span>
<span id="cb37-11"><a href="#cb37-11" aria-hidden="true" tabindex="-1"></a>  <span class="at">K_ot =</span> K_ot,</span>
<span id="cb37-12"><a href="#cb37-12" aria-hidden="true" tabindex="-1"></a>  <span class="at">M_ot =</span> M_ot,</span>
<span id="cb37-13"><a href="#cb37-13" aria-hidden="true" tabindex="-1"></a>  <span class="at">C_ot =</span> C_ot,</span>
<span id="cb37-14"><a href="#cb37-14" aria-hidden="true" tabindex="-1"></a>  <span class="at">Q_min =</span> Q_min, </span>
<span id="cb37-15"><a href="#cb37-15" aria-hidden="true" tabindex="-1"></a>  <span class="at">Q_max =</span> Q_max, </span>
<span id="cb37-16"><a href="#cb37-16" aria-hidden="true" tabindex="-1"></a>  <span class="at">K_c =</span> K_c,                                          <span class="co"># THIS IS A PLACEHOLDER</span></span>
<span id="cb37-17"><a href="#cb37-17" aria-hidden="true" tabindex="-1"></a>  <span class="at">mu =</span> mu,          <span class="co"># From tetrasporophyte</span></span>
<span id="cb37-18"><a href="#cb37-18" aria-hidden="true" tabindex="-1"></a>  <span class="co"># N_min = N_min, </span></span>
<span id="cb37-19"><a href="#cb37-19" aria-hidden="true" tabindex="-1"></a>  <span class="co"># N_max = N_max, </span></span>
<span id="cb37-20"><a href="#cb37-20" aria-hidden="true" tabindex="-1"></a>  <span class="at">D_m =</span> D_m,</span>
<span id="cb37-21"><a href="#cb37-21" aria-hidden="true" tabindex="-1"></a>  <span class="at">D_ve =</span> D_ve, </span>
<span id="cb37-22"><a href="#cb37-22" aria-hidden="true" tabindex="-1"></a>  <span class="at">D_lo =</span> D_lo,</span>
<span id="cb37-23"><a href="#cb37-23" aria-hidden="true" tabindex="-1"></a>  <span class="at">D_mi =</span> D_mi,</span>
<span id="cb37-24"><a href="#cb37-24" aria-hidden="true" tabindex="-1"></a>  <span class="at">D_hi =</span> D_hi,</span>
<span id="cb37-25"><a href="#cb37-25" aria-hidden="true" tabindex="-1"></a>  <span class="at">a_cs =</span> a_cs,                  <span class="co"># or a_cs from figueroa 2006</span></span>
<span id="cb37-26"><a href="#cb37-26" aria-hidden="true" tabindex="-1"></a>  <span class="at">I_o =</span> I_o,</span>
<span id="cb37-27"><a href="#cb37-27" aria-hidden="true" tabindex="-1"></a>  <span class="at">T_opt =</span> T_opt_arma,</span>
<span id="cb37-28"><a href="#cb37-28" aria-hidden="true" tabindex="-1"></a>  <span class="at">T_min =</span> T_min_arma,</span>
<span id="cb37-29"><a href="#cb37-29" aria-hidden="true" tabindex="-1"></a>  <span class="at">T_max =</span> T_max_arma,</span>
<span id="cb37-30"><a href="#cb37-30" aria-hidden="true" tabindex="-1"></a>  <span class="at">S_opt =</span> S_opt,</span>
<span id="cb37-31"><a href="#cb37-31" aria-hidden="true" tabindex="-1"></a>  <span class="at">S_min =</span> S_min,</span>
<span id="cb37-32"><a href="#cb37-32" aria-hidden="true" tabindex="-1"></a>  <span class="at">S_max =</span> S_max,</span>
<span id="cb37-33"><a href="#cb37-33" aria-hidden="true" tabindex="-1"></a>  <span class="at">h_a =</span> h_a,</span>
<span id="cb37-34"><a href="#cb37-34" aria-hidden="true" tabindex="-1"></a>  <span class="at">h_b =</span> h_b,</span>
<span id="cb37-35"><a href="#cb37-35" aria-hidden="true" tabindex="-1"></a>  <span class="at">h_c =</span> h_c,</span>
<span id="cb37-36"><a href="#cb37-36" aria-hidden="true" tabindex="-1"></a>  <span class="at">h_max =</span> h_max,</span>
<span id="cb37-37"><a href="#cb37-37" aria-hidden="true" tabindex="-1"></a>  <span class="at">DWWW =</span> DWWW</span>
<span id="cb37-38"><a href="#cb37-38" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb37-39"><a href="#cb37-39" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb37-40"><a href="#cb37-40" aria-hidden="true" tabindex="-1"></a>a_taxiformis_gametophyte <span class="ot">&lt;-</span> a_armata_gametophyte</span>
<span id="cb37-41"><a href="#cb37-41" aria-hidden="true" tabindex="-1"></a>a_taxiformis_gametophyte[<span class="st">'T_opt'</span>] <span class="ot">&lt;-</span> T_opt_taxi</span>
<span id="cb37-42"><a href="#cb37-42" aria-hidden="true" tabindex="-1"></a>a_taxiformis_gametophyte[<span class="st">'T_max'</span>] <span class="ot">&lt;-</span> T_max_taxi</span>
<span id="cb37-43"><a href="#cb37-43" aria-hidden="true" tabindex="-1"></a>a_taxiformis_gametophyte[<span class="st">'T_min'</span>] <span class="ot">&lt;-</span> T_min_taxi</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<div class="cell">
<details class="code-fold">
<summary>Save named vector and .csv file</summary>
<div class="sourceCode cell-code" id="cb38"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb38-1"><a href="#cb38-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Save data as a named vector - model uses this</span></span>
<span id="cb38-2"><a href="#cb38-2" aria-hidden="true" tabindex="-1"></a><span class="co"># save(a_armata_gametophyte, file = file.path(base_path, "data", "asparagopsis_armata_gametophyte.rda"))</span></span>
<span id="cb38-3"><a href="#cb38-3" aria-hidden="true" tabindex="-1"></a><span class="co"># load(file.path(base_path, "data", "asparagopsis_armata_gametophyte.rda")) # use this to load data from file</span></span>
<span id="cb38-4"><a href="#cb38-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb38-5"><a href="#cb38-5" aria-hidden="true" tabindex="-1"></a><span class="co"># Save duplicate as CSV - easier for humans to read</span></span>
<span id="cb38-6"><a href="#cb38-6" aria-hidden="true" tabindex="-1"></a>df <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="at">parameter =</span> <span class="fu">names</span>(a_armata_gametophyte),</span>
<span id="cb38-7"><a href="#cb38-7" aria-hidden="true" tabindex="-1"></a>           <span class="at">unit =</span> <span class="fu">as.character</span>(param_units),</span>
<span id="cb38-8"><a href="#cb38-8" aria-hidden="true" tabindex="-1"></a>           <span class="at">value =</span> (<span class="fu">unname</span>(a_armata_gametophyte)))</span>
<span id="cb38-9"><a href="#cb38-9" aria-hidden="true" tabindex="-1"></a><span class="fu">write.csv</span>(df, <span class="fu">file.path</span>(<span class="st">"data"</span>, <span class="st">"asparagopsis_armata_gametophyte.csv"</span>))</span>
<span id="cb38-10"><a href="#cb38-10" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb38-11"><a href="#cb38-11" aria-hidden="true" tabindex="-1"></a><span class="co"># save(a_taxiformis_gametophyte, file = file.path(base_path, "data", "asparagopsis_taxiformis_gametophyte.rda"))</span></span>
<span id="cb38-12"><a href="#cb38-12" aria-hidden="true" tabindex="-1"></a>df <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="at">parameter =</span> <span class="fu">names</span>(a_taxiformis_gametophyte),</span>
<span id="cb38-13"><a href="#cb38-13" aria-hidden="true" tabindex="-1"></a>           <span class="at">unit =</span> <span class="fu">as.character</span>(param_units),</span>
<span id="cb38-14"><a href="#cb38-14" aria-hidden="true" tabindex="-1"></a>           <span class="at">value =</span> (<span class="fu">unname</span>(a_taxiformis_gametophyte)))</span>
<span id="cb38-15"><a href="#cb38-15" aria-hidden="true" tabindex="-1"></a><span class="fu">write.csv</span>(df, <span class="fu">file.path</span>(<span class="st">"data"</span>, <span class="st">"asparagopsis_taxiformis_gametophyte.csv"</span>))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
</section>
</section>
<section id="potential-future-extensions" class="level1">
<h1>Potential future extensions</h1>
<p>Some of these have been discussed in meetings, but this section is more a dumping ground for resources/data I find along the way that might be useful later.</p>
<ul>
<li>Dissolved inorganic carbon may be a major (perhaps determining?) limiting factor for growth and photosynthesis in <em>Asparagopsis</em> <span class="citation" data-cites="zanolla_concise_2022">(<a href="#ref-zanolla_concise_2022" role="doc-biblioref">Zanolla et al. 2022</a>)</span>
<ul>
<li><span class="citation" data-cites="sainz-villegas_modelling_2024">Sainz-Villegas (<a href="#ref-sainz-villegas_modelling_2024" role="doc-biblioref">2024</a>)</span> and <span class="citation" data-cites="sainz-villegas_exploring_2024">Sainz-Villegas et al. (<a href="#ref-sainz-villegas_exploring_2024" role="doc-biblioref">2024</a>)</span> investigated the survival of vegetative propagules (juvenile gametophytes) - potential to estimate propagation/spread based on wave exposure</li>
</ul></li>
<li><span class="citation" data-cites="mata_within-species_2017">Mata et al. (<a href="#ref-mata_within-species_2017" role="doc-biblioref">2017</a>)</span> looked at temperature effects on the production of bromoforms</li>
<li><span class="citation" data-cites="stengel_short-term_2014">Stengel et al. (<a href="#ref-stengel_short-term_2014" role="doc-biblioref">2014</a>)</span> - a bunch of stuff for <em>Ulva</em></li>
</ul>
</section>
<section id="code" class="level1">
<h1>Code</h1>
<p>This document was produced using <code>quarto</code> <span class="citation" data-cites="quarto">(<a href="#ref-quarto" role="doc-biblioref">Allaire et al. 2024</a>)</span>. The code in this document was run in R version 4.4.2 <span class="citation" data-cites="R_base">(<a href="#ref-R_base" role="doc-biblioref">R Core Team 2019</a>)</span> using the packages <code>cowplot</code> version 0.8.7 <span class="citation" data-cites="cowplot">(<a href="#ref-cowplot" role="doc-biblioref">Wilke 2024</a>)</span>, <code>formatdown</code> version 0.8.7 <span class="citation" data-cites="formatdown">(<a href="#ref-formatdown" role="doc-biblioref">Layton 2024</a>)</span>, <code>raster</code> version 0.8.7 <span class="citation" data-cites="raster">(<a href="#ref-raster" role="doc-biblioref"><strong>raster?</strong></a>)</span>, <!-- `sp` version 0.8.7 [@sp],  --> and <code>units</code> version 0.8.7 <span class="citation" data-cites="units">(<a href="#ref-units" role="doc-biblioref">Pebesma, Mailund, and Hiebert 2016</a>)</span>. <code>tidyverse</code> includes the following packages:</p>
<table class="caption-top table">
<colgroup>
<col style="width: 11%">
<col style="width: 9%">
<col style="width: 30%">
<col style="width: 11%">
<col style="width: 9%">
<col style="width: 30%">
</colgroup>
<thead>
<tr class="header">
<th>Package</th>
<th>Version</th>
<th>Reference</th>
<th>Package</th>
<th>Version</th>
<th>Reference</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td><code>broom</code></td>
<td>1.0.7</td>
<td><span class="citation" data-cites="broom">Robinson, Hayes, and Couch (<a href="#ref-broom" role="doc-biblioref">2024</a>)</span></td>
<td><code>conflicted</code></td>
<td>1.2.0</td>
<td><span class="citation" data-cites="conflicted">Wickham (<a href="#ref-conflicted" role="doc-biblioref">2023a</a>)</span></td>
</tr>
<tr class="even">
<td><code>cli</code></td>
<td>3.6.4</td>
<td><span class="citation" data-cites="cli">Csárdi (<a href="#ref-cli" role="doc-biblioref">2024</a>)</span></td>
<td><code>dbplyr</code></td>
<td>2.5.0</td>
<td><span class="citation" data-cites="dbplyr">Wickham, Girlich, and Ruiz (<a href="#ref-dbplyr" role="doc-biblioref">2024</a>)</span></td>
</tr>
<tr class="odd">
<td><code>dplyr</code></td>
<td>1.1.4</td>
<td><span class="citation" data-cites="dplyr">Wickham, François, et al. (<a href="#ref-dplyr" role="doc-biblioref">2023</a>)</span></td>
<td><code>dtplyr</code></td>
<td>1.3.1</td>
<td><span class="citation" data-cites="dtplyr">Wickham, Girlich, et al. (<a href="#ref-dtplyr" role="doc-biblioref">2023</a>)</span></td>
</tr>
<tr class="even">
<td><code>forcats</code></td>
<td>1.0.0</td>
<td><span class="citation" data-cites="forcats">Wickham (<a href="#ref-forcats" role="doc-biblioref">2023b</a>)</span></td>
<td><code>ggplot2</code></td>
<td>3.5.1</td>
<td><span class="citation" data-cites="ggplot2">Wickham (<a href="#ref-ggplot2" role="doc-biblioref">2016</a>)</span></td>
</tr>
<tr class="odd">
<td><code>googledrive</code></td>
<td>2.1.1</td>
<td><span class="citation" data-cites="googledrive">D’Agostino McGowan and Bryan (<a href="#ref-googledrive" role="doc-biblioref">2023</a>)</span></td>
<td><code>googlesheets4</code></td>
<td>1.1.1</td>
<td><span class="citation" data-cites="googlesheets4">Bryan (<a href="#ref-googlesheets4" role="doc-biblioref">2023</a>)</span></td>
</tr>
<tr class="even">
<td><code>haven</code></td>
<td>2.5.4</td>
<td><span class="citation" data-cites="haven">Wickham, Miller, and Smith (<a href="#ref-haven" role="doc-biblioref">2023</a>)</span></td>
<td><code>hms</code></td>
<td>1.1.3</td>
<td><span class="citation" data-cites="hms">Müller (<a href="#ref-hms" role="doc-biblioref">2023</a>)</span></td>
</tr>
<tr class="odd">
<td><code>httr</code></td>
<td>1.4.7</td>
<td><span class="citation" data-cites="httr">Wickham (<a href="#ref-httr" role="doc-biblioref">2023c</a>)</span></td>
<td><code>jsonlite</code></td>
<td>1.9.1</td>
<td><span class="citation" data-cites="jsonlite">Ooms (<a href="#ref-jsonlite" role="doc-biblioref">2014</a>)</span></td>
</tr>
<tr class="even">
<td><code>lubridate</code></td>
<td>1.9.4</td>
<td><span class="citation" data-cites="lubridate">Grolemund and Wickham (<a href="#ref-lubridate" role="doc-biblioref">2011</a>)</span></td>
<td><code>magrittr</code></td>
<td>2.0.3</td>
<td><span class="citation" data-cites="magrittr">Bache and Wickham (<a href="#ref-magrittr" role="doc-biblioref">2022</a>)</span></td>
</tr>
<tr class="odd">
<td><code>modelr</code></td>
<td>0.1.11</td>
<td><span class="citation" data-cites="modelr">Wickham (<a href="#ref-modelr" role="doc-biblioref">2023d</a>)</span></td>
<td><code>pillar</code></td>
<td>1.10.1</td>
<td><span class="citation" data-cites="pillar">Müller and Wickham (<a href="#ref-pillar" role="doc-biblioref">2023a</a>)</span></td>
</tr>
<tr class="even">
<td><code>purrr</code></td>
<td>1.0.4</td>
<td><span class="citation" data-cites="purrr">Wickham and Henry (<a href="#ref-purrr" role="doc-biblioref">2023</a>)</span></td>
<td><code>ragg</code></td>
<td>1.3.3</td>
<td><span class="citation" data-cites="ragg">Pedersen and Shemanarev (<a href="#ref-ragg" role="doc-biblioref">2024</a>)</span></td>
</tr>
<tr class="odd">
<td><code>readr</code></td>
<td>2.1.5</td>
<td><span class="citation" data-cites="readr">Wickham, Hester, and Bryan (<a href="#ref-readr" role="doc-biblioref">2024</a>)</span></td>
<td><code>readxl</code></td>
<td>1.4.5</td>
<td><span class="citation" data-cites="readxl">Wickham and Bryan (<a href="#ref-readxl" role="doc-biblioref">2023</a>)</span></td>
</tr>
<tr class="even">
<td><code>reprex</code></td>
<td>2.1.1</td>
<td><span class="citation" data-cites="reprex">Bryan et al. (<a href="#ref-reprex" role="doc-biblioref">2024</a>)</span></td>
<td><code>rlang</code></td>
<td>1.1.5</td>
<td><span class="citation" data-cites="rlang">Henry and Wickham (<a href="#ref-rlang" role="doc-biblioref">2024</a>)</span></td>
</tr>
<tr class="odd">
<td><code>rstudioapi</code></td>
<td>0.17.1</td>
<td><span class="citation" data-cites="rstudioapi">Ushey et al. (<a href="#ref-rstudioapi" role="doc-biblioref">2024</a>)</span></td>
<td><code>rvest</code></td>
<td>1.0.4</td>
<td><span class="citation" data-cites="rvest">Wickham (<a href="#ref-rvest" role="doc-biblioref">2024</a>)</span></td>
</tr>
<tr class="even">
<td><code>stringr</code></td>
<td>1.5.1</td>
<td><span class="citation" data-cites="stringr">Wickham (<a href="#ref-stringr" role="doc-biblioref">2023e</a>)</span></td>
<td><code>tibble</code></td>
<td>3.2.1</td>
<td><span class="citation" data-cites="tibble">Müller and Wickham (<a href="#ref-tibble" role="doc-biblioref">2023b</a>)</span></td>
</tr>
<tr class="odd">
<td><code>tidyr</code></td>
<td>1.3.1</td>
<td><span class="citation" data-cites="tidyr">Wickham, Vaughan, and Girlich (<a href="#ref-tidyr" role="doc-biblioref">2024</a>)</span></td>
<td><code>xml2</code></td>
<td>1.3.8</td>
<td><span class="citation" data-cites="xml2">Wickham, Hester, and Ooms (<a href="#ref-xml2" role="doc-biblioref">2023</a>)</span></td>
</tr>
</tbody>
</table>
</section>
<section id="references" class="level1 unnumbered">


</section>

<div id="quarto-appendix" class="default"><section class="quarto-appendix-contents" role="doc-bibliography" id="quarto-bibliography"><h2 class="anchored quarto-appendix-heading">References</h2><div id="refs" class="references csl-bib-body hanging-indent" data-entry-spacing="0" role="list">
<div id="ref-quarto" class="csl-entry" role="listitem">
Allaire, J. J., Charles Teague, Carlos Scheidegger, Yihui Xie, and Christophe Dervieux. 2024. <span>‘<span>Quarto</span>’</span>. <a href="https://doi.org/10.5281/zenodo.5960048">https://doi.org/10.5281/zenodo.5960048</a>.
</div>
<div id="ref-angell_protein_2016" class="csl-entry" role="listitem">
Angell, Alex R., Leonardo Mata, Rocky De Nys, and Nicholas A. Paul. 2016. <span>‘The Protein Content of Seaweeds: A Universal Nitrogen-to-Protein Conversion Factor of Five’</span>. <em>Journal of Applied Phycology</em> 28 (1): 511–24. <a href="http://doi.org/10.1007/s10811-015-0650-1">http://doi.org/10.1007/s10811-015-0650-1</a>.
</div>
<div id="ref-atkinson_cnp_1983" class="csl-entry" role="listitem">
Atkinson, M. J., and S. V. Smith. 1983. <span>‘C:<span>N</span>:<span>P</span> Ratios of Benthic Marine Plants’</span>. <em>Limnology and Oceanography</em> 28 (3): 568–74. <a href="https://doi.org/10.4319/lo.1983.28.3.0568">https://doi.org/10.4319/lo.1983.28.3.0568</a>.
</div>
<div id="ref-magrittr" class="csl-entry" role="listitem">
Bache, Stefan Milton, and Hadley Wickham. 2022. <em>Magrittr: A Forward-Pipe Operator for r</em>. <a href="https://CRAN.R-project.org/package=magrittr">https://CRAN.R-project.org/package=magrittr</a>.
</div>
<div id="ref-brooke_methane_2020" class="csl-entry" role="listitem">
Brooke, Charles G., Breanna M. Roque, Claire Shaw, Negeen Najafi, Maria Gonzalez, Abigail Pfefferlen, Vannesa De Anda, et al. 2020. <span>‘Methane <span>Reduction</span> <span>Potential</span> of <span>Two</span> <span>Pacific</span> <span>Coast</span> <span>Macroalgae</span> <span>During</span> in Vitro <span>Ruminant</span> <span>Fermentation</span>’</span>. <em>Frontiers in Marine Science</em> 7. <a href="https://doi.org/10.3389/fmars.2020.00561">https://doi.org/10.3389/fmars.2020.00561</a>.
</div>
<div id="ref-googlesheets4" class="csl-entry" role="listitem">
Bryan, Jennifer. 2023. <em>Googlesheets4: Access Google Sheets Using the Sheets API V4</em>. <a href="https://CRAN.R-project.org/package=googlesheets4">https://CRAN.R-project.org/package=googlesheets4</a>.
</div>
<div id="ref-reprex" class="csl-entry" role="listitem">
Bryan, Jennifer, Jim Hester, David Robinson, Hadley Wickham, and Christophe Dervieux. 2024. <em>Reprex: Prepare Reproducible Example Code via the Clipboard</em>. <a href="https://CRAN.R-project.org/package=reprex">https://CRAN.R-project.org/package=reprex</a>.
</div>
<div id="ref-chualain_invasive_2004" class="csl-entry" role="listitem">
Chualáin, Fionnuala Ní, Christine A. Maggs, Gary W. Saunders, and Michael D. Guiry. 2004. <span>‘The <span>Invasive</span> <span>Genus</span> <em>Asparagopsis</em> (Bonnemaisoniaceae, <span>Rhodophyta</span>): <span>Molecular</span> <span>Systematics</span>, <span>Morphology</span>, and <span>Ecophysiology</span> of <em>Falkenbergia</em> <span>Isolates</span>’</span>. <em>Journal of Phycology</em> 40 (6): 1112–26. <a href="https://doi.org/10.1111/j.1529-8817.2004.03135.x">https://doi.org/10.1111/j.1529-8817.2004.03135.x</a>.
</div>
<div id="ref-cole_using_2014" class="csl-entry" role="listitem">
Cole, Andrew J., Leonardo Mata, Nicholas A. Paul, and Rocky De Nys. 2014. <span>‘Using <span>CO</span><span class="math inline">\(_{\textrm{2}}\)</span> to Enhance Carbon Capture and Biomass Applications of Freshwater Macroalgae’</span>. <em>GCB Bioenergy</em> 6 (6): 637–45. <a href="https://doi.org/10.1111/gcbb.12097">https://doi.org/10.1111/gcbb.12097</a>.
</div>
<div id="ref-cli" class="csl-entry" role="listitem">
Csárdi, Gábor. 2024. <em>Cli: Helpers for Developing Command Line Interfaces</em>. <a href="https://CRAN.R-project.org/package=cli">https://CRAN.R-project.org/package=cli</a>.
</div>
<div id="ref-googledrive" class="csl-entry" role="listitem">
D’Agostino McGowan, Lucy, and Jennifer Bryan. 2023. <em>Googledrive: An Interface to Google Drive</em>. <a href="https://CRAN.R-project.org/package=googledrive">https://CRAN.R-project.org/package=googledrive</a>.
</div>
<div id="ref-dijoux_more_2014" class="csl-entry" role="listitem">
Dijoux, Laury, Frédérique Viard, and Claude Payri. 2014. <span>‘The <span>More</span> <span>We</span> <span>Search</span>, the <span>More</span> <span>We</span> <span>Find</span>: <span>Discovery</span> of a <span>New</span> <span>Lineage</span> and a <span>New</span> <span>Species</span> <span>Complex</span> in the <span>Genus</span> <em>Asparagopsis</em>’</span>. <em>PLOS ONE</em> 9 (7): e103826. <a href="https://doi.org/10.1371/journal.pone.0103826">https://doi.org/10.1371/journal.pone.0103826</a>.
</div>
<div id="ref-dishon_effect_2023" class="csl-entry" role="listitem">
Dishon, Gal, Hannah M. Resetarits, Brandon Tsai, Ashley L. Jones, Vinayak Agarwal, and Jennifer E. Smith. 2023. <span>‘The Effect of Light Intensity, Spectrum, and Photoperiod on the Physiological Performance of <em><span>Asparagopsis</span> Taxiformis</em> Tetrasporophytes’</span>. <em>Algal Research</em> 76: 103304. <a href="https://doi.org/10.1016/j.algal.2023.103304">https://doi.org/10.1016/j.algal.2023.103304</a>.
</div>
<div id="ref-figueroa_use_2006" class="csl-entry" role="listitem">
Figueroa, Félix L., Rui Santos, Rafael Conde-Álvarez, Leonardo Mata, Juan Luis Gómez Pinchetti, Joana Matos, Pirjo Huovinen, Andreas Schuenhoff, and João Silva. 2006. <span>‘The Use of Chlorophyll Fluorescence for Monitoring Photosynthetic Condition of Two Tank-Cultivated Red Macroalgae Using Fishpond Effluents’</span>. <em>Botanica Marina</em> 49 (4): 275–82. <a href="https://doi.org/10.1515/BOT.2006.035">https://doi.org/10.1515/BOT.2006.035</a>.
</div>
<div id="ref-grisenthwaite_establishing_2023" class="csl-entry" role="listitem">
Grisenthwaite, Robert David Karr. 2023. <span>‘Establishing Land Tank Cultivation of <span>Asparagopsis</span> Taxiformis in <span>Scotland</span>’</span>. Master of {Research}, Scotland, UK: University of the Highlands; Islands.
</div>
<div id="ref-lubridate" class="csl-entry" role="listitem">
Grolemund, Garrett, and Hadley Wickham. 2011. <span>‘Dates and Times Made Easy with <span class="nocase">lubridate</span>’</span>. <em>Journal of Statistical Software</em> 40 (3): 1–25. <a href="https://www.jstatsoft.org/v40/i03/">https://www.jstatsoft.org/v40/i03/</a>.
</div>
<div id="ref-hall_investigating_2023" class="csl-entry" role="listitem">
Hall, Ailise Maree. 2023. <span>‘Investigating the Effects of Water Motion on Fragmentation of the Red Seaweed <em><span>Asparagopsis</span> Armata</em>’</span>. Master of {Science}, Auckland, NZ: University of Auckland.
</div>
<div id="ref-rlang" class="csl-entry" role="listitem">
Henry, Lionel, and Hadley Wickham. 2024. <em>Rlang: Functions for Base Types and Core r and ’Tidyverse’ Features</em>. <a href="https://CRAN.R-project.org/package=rlang">https://CRAN.R-project.org/package=rlang</a>.
</div>
<div id="ref-kraan_commercial_2005" class="csl-entry" role="listitem">
Kraan, Stefan, and Kelly A. Barrington. 2005. <span>‘Commercial Farming of <em><span>Asparagopsis</span> Armata</em> (<span>Bonnemaisoniceae</span>, <span>Rhodophyta</span>) in <span>Ireland</span>, Maintenance of an Introduced Species?’</span> <em>Journal of Applied Phycology</em> 17 (2): 103–10. <a href="https://doi.org/10.1007/s10811-005-2799-5">https://doi.org/10.1007/s10811-005-2799-5</a>.
</div>
<div id="ref-formatdown" class="csl-entry" role="listitem">
Layton, Richard. 2024. <em>Formatdown: Formatting Numbers in ’Rmarkdown’ Documents</em>. <a href="https://CRAN.R-project.org/package=formatdown">https://CRAN.R-project.org/package=formatdown</a>.
</div>
<div id="ref-mata_integrated_2008" class="csl-entry" role="listitem">
Mata, Leonardo. 2008. <span>‘Integrated Aquaculture of <span>Bonnemaisoniaceae</span>: Physiological and Nutritional Controls of Biomass Production and of Halogenated Metabolite Content’</span>. Doctor of {Philosophy} in {Marine} {Sciences}, Faro, Portugal: University of Algarve.
</div>
<div id="ref-mata_within-species_2017" class="csl-entry" role="listitem">
Mata, Leonardo, Rebecca J. Lawton, Marie Magnusson, Nikos Andreakis, Rocky De Nys, and Nicholas A. Paul. 2017. <span>‘Within-Species and Temperature-Related Variation in the Growth and Natural Products of the Red Alga <em><span>Asparagopsis</span> Taxiformis</em>’</span>. <em>Journal of Applied Phycology</em> 29 (3): 1437–47. <a href="http://doi.org/10.1007/s10811-016-1017-y">http://doi.org/10.1007/s10811-016-1017-y</a>.
</div>
<div id="ref-mata_direct_2010" class="csl-entry" role="listitem">
Mata, Leonardo, Andreas Schuenhoff, and Rui Santos. 2010. <span>‘A Direct Comparison of the Performance of the Seaweed Biofilters, <em><span>Asparagopsis</span> Armata</em> and <em><span>Ulva</span> Rigida</em>’</span>. <em>Journal of Applied Phycology</em> 22 (5): 639–44. <a href="https://doi.org/10.1007/s10811-010-9504-z">https://doi.org/10.1007/s10811-010-9504-z</a>.
</div>
<div id="ref-mata_effects_2006" class="csl-entry" role="listitem">
Mata, Leonardo, João Silva, Andreas Schuenhoff, and Rui Santos. 2006. <span>‘The Effects of Light and Temperature on the Photosynthesis of the <em><span>Asparagopsis</span> Armata</em> Tetrasporophyte (<em><span>Falkenbergia</span> Rufolanosa</em>), Cultivated in Tanks’</span>. <em>Aquaculture</em>, Seaweed-<span>Based</span> <span>Integrated</span> <span>Mariculture</span>, 252 (1): 12–19. <a href="https://doi.org/10.1016/j.aquaculture.2005.11.045">https://doi.org/10.1016/j.aquaculture.2005.11.045</a>.
</div>
<div id="ref-mcdermid_nutritional_2003" class="csl-entry" role="listitem">
McDermid, Karla J., and Brooke Stuercke. 2003. <span>‘Nutritional Composition of Edible <span>Hawaiian</span> Seaweeds’</span>. <em>Journal of Applied Phycology</em> 15 (6): 513–24. <a href="https://doi.org/10.1023/B:JAPH.0000004345.31686.7f">https://doi.org/10.1023/B:JAPH.0000004345.31686.7f</a>.
</div>
<div id="ref-mihaila_new_2022" class="csl-entry" role="listitem">
Mihaila, Alisa A., Christopher R. K. Glasson, Rebecca Lawton, Stefan Muetzel, German Molano, and Marie Magnusson. 2022. <span>‘New Temperate Seaweed Targets for Mitigation of Ruminant Methane Emissions: An in Vitro Assessment’</span>. <em>Applied Phycology</em> 3 (1): 274–84. <a href="https://doi.org/10.1080/26388081.2022.2059700">https://doi.org/10.1080/26388081.2022.2059700</a>.
</div>
<div id="ref-mihaila_moderate_2024" class="csl-entry" role="listitem">
Mihaila, Alisa A., Rebecca J. Lawton, Christopher R. K. Glasson, and Marie Magnusson. 2024. <span>‘Moderate Temperature and Water Flow Increase Growth During the Nursery Phase of <em><span>Asparagopsis</span> Armata</em>’</span>. <em>Algal Research</em> 78: 103380. <a href="https://doi.org/10.1016/j.algal.2023.103380">https://doi.org/10.1016/j.algal.2023.103380</a>.
</div>
<div id="ref-monro_evolutionary_2007" class="csl-entry" role="listitem">
Monro, Keyne. 2007. <span>‘Evolutionary Consequences of Growth-Form Plasticity in a Red Seaweed’</span>. Doctor of {Philosophy}, Sydney, Australia: University of New South Wales. <a href="http://doi.org/10.26190/unsworks/17535">http://doi.org/10.26190/unsworks/17535</a>.
</div>
<div id="ref-hms" class="csl-entry" role="listitem">
Müller, Kirill. 2023. <em>Hms: Pretty Time of Day</em>. <a href="https://CRAN.R-project.org/package=hms">https://CRAN.R-project.org/package=hms</a>.
</div>
<div id="ref-pillar" class="csl-entry" role="listitem">
Müller, Kirill, and Hadley Wickham. 2023a. <em>Pillar: Coloured Formatting for Columns</em>. <a href="https://CRAN.R-project.org/package=pillar">https://CRAN.R-project.org/package=pillar</a>.
</div>
<div id="ref-tibble" class="csl-entry" role="listitem">
———. 2023b. <em>Tibble: Simple Data Frames</em>. <a href="https://CRAN.R-project.org/package=tibble">https://CRAN.R-project.org/package=tibble</a>.
</div>
<div id="ref-nunes_evaluation_2024" class="csl-entry" role="listitem">
Nunes, Helder P. B., Cristiana S. A. M. Maduro Dias, Nuno V. Álvaro, and Alfredo E. S. Borba. 2024. <span>‘Evaluation of <span>Two</span> <span>Species</span> of <span>Macroalgae</span> from <span>Azores</span> <span>Sea</span> as <span>Potential</span> <span>Reducers</span> of <span>Ruminal</span> <span>Methane</span> <span>Production</span>: <span>In</span> <span>Vitro</span> <span>Ruminal</span> <span>Assay</span>’</span>. <em>Animals</em> 14 (6): 967. <a href="https://doi.org/10.3390/ani14060967">https://doi.org/10.3390/ani14060967</a>.
</div>
<div id="ref-jsonlite" class="csl-entry" role="listitem">
Ooms, Jeroen. 2014. <span>‘The Jsonlite Package: A Practical and Consistent Mapping Between JSON Data and r Objects’</span>. <em>arXiv:1403.2805 [Stat.CO]</em>. <a href="https://arxiv.org/abs/1403.2805">https://arxiv.org/abs/1403.2805</a>.
</div>
<div id="ref-pacheco_invasive_2020" class="csl-entry" role="listitem">
Pacheco, Diana, Glacio Souza Araújo, João Cotas, Rui Gaspar, João M. Neto, and Leonel Pereira. 2020. <span>‘Invasive <span>Seaweeds</span> in the <span>Iberian</span> <span>Peninsula</span>: <span>A</span> <span>Contribution</span> for <span>Food</span> <span>Supply</span>’</span>. <em>Marine Drugs</em> 18 (11): 560. <a href="https://doi.org/10.3390/md18110560">https://doi.org/10.3390/md18110560</a>.
</div>
<div id="ref-padilla-gamino_seasonal_2007" class="csl-entry" role="listitem">
Padilla-Gamiño, Jacqueline L., and Robert C. Carpenter. 2007. <span>‘Seasonal Acclimatization of <em><span>Asparagopsis</span> Taxiformis</em> (<span>Rhodophyta</span>) from Different Biogeographic Regions’</span>. <em>Limnology and Oceanography</em> 52 (2): 833–42. <a href="https://doi.org/10.4319/lo.2007.52.2.0833">https://doi.org/10.4319/lo.2007.52.2.0833</a>.
</div>
<div id="ref-units" class="csl-entry" role="listitem">
Pebesma, Edzer, Thomas Mailund, and James Hiebert. 2016. <span>‘Measurement Units in <span>R</span>’</span>. <em>R Journal</em> 8 (2): 486–94. <a href="https://doi.org/10.32614/RJ-2016-061">https://doi.org/10.32614/RJ-2016-061</a>.
</div>
<div id="ref-ragg" class="csl-entry" role="listitem">
Pedersen, Thomas Lin, and Maxim Shemanarev. 2024. <em>Ragg: Graphic Devices Based on AGG</em>. <a href="https://CRAN.R-project.org/package=ragg">https://CRAN.R-project.org/package=ragg</a>.
</div>
<div id="ref-R_base" class="csl-entry" role="listitem">
R Core Team. 2019. <em>R: A Language and Environment for Statistical Computing</em>. Vienna, Austria: R Foundation for Statistical Computing. <a href="https://www.R-project.org">https://www.R-project.org</a>.
</div>
<div id="ref-broom" class="csl-entry" role="listitem">
Robinson, David, Alex Hayes, and Simon Couch. 2024. <em>Broom: Convert Statistical Objects into Tidy Tibbles</em>. <a href="https://CRAN.R-project.org/package=broom">https://CRAN.R-project.org/package=broom</a>.
</div>
<div id="ref-roque_effect_2019" class="csl-entry" role="listitem">
Roque, Breanna Michell, Charles Garrett Brooke, Joshua Ladau, Tamsen Polley, Lyndsey Jean Marsh, Negeen Najafi, Pramod Pandey, et al. 2019. <span>‘Effect of the Macroalgae <em><span>Asparagopsis</span> Taxiformis</em> on Methane Production and Rumen Microbiome Assemblage’</span>. <em>Animal Microbiome</em> 1 (1): 3. <a href="https://doi.org/10.1186/s42523-019-0004-4">https://doi.org/10.1186/s42523-019-0004-4</a>.
</div>
<div id="ref-sainz-villegas_modelling_2024" class="csl-entry" role="listitem">
Sainz-Villegas, Samuel. 2024. <span>‘Modelling the <span>Effects</span> of <span>Climate</span> <span>Change</span> on the <span>Distribution</span> of <span>Subtidal</span> <span>Macroalgae</span>: <span>The</span> <span>Role</span> of <span>Vegetative</span> <span>Propagation</span>’</span>. Doctor of {Philosophy}, Santander, Spain: Universidad de Cantabria.
</div>
<div id="ref-sainz-villegas_exploring_2024" class="csl-entry" role="listitem">
Sainz-Villegas, Samuel, Begoña Sánchez-Astráin, Araceli Puente, and José A. Juanes. 2024. <span>‘Exploring the Effects of Temperature and Light Availability on the Vegetative Propagation Processes of the Non-Native Species <em><span>Asparagopsis</span> Armata</em>’</span>. <em>Frontiers in Marine Science</em> 10: 1343353. <a href="https://doi.org/10.3389/fmars.2023.1343353">https://doi.org/10.3389/fmars.2023.1343353</a>.
</div>
<div id="ref-schuenhoff_tetrasporophyte_2006" class="csl-entry" role="listitem">
Schuenhoff, Andreas, Leonardo Mata, and Rui Santos. 2006. <span>‘The Tetrasporophyte of <em><span>Asparagopsis</span> Armata</em> as a Novel Seaweed Biofilter’</span>. <em>Aquaculture</em>, Seaweed-<span>Based</span> <span>Integrated</span> <span>Mariculture</span>, 252 (1): 3–11. <a href="https://doi.org/10.1016/j.aquaculture.2005.11.044">https://doi.org/10.1016/j.aquaculture.2005.11.044</a>.
</div>
<div id="ref-stengel_short-term_2014" class="csl-entry" role="listitem">
Stengel, D. B., R. Conde-Álvarez, S. Connan, U. Nitschke, F. Arenas, H. Abreu, J. Bonomi Barufi, Fungyi Chow, D. Robledo, and E. J. Malta. 2014. <span>‘Short-Term Effects of <span>CO</span><span class="math inline">\(_{\textrm{2}}\)</span>, Nutrients and Temperature on Three Marine Macroalgae Under Solar Radiation’</span>. <em>Aquatic Biology</em> 22: 159–76. <a href="https://doi.org/10.3354/ab00576">https://doi.org/10.3354/ab00576</a>.
</div>
<div id="ref-torres_effects_2024" class="csl-entry" role="listitem">
Torres, Raquel C., Ana M. Campos, Jacob Goldman, Isabel Barrote, Leonardo Mata, and João Silva. 2024. <span>‘Effects of Light Quality and Intensity on Growth and Bromoform Content of the Red Seaweed <em><span>Asparagopsis</span> Taxiformis</em>’</span>. <em>Journal of Applied Phycology</em> 36 (2): 627–37. <a href="https://doi.org/10.1007/s10811-023-03052-6">https://doi.org/10.1007/s10811-023-03052-6</a>.
</div>
<div id="ref-torres_nitrogen_2021" class="csl-entry" role="listitem">
Torres, Raquel C., Leonardo Mata, Rui Santos, and Ana Alexandre. 2021. <span>‘Nitrogen Uptake Kinetics of an Enteric Methane Inhibitor, the Red Seaweed <em><span>Asparagopsis</span> Armata</em>’</span>. <em>Journal of Applied Phycology</em> 33 (6): 4001–9. <a href="https://doi.org/10.1007/s10811-021-02604-y">https://doi.org/10.1007/s10811-021-02604-y</a>.
</div>
<div id="ref-rstudioapi" class="csl-entry" role="listitem">
Ushey, Kevin, JJ Allaire, Hadley Wickham, and Gary Ritchie. 2024. <em>Rstudioapi: Safely Access the RStudio API</em>. <a href="https://CRAN.R-project.org/package=rstudioapi">https://CRAN.R-project.org/package=rstudioapi</a>.
</div>
<div id="ref-ggplot2" class="csl-entry" role="listitem">
Wickham, Hadley. 2016. <em>Ggplot2: Elegant Graphics for Data Analysis</em>. Springer-Verlag New York. <a href="https://ggplot2.tidyverse.org">https://ggplot2.tidyverse.org</a>.
</div>
<div id="ref-conflicted" class="csl-entry" role="listitem">
———. 2023a. <em>Conflicted: An Alternative Conflict Resolution Strategy</em>. <a href="https://CRAN.R-project.org/package=conflicted">https://CRAN.R-project.org/package=conflicted</a>.
</div>
<div id="ref-forcats" class="csl-entry" role="listitem">
———. 2023b. <em>Forcats: Tools for Working with Categorical Variables (Factors)</em>. <a href="https://CRAN.R-project.org/package=forcats">https://CRAN.R-project.org/package=forcats</a>.
</div>
<div id="ref-httr" class="csl-entry" role="listitem">
———. 2023c. <em>Httr: Tools for Working with URLs and HTTP</em>. <a href="https://CRAN.R-project.org/package=httr">https://CRAN.R-project.org/package=httr</a>.
</div>
<div id="ref-modelr" class="csl-entry" role="listitem">
———. 2023d. <em>Modelr: Modelling Functions That Work with the Pipe</em>. <a href="https://CRAN.R-project.org/package=modelr">https://CRAN.R-project.org/package=modelr</a>.
</div>
<div id="ref-stringr" class="csl-entry" role="listitem">
———. 2023e. <em>Stringr: Simple, Consistent Wrappers for Common String Operations</em>. <a href="https://CRAN.R-project.org/package=stringr">https://CRAN.R-project.org/package=stringr</a>.
</div>
<div id="ref-rvest" class="csl-entry" role="listitem">
———. 2024. <em>Rvest: Easily Harvest (Scrape) Web Pages</em>. <a href="https://CRAN.R-project.org/package=rvest">https://CRAN.R-project.org/package=rvest</a>.
</div>
<div id="ref-readxl" class="csl-entry" role="listitem">
Wickham, Hadley, and Jennifer Bryan. 2023. <em>Readxl: Read Excel Files</em>. <a href="https://CRAN.R-project.org/package=readxl">https://CRAN.R-project.org/package=readxl</a>.
</div>
<div id="ref-dplyr" class="csl-entry" role="listitem">
Wickham, Hadley, Romain François, Lionel Henry, Kirill Müller, and Davis Vaughan. 2023. <em>Dplyr: A Grammar of Data Manipulation</em>. <a href="https://CRAN.R-project.org/package=dplyr">https://CRAN.R-project.org/package=dplyr</a>.
</div>
<div id="ref-dtplyr" class="csl-entry" role="listitem">
Wickham, Hadley, Maximilian Girlich, Mark Fairbanks, and Ryan Dickerson. 2023. <em>Dtplyr: Data Table Back-End for ’Dplyr’</em>. <a href="https://CRAN.R-project.org/package=dtplyr">https://CRAN.R-project.org/package=dtplyr</a>.
</div>
<div id="ref-dbplyr" class="csl-entry" role="listitem">
Wickham, Hadley, Maximilian Girlich, and Edgar Ruiz. 2024. <em>Dbplyr: A ’Dplyr’ Back End for Databases</em>. <a href="https://CRAN.R-project.org/package=dbplyr">https://CRAN.R-project.org/package=dbplyr</a>.
</div>
<div id="ref-purrr" class="csl-entry" role="listitem">
Wickham, Hadley, and Lionel Henry. 2023. <em>Purrr: Functional Programming Tools</em>. <a href="https://CRAN.R-project.org/package=purrr">https://CRAN.R-project.org/package=purrr</a>.
</div>
<div id="ref-readr" class="csl-entry" role="listitem">
Wickham, Hadley, Jim Hester, and Jennifer Bryan. 2024. <em>Readr: Read Rectangular Text Data</em>. <a href="https://CRAN.R-project.org/package=readr">https://CRAN.R-project.org/package=readr</a>.
</div>
<div id="ref-xml2" class="csl-entry" role="listitem">
Wickham, Hadley, Jim Hester, and Jeroen Ooms. 2023. <em>Xml2: Parse XML</em>. <a href="https://CRAN.R-project.org/package=xml2">https://CRAN.R-project.org/package=xml2</a>.
</div>
<div id="ref-haven" class="csl-entry" role="listitem">
Wickham, Hadley, Evan Miller, and Danny Smith. 2023. <em>Haven: Import and Export ’SPSS’, ’Stata’ and ’SAS’ Files</em>. <a href="https://CRAN.R-project.org/package=haven">https://CRAN.R-project.org/package=haven</a>.
</div>
<div id="ref-tidyr" class="csl-entry" role="listitem">
Wickham, Hadley, Davis Vaughan, and Maximilian Girlich. 2024. <em>Tidyr: Tidy Messy Data</em>. <a href="https://CRAN.R-project.org/package=tidyr">https://CRAN.R-project.org/package=tidyr</a>.
</div>
<div id="ref-cowplot" class="csl-entry" role="listitem">
Wilke, Claus O. 2024. <em>Cowplot: Streamlined Plot Theme and Plot Annotations for ’Ggplot2’</em>. <a href="https://CRAN.R-project.org/package=cowplot">https://CRAN.R-project.org/package=cowplot</a>.
</div>
<div id="ref-wiltshire_feasibility_2015" class="csl-entry" role="listitem">
Wiltshire, Kathryn H., Jason E. Tanner, C. Fred D. Gurgel, and Marty R. Deveney. 2015. <span>‘Feasibility Study for Integrated Multitrophic Aquaculture in Southern <span>Australia</span>’</span>. {SARDI} {Publication} F2015/000786-1.
</div>
<div id="ref-wright_asexual_2022" class="csl-entry" role="listitem">
Wright, Jeffrey T., Elysha J. Kennedy, Rocky de Nys, and Masayuki Tatsumi. 2022. <span>‘Asexual Propagation of <em><span>Asparagopsis</span> Armata</em> Gametophytes: Fragmentation, Regrowth and Attachment Mechanisms for Sea-Based Cultivation’</span>. <em>Journal of Applied Phycology</em> 34 (4): 2135–44. <a href="https://doi.org/10.1007/s10811-022-02763-6">https://doi.org/10.1007/s10811-022-02763-6</a>.
</div>
<div id="ref-zanolla_assessing_2018" class="csl-entry" role="listitem">
Zanolla, Marianela, María Altamirano, Raquel Carmona, Julio De La Rosa, Virginia Souza‐Egipsy, Alison Sherwood, Konstantinos Tsiamis, Ana Márcia Barbosa, Antonio Román Muñoz, and Nikos Andreakis. 2018. <span>‘Assessing Global Range Expansion in a Cryptic Species Complex: Insights from the Red Seaweed Genus <em>Asparagopsis</em> (<span>Florideophyceae</span>)’</span>. Edited by K. Müller. <em>Journal of Phycology</em> 54 (1): 12–24. <a href="https://doi.org/10.1111/jpy.12598">https://doi.org/10.1111/jpy.12598</a>.
</div>
<div id="ref-zanolla_size_2018" class="csl-entry" role="listitem">
Zanolla, Marianela, María Altamirano, Julio De La Rosa, Francisco X. Niell, and Raquel Carmona. 2018. <span>‘Size Structure and Dynamics of an Invasive Population of Lineage 2 of <em><span>Asparagopsis</span> Taxiformis</em> (<span>Florideophyceae</span>) in the <span>Alboran</span> <span>Sea</span>’</span>. <em>Phycological Research</em> 66 (1): 45–51. <a href="https://doi.org/10.1111/pre.12189">https://doi.org/10.1111/pre.12189</a>.
</div>
<div id="ref-zanolla_photosynthetic_2015" class="csl-entry" role="listitem">
Zanolla, Marianela, M. Altamirano, R. Carmona, J. De La Rosa, A. Sherwood, and N. Andreakis. 2015. <span>‘Photosynthetic Plasticity of the Genus <em>Asparagopsis</em> (<span>Bonnemaisoniales</span>, <span>Rhodophyta</span>) in Response to Temperature: Implications for Invasiveness’</span>. <em>Biological Invasions</em> 17 (5): 1341–53. <a href="https://doi.org/10.1007/s10530-014-0797-8">https://doi.org/10.1007/s10530-014-0797-8</a>.
</div>
<div id="ref-zanolla_there_2019" class="csl-entry" role="listitem">
Zanolla, Marianela, M. Altamirano, F. X. Niell, and R. Carmona. 2019. <span>‘There Is More Than Meets the Eye: <span>Primary</span> Production of the Invasive Seaweed <em><span>Asparagopsis</span> Taxiformis</em> (<span>Bonnemaisoniaceae</span>, <span>Rhodophyta</span>) Is Provided by Six Cohorts with Distinctive Characteristics’</span>. <em>Aquatic Botany</em> 153: 24–28. <a href="https://doi.org/10.1016/j.aquabot.2018.11.007">https://doi.org/10.1016/j.aquabot.2018.11.007</a>.
</div>
<div id="ref-zanolla_structure_2018" class="csl-entry" role="listitem">
Zanolla, Marianela, Raquel Carmona, Julio De La Rosa, and María Altamirano. 2018. <span>‘Structure and Temporal Dynamics of a Seaweed Assemblage Dominated by the Invasive Lineage 2 of <em><span>Asparagopsis</span> Taxiformis</em> (<span>Bonnemaisoniaceae</span>, <span>Rhodophyta</span>) in the <span>Alboran</span> <span>Sea</span>’</span>. <em>Mediterranean Marine Science</em> 19 (1): 147. <a href="https://doi.org/10.12681/mms.1892">https://doi.org/10.12681/mms.1892</a>.
</div>
<div id="ref-zanolla_concise_2022" class="csl-entry" role="listitem">
Zanolla, Marianela, Raquel Carmona, Leonardo Mata, Julio De La Rosa, Alison Sherwood, Carlos Navarro Barranco, Antonio Román Muñoz, and María Altamirano. 2022. <span>‘Concise Review of the Genus <em>Asparagopsis</em> <span>Montagne</span>, 1840’</span>. <em>Journal of Applied Phycology</em> 34 (1): 1–17. <a href="https://doi.org/10.1007/s10811-021-02665-z">https://doi.org/10.1007/s10811-021-02665-z</a>.
</div>
<div id="ref-zemke-white_chlorophyte_1999" class="csl-entry" role="listitem">
Zemke-White, W. Lindsey, and K. D. Clements. 1999. <span>‘Chlorophyte and Rhodophyte Starches as Factors in Diet Choice by Marine Herbivorous Fish’</span>. <em>Journal of Experimental Marine Biology and Ecology</em> 240 (1): 137–49. <a href="https://doi.org/10.1016/S0022-0981(99)00056-8">https://doi.org/10.1016/S0022-0981(99)00056-8</a>.
</div>
</div></section></div></main>
<!-- /main column -->
<script id="quarto-html-after-body" type="application/javascript">
window.document.addEventListener("DOMContentLoaded", function (event) {
  const toggleBodyColorMode = (bsSheetEl) => {
    const mode = bsSheetEl.getAttribute("data-mode");
    const bodyEl = window.document.querySelector("body");
    if (mode === "dark") {
      bodyEl.classList.add("quarto-dark");
      bodyEl.classList.remove("quarto-light");
    } else {
      bodyEl.classList.add("quarto-light");
      bodyEl.classList.remove("quarto-dark");
    }
  }
  const toggleBodyColorPrimary = () => {
    const bsSheetEl = window.document.querySelector("link#quarto-bootstrap");
    if (bsSheetEl) {
      toggleBodyColorMode(bsSheetEl);
    }
  }
  toggleBodyColorPrimary();  
  const icon = "";
  const anchorJS = new window.AnchorJS();
  anchorJS.options = {
    placement: 'right',
    icon: icon
  };
  anchorJS.add('.anchored');
  const isCodeAnnotation = (el) => {
    for (const clz of el.classList) {
      if (clz.startsWith('code-annotation-')) {                     
        return true;
      }
    }
    return false;
  }
  const onCopySuccess = function(e) {
    // button target
    const button = e.trigger;
    // don't keep focus
    button.blur();
    // flash "checked"
    button.classList.add('code-copy-button-checked');
    var currentTitle = button.getAttribute("title");
    button.setAttribute("title", "Copied!");
    let tooltip;
    if (window.bootstrap) {
      button.setAttribute("data-bs-toggle", "tooltip");
      button.setAttribute("data-bs-placement", "left");
      button.setAttribute("data-bs-title", "Copied!");
      tooltip = new bootstrap.Tooltip(button, 
        { trigger: "manual", 
          customClass: "code-copy-button-tooltip",
          offset: [0, -8]});
      tooltip.show();    
    }
    setTimeout(function() {
      if (tooltip) {
        tooltip.hide();
        button.removeAttribute("data-bs-title");
        button.removeAttribute("data-bs-toggle");
        button.removeAttribute("data-bs-placement");
      }
      button.setAttribute("title", currentTitle);
      button.classList.remove('code-copy-button-checked');
    }, 1000);
    // clear code selection
    e.clearSelection();
  }
  const getTextToCopy = function(trigger) {
      const codeEl = trigger.previousElementSibling.cloneNode(true);
      for (const childEl of codeEl.children) {
        if (isCodeAnnotation(childEl)) {
          childEl.remove();
        }
      }
      return codeEl.innerText;
  }
  const clipboard = new window.ClipboardJS('.code-copy-button:not([data-in-quarto-modal])', {
    text: getTextToCopy
  });
  clipboard.on('success', onCopySuccess);
  if (window.document.getElementById('quarto-embedded-source-code-modal')) {
    // For code content inside modals, clipBoardJS needs to be initialized with a container option
    // TODO: Check when it could be a function (https://github.com/zenorocha/clipboard.js/issues/860)
    const clipboardModal = new window.ClipboardJS('.code-copy-button[data-in-quarto-modal]', {
      text: getTextToCopy,
      container: window.document.getElementById('quarto-embedded-source-code-modal')
    });
    clipboardModal.on('success', onCopySuccess);
  }
    var localhostRegex = new RegExp(/^(?:http|https):\/\/localhost\:?[0-9]*\//);
    var mailtoRegex = new RegExp(/^mailto:/);
      var filterRegex = new RegExp('/' + window.location.host + '/');
    var isInternal = (href) => {
        return filterRegex.test(href) || localhostRegex.test(href) || mailtoRegex.test(href);
    }
    // Inspect non-navigation links and adorn them if external
 	var links = window.document.querySelectorAll('a[href]:not(.nav-link):not(.navbar-brand):not(.toc-action):not(.sidebar-link):not(.sidebar-item-toggle):not(.pagination-link):not(.no-external):not([aria-hidden]):not(.dropdown-item):not(.quarto-navigation-tool):not(.about-link)');
    for (var i=0; i<links.length; i++) {
      const link = links[i];
      if (!isInternal(link.href)) {
        // undo the damage that might have been done by quarto-nav.js in the case of
        // links that we want to consider external
        if (link.dataset.originalHref !== undefined) {
          link.href = link.dataset.originalHref;
        }
      }
    }
  function tippyHover(el, contentFn, onTriggerFn, onUntriggerFn) {
    const config = {
      allowHTML: true,
      maxWidth: 500,
      delay: 100,
      arrow: false,
      appendTo: function(el) {
          return el.parentElement;
      },
      interactive: true,
      interactiveBorder: 10,
      theme: 'quarto',
      placement: 'bottom-start',
    };
    if (contentFn) {
      config.content = contentFn;
    }
    if (onTriggerFn) {
      config.onTrigger = onTriggerFn;
    }
    if (onUntriggerFn) {
      config.onUntrigger = onUntriggerFn;
    }
    window.tippy(el, config); 
  }
  const noterefs = window.document.querySelectorAll('a[role="doc-noteref"]');
  for (var i=0; i<noterefs.length; i++) {
    const ref = noterefs[i];
    tippyHover(ref, function() {
      // use id or data attribute instead here
      let href = ref.getAttribute('data-footnote-href') || ref.getAttribute('href');
      try { href = new URL(href).hash; } catch {}
      const id = href.replace(/^#\/?/, "");
      const note = window.document.getElementById(id);
      if (note) {
        return note.innerHTML;
      } else {
        return "";
      }
    });
  }
  const xrefs = window.document.querySelectorAll('a.quarto-xref');
  const processXRef = (id, note) => {
    // Strip column container classes
    const stripColumnClz = (el) => {
      el.classList.remove("page-full", "page-columns");
      if (el.children) {
        for (const child of el.children) {
          stripColumnClz(child);
        }
      }
    }
    stripColumnClz(note)
    if (id === null || id.startsWith('sec-')) {
      // Special case sections, only their first couple elements
      const container = document.createElement("div");
      if (note.children && note.children.length > 2) {
        container.appendChild(note.children[0].cloneNode(true));
        for (let i = 1; i < note.children.length; i++) {
          const child = note.children[i];
          if (child.tagName === "P" && child.innerText === "") {
            continue;
          } else {
            container.appendChild(child.cloneNode(true));
            break;
          }
        }
        if (window.Quarto?.typesetMath) {
          window.Quarto.typesetMath(container);
        }
        return container.innerHTML
      } else {
        if (window.Quarto?.typesetMath) {
          window.Quarto.typesetMath(note);
        }
        return note.innerHTML;
      }
    } else {
      // Remove any anchor links if they are present
      const anchorLink = note.querySelector('a.anchorjs-link');
      if (anchorLink) {
        anchorLink.remove();
      }
      if (window.Quarto?.typesetMath) {
        window.Quarto.typesetMath(note);
      }
      // TODO in 1.5, we should make sure this works without a callout special case
      if (note.classList.contains("callout")) {
        return note.outerHTML;
      } else {
        return note.innerHTML;
      }
    }
  }
  for (var i=0; i<xrefs.length; i++) {
    const xref = xrefs[i];
    tippyHover(xref, undefined, function(instance) {
      instance.disable();
      let url = xref.getAttribute('href');
      let hash = undefined; 
      if (url.startsWith('#')) {
        hash = url;
      } else {
        try { hash = new URL(url).hash; } catch {}
      }
      if (hash) {
        const id = hash.replace(/^#\/?/, "");
        const note = window.document.getElementById(id);
        if (note !== null) {
          try {
            const html = processXRef(id, note.cloneNode(true));
            instance.setContent(html);
          } finally {
            instance.enable();
            instance.show();
          }
        } else {
          // See if we can fetch this
          fetch(url.split('#')[0])
          .then(res => res.text())
          .then(html => {
            const parser = new DOMParser();
            const htmlDoc = parser.parseFromString(html, "text/html");
            const note = htmlDoc.getElementById(id);
            if (note !== null) {
              const html = processXRef(id, note);
              instance.setContent(html);
            } 
          }).finally(() => {
            instance.enable();
            instance.show();
          });
        }
      } else {
        // See if we can fetch a full url (with no hash to target)
        // This is a special case and we should probably do some content thinning / targeting
        fetch(url)
        .then(res => res.text())
        .then(html => {
          const parser = new DOMParser();
          const htmlDoc = parser.parseFromString(html, "text/html");
          const note = htmlDoc.querySelector('main.content');
          if (note !== null) {
            // This should only happen for chapter cross references
            // (since there is no id in the URL)
            // remove the first header
            if (note.children.length > 0 && note.children[0].tagName === "HEADER") {
              note.children[0].remove();
            }
            const html = processXRef(null, note);
            instance.setContent(html);
          } 
        }).finally(() => {
          instance.enable();
          instance.show();
        });
      }
    }, function(instance) {
    });
  }
      let selectedAnnoteEl;
      const selectorForAnnotation = ( cell, annotation) => {
        let cellAttr = 'data-code-cell="' + cell + '"';
        let lineAttr = 'data-code-annotation="' +  annotation + '"';
        const selector = 'span[' + cellAttr + '][' + lineAttr + ']';
        return selector;
      }
      const selectCodeLines = (annoteEl) => {
        const doc = window.document;
        const targetCell = annoteEl.getAttribute("data-target-cell");
        const targetAnnotation = annoteEl.getAttribute("data-target-annotation");
        const annoteSpan = window.document.querySelector(selectorForAnnotation(targetCell, targetAnnotation));
        const lines = annoteSpan.getAttribute("data-code-lines").split(",");
        const lineIds = lines.map((line) => {
          return targetCell + "-" + line;
        })
        let top = null;
        let height = null;
        let parent = null;
        if (lineIds.length > 0) {
            //compute the position of the single el (top and bottom and make a div)
            const el = window.document.getElementById(lineIds[0]);
            top = el.offsetTop;
            height = el.offsetHeight;
            parent = el.parentElement.parentElement;
          if (lineIds.length > 1) {
            const lastEl = window.document.getElementById(lineIds[lineIds.length - 1]);
            const bottom = lastEl.offsetTop + lastEl.offsetHeight;
            height = bottom - top;
          }
          if (top !== null && height !== null && parent !== null) {
            // cook up a div (if necessary) and position it 
            let div = window.document.getElementById("code-annotation-line-highlight");
            if (div === null) {
              div = window.document.createElement("div");
              div.setAttribute("id", "code-annotation-line-highlight");
              div.style.position = 'absolute';
              parent.appendChild(div);
            }
            div.style.top = top - 2 + "px";
            div.style.height = height + 4 + "px";
            div.style.left = 0;
            let gutterDiv = window.document.getElementById("code-annotation-line-highlight-gutter");
            if (gutterDiv === null) {
              gutterDiv = window.document.createElement("div");
              gutterDiv.setAttribute("id", "code-annotation-line-highlight-gutter");
              gutterDiv.style.position = 'absolute';
              const codeCell = window.document.getElementById(targetCell);
              const gutter = codeCell.querySelector('.code-annotation-gutter');
              gutter.appendChild(gutterDiv);
            }
            gutterDiv.style.top = top - 2 + "px";
            gutterDiv.style.height = height + 4 + "px";
          }
          selectedAnnoteEl = annoteEl;
        }
      };
      const unselectCodeLines = () => {
        const elementsIds = ["code-annotation-line-highlight", "code-annotation-line-highlight-gutter"];
        elementsIds.forEach((elId) => {
          const div = window.document.getElementById(elId);
          if (div) {
            div.remove();
          }
        });
        selectedAnnoteEl = undefined;
      };
        // Handle positioning of the toggle
    window.addEventListener(
      "resize",
      throttle(() => {
        elRect = undefined;
        if (selectedAnnoteEl) {
          selectCodeLines(selectedAnnoteEl);
        }
      }, 10)
    );
    function throttle(fn, ms) {
    let throttle = false;
    let timer;
      return (...args) => {
        if(!throttle) { // first call gets through
            fn.apply(this, args);
            throttle = true;
        } else { // all the others get throttled
            if(timer) clearTimeout(timer); // cancel #2
            timer = setTimeout(() => {
              fn.apply(this, args);
              timer = throttle = false;
            }, ms);
        }
      };
    }
      // Attach click handler to the DT
      const annoteDls = window.document.querySelectorAll('dt[data-target-cell]');
      for (const annoteDlNode of annoteDls) {
        annoteDlNode.addEventListener('click', (event) => {
          const clickedEl = event.target;
          if (clickedEl !== selectedAnnoteEl) {
            unselectCodeLines();
            const activeEl = window.document.querySelector('dt[data-target-cell].code-annotation-active');
            if (activeEl) {
              activeEl.classList.remove('code-annotation-active');
            }
            selectCodeLines(clickedEl);
            clickedEl.classList.add('code-annotation-active');
          } else {
            // Unselect the line
            unselectCodeLines();
            clickedEl.classList.remove('code-annotation-active');
          }
        });
      }
  const findCites = (el) => {
    const parentEl = el.parentElement;
    if (parentEl) {
      const cites = parentEl.dataset.cites;
      if (cites) {
        return {
          el,
          cites: cites.split(' ')
        };
      } else {
        return findCites(el.parentElement)
      }
    } else {
      return undefined;
    }
  };
  var bibliorefs = window.document.querySelectorAll('a[role="doc-biblioref"]');
  for (var i=0; i<bibliorefs.length; i++) {
    const ref = bibliorefs[i];
    const citeInfo = findCites(ref);
    if (citeInfo) {
      tippyHover(citeInfo.el, function() {
        var popup = window.document.createElement('div');
        citeInfo.cites.forEach(function(cite) {
          var citeDiv = window.document.createElement('div');
          citeDiv.classList.add('hanging-indent');
          citeDiv.classList.add('csl-entry');
          var biblioDiv = window.document.getElementById('ref-' + cite);
          if (biblioDiv) {
            citeDiv.innerHTML = biblioDiv.innerHTML;
          }
          popup.appendChild(citeDiv);
        });
        return popup.innerHTML;
      });
    }
  }
});
</script>
</div> <!-- /content -->




</body></html>